// @aws-sdk/client-cloudformation/dist-cjs/index.js
var __exportStar = (m, e) => { Object.assign(e, m); };
const { getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, Command, createWaiter, checkExceptions, WaiterState, createAggregatedClient } = require("@smithy/core/client");
exports.$Command = Command;
exports.__Client = Client;
const { resolveRegionConfig } = require("@smithy/core/config");
const { resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveHttpAuthSchemeConfig, defaultCloudFormationHttpAuthSchemeParametersProvider } = require("./auth/httpAuthSchemeProvider");
const { getRuntimeConfig } = require("./runtimeConfig");
const { ActivateOrganizationsAccess$, ActivateType$, BatchDescribeTypeConfigurations$, CancelUpdateStack$, ContinueUpdateRollback$, CreateChangeSet$, CreateGeneratedTemplate$, CreateStack$, CreateStackInstances$, CreateStackRefactor$, CreateStackSet$, DeactivateOrganizationsAccess$, DeactivateType$, DeleteChangeSet$, DeleteGeneratedTemplate$, DeleteStack$, DeleteStackInstances$, DeleteStackSet$, DeregisterType$, DescribeAccountLimits$, DescribeChangeSet$, DescribeChangeSetHooks$, DescribeEvents$, DescribeGeneratedTemplate$, DescribeOrganizationsAccess$, DescribePublisher$, DescribeResourceScan$, DescribeStackDriftDetectionStatus$, DescribeStackEvents$, DescribeStackInstance$, DescribeStackRefactor$, DescribeStackResource$, DescribeStackResourceDrifts$, DescribeStackResources$, DescribeStacks$, DescribeStackSet$, DescribeStackSetOperation$, DescribeType$, DescribeTypeRegistration$, DetectStackDrift$, DetectStackResourceDrift$, DetectStackSetDrift$, EstimateTemplateCost$, ExecuteChangeSet$, ExecuteStackRefactor$, GetGeneratedTemplate$, GetHookResult$, GetStackPolicy$, GetTemplate$, GetTemplateSummary$, ImportStacksToStackSet$, ListChangeSets$, ListExports$, ListGeneratedTemplates$, ListHookResults$, ListImports$, ListResourceScanRelatedResources$, ListResourceScanResources$, ListResourceScans$, ListStackInstanceResourceDrifts$, ListStackInstances$, ListStackRefactorActions$, ListStackRefactors$, ListStackResources$, ListStacks$, ListStackSetAutoDeploymentTargets$, ListStackSetOperationResults$, ListStackSetOperations$, ListStackSets$, ListTypeRegistrations$, ListTypes$, ListTypeVersions$, PublishType$, RecordHandlerProgress$, RegisterPublisher$, RegisterType$, RollbackStack$, SetStackPolicy$, SetTypeConfiguration$, SetTypeDefaultVersion$, SignalResource$, StartResourceScan$, StopStackSetOperation$, TestType$, UpdateGeneratedTemplate$, UpdateStack$, UpdateStackInstances$, UpdateStackSet$, UpdateTerminationProtection$, ValidateTemplate$ } = require("./schemas/schemas_0");
__exportStar(require("./schemas/schemas_0"), exports);
__exportStar(require("./models/errors"), exports);
const { CloudFormationServiceException } = require("./models/CloudFormationServiceException");
exports.CloudFormationServiceException = CloudFormationServiceException;

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "cloudformation",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class CloudFormationClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultCloudFormationHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class ActivateOrganizationsAccessCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ActivateOrganizationsAccess", {})
    .n("CloudFormationClient", "ActivateOrganizationsAccessCommand")
    .sc(ActivateOrganizationsAccess$)
    .build() {
}

class ActivateTypeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ActivateType", {})
    .n("CloudFormationClient", "ActivateTypeCommand")
    .sc(ActivateType$)
    .build() {
}

class BatchDescribeTypeConfigurationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "BatchDescribeTypeConfigurations", {})
    .n("CloudFormationClient", "BatchDescribeTypeConfigurationsCommand")
    .sc(BatchDescribeTypeConfigurations$)
    .build() {
}

class CancelUpdateStackCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "CancelUpdateStack", {})
    .n("CloudFormationClient", "CancelUpdateStackCommand")
    .sc(CancelUpdateStack$)
    .build() {
}

class ContinueUpdateRollbackCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ContinueUpdateRollback", {})
    .n("CloudFormationClient", "ContinueUpdateRollbackCommand")
    .sc(ContinueUpdateRollback$)
    .build() {
}

class CreateChangeSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "CreateChangeSet", {})
    .n("CloudFormationClient", "CreateChangeSetCommand")
    .sc(CreateChangeSet$)
    .build() {
}

class CreateGeneratedTemplateCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "CreateGeneratedTemplate", {})
    .n("CloudFormationClient", "CreateGeneratedTemplateCommand")
    .sc(CreateGeneratedTemplate$)
    .build() {
}

class CreateStackCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "CreateStack", {})
    .n("CloudFormationClient", "CreateStackCommand")
    .sc(CreateStack$)
    .build() {
}

class CreateStackInstancesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "CreateStackInstances", {})
    .n("CloudFormationClient", "CreateStackInstancesCommand")
    .sc(CreateStackInstances$)
    .build() {
}

class CreateStackRefactorCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "CreateStackRefactor", {})
    .n("CloudFormationClient", "CreateStackRefactorCommand")
    .sc(CreateStackRefactor$)
    .build() {
}

class CreateStackSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "CreateStackSet", {})
    .n("CloudFormationClient", "CreateStackSetCommand")
    .sc(CreateStackSet$)
    .build() {
}

class DeactivateOrganizationsAccessCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DeactivateOrganizationsAccess", {})
    .n("CloudFormationClient", "DeactivateOrganizationsAccessCommand")
    .sc(DeactivateOrganizationsAccess$)
    .build() {
}

class DeactivateTypeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DeactivateType", {})
    .n("CloudFormationClient", "DeactivateTypeCommand")
    .sc(DeactivateType$)
    .build() {
}

class DeleteChangeSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DeleteChangeSet", {})
    .n("CloudFormationClient", "DeleteChangeSetCommand")
    .sc(DeleteChangeSet$)
    .build() {
}

class DeleteGeneratedTemplateCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DeleteGeneratedTemplate", {})
    .n("CloudFormationClient", "DeleteGeneratedTemplateCommand")
    .sc(DeleteGeneratedTemplate$)
    .build() {
}

class DeleteStackCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DeleteStack", {})
    .n("CloudFormationClient", "DeleteStackCommand")
    .sc(DeleteStack$)
    .build() {
}

class DeleteStackInstancesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DeleteStackInstances", {})
    .n("CloudFormationClient", "DeleteStackInstancesCommand")
    .sc(DeleteStackInstances$)
    .build() {
}

class DeleteStackSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DeleteStackSet", {})
    .n("CloudFormationClient", "DeleteStackSetCommand")
    .sc(DeleteStackSet$)
    .build() {
}

class DeregisterTypeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DeregisterType", {})
    .n("CloudFormationClient", "DeregisterTypeCommand")
    .sc(DeregisterType$)
    .build() {
}

class DescribeAccountLimitsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeAccountLimits", {})
    .n("CloudFormationClient", "DescribeAccountLimitsCommand")
    .sc(DescribeAccountLimits$)
    .build() {
}

class DescribeChangeSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeChangeSet", {})
    .n("CloudFormationClient", "DescribeChangeSetCommand")
    .sc(DescribeChangeSet$)
    .build() {
}

class DescribeChangeSetHooksCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeChangeSetHooks", {})
    .n("CloudFormationClient", "DescribeChangeSetHooksCommand")
    .sc(DescribeChangeSetHooks$)
    .build() {
}

class DescribeEventsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeEvents", {})
    .n("CloudFormationClient", "DescribeEventsCommand")
    .sc(DescribeEvents$)
    .build() {
}

class DescribeGeneratedTemplateCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeGeneratedTemplate", {})
    .n("CloudFormationClient", "DescribeGeneratedTemplateCommand")
    .sc(DescribeGeneratedTemplate$)
    .build() {
}

class DescribeOrganizationsAccessCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeOrganizationsAccess", {})
    .n("CloudFormationClient", "DescribeOrganizationsAccessCommand")
    .sc(DescribeOrganizationsAccess$)
    .build() {
}

class DescribePublisherCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribePublisher", {})
    .n("CloudFormationClient", "DescribePublisherCommand")
    .sc(DescribePublisher$)
    .build() {
}

class DescribeResourceScanCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeResourceScan", {})
    .n("CloudFormationClient", "DescribeResourceScanCommand")
    .sc(DescribeResourceScan$)
    .build() {
}

class DescribeStackDriftDetectionStatusCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackDriftDetectionStatus", {})
    .n("CloudFormationClient", "DescribeStackDriftDetectionStatusCommand")
    .sc(DescribeStackDriftDetectionStatus$)
    .build() {
}

class DescribeStackEventsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackEvents", {})
    .n("CloudFormationClient", "DescribeStackEventsCommand")
    .sc(DescribeStackEvents$)
    .build() {
}

class DescribeStackInstanceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackInstance", {})
    .n("CloudFormationClient", "DescribeStackInstanceCommand")
    .sc(DescribeStackInstance$)
    .build() {
}

class DescribeStackRefactorCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackRefactor", {})
    .n("CloudFormationClient", "DescribeStackRefactorCommand")
    .sc(DescribeStackRefactor$)
    .build() {
}

class DescribeStackResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackResource", {})
    .n("CloudFormationClient", "DescribeStackResourceCommand")
    .sc(DescribeStackResource$)
    .build() {
}

class DescribeStackResourceDriftsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackResourceDrifts", {})
    .n("CloudFormationClient", "DescribeStackResourceDriftsCommand")
    .sc(DescribeStackResourceDrifts$)
    .build() {
}

class DescribeStackResourcesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackResources", {})
    .n("CloudFormationClient", "DescribeStackResourcesCommand")
    .sc(DescribeStackResources$)
    .build() {
}

class DescribeStacksCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStacks", {})
    .n("CloudFormationClient", "DescribeStacksCommand")
    .sc(DescribeStacks$)
    .build() {
}

class DescribeStackSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackSet", {})
    .n("CloudFormationClient", "DescribeStackSetCommand")
    .sc(DescribeStackSet$)
    .build() {
}

class DescribeStackSetOperationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeStackSetOperation", {})
    .n("CloudFormationClient", "DescribeStackSetOperationCommand")
    .sc(DescribeStackSetOperation$)
    .build() {
}

class DescribeTypeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeType", {})
    .n("CloudFormationClient", "DescribeTypeCommand")
    .sc(DescribeType$)
    .build() {
}

class DescribeTypeRegistrationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DescribeTypeRegistration", {})
    .n("CloudFormationClient", "DescribeTypeRegistrationCommand")
    .sc(DescribeTypeRegistration$)
    .build() {
}

class DetectStackDriftCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DetectStackDrift", {})
    .n("CloudFormationClient", "DetectStackDriftCommand")
    .sc(DetectStackDrift$)
    .build() {
}

class DetectStackResourceDriftCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DetectStackResourceDrift", {})
    .n("CloudFormationClient", "DetectStackResourceDriftCommand")
    .sc(DetectStackResourceDrift$)
    .build() {
}

class DetectStackSetDriftCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "DetectStackSetDrift", {})
    .n("CloudFormationClient", "DetectStackSetDriftCommand")
    .sc(DetectStackSetDrift$)
    .build() {
}

class EstimateTemplateCostCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "EstimateTemplateCost", {})
    .n("CloudFormationClient", "EstimateTemplateCostCommand")
    .sc(EstimateTemplateCost$)
    .build() {
}

class ExecuteChangeSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ExecuteChangeSet", {})
    .n("CloudFormationClient", "ExecuteChangeSetCommand")
    .sc(ExecuteChangeSet$)
    .build() {
}

class ExecuteStackRefactorCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ExecuteStackRefactor", {})
    .n("CloudFormationClient", "ExecuteStackRefactorCommand")
    .sc(ExecuteStackRefactor$)
    .build() {
}

class GetGeneratedTemplateCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "GetGeneratedTemplate", {})
    .n("CloudFormationClient", "GetGeneratedTemplateCommand")
    .sc(GetGeneratedTemplate$)
    .build() {
}

class GetHookResultCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "GetHookResult", {})
    .n("CloudFormationClient", "GetHookResultCommand")
    .sc(GetHookResult$)
    .build() {
}

class GetStackPolicyCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "GetStackPolicy", {})
    .n("CloudFormationClient", "GetStackPolicyCommand")
    .sc(GetStackPolicy$)
    .build() {
}

class GetTemplateCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "GetTemplate", {})
    .n("CloudFormationClient", "GetTemplateCommand")
    .sc(GetTemplate$)
    .build() {
}

class GetTemplateSummaryCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "GetTemplateSummary", {})
    .n("CloudFormationClient", "GetTemplateSummaryCommand")
    .sc(GetTemplateSummary$)
    .build() {
}

class ImportStacksToStackSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ImportStacksToStackSet", {})
    .n("CloudFormationClient", "ImportStacksToStackSetCommand")
    .sc(ImportStacksToStackSet$)
    .build() {
}

class ListChangeSetsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListChangeSets", {})
    .n("CloudFormationClient", "ListChangeSetsCommand")
    .sc(ListChangeSets$)
    .build() {
}

class ListExportsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListExports", {})
    .n("CloudFormationClient", "ListExportsCommand")
    .sc(ListExports$)
    .build() {
}

class ListGeneratedTemplatesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListGeneratedTemplates", {})
    .n("CloudFormationClient", "ListGeneratedTemplatesCommand")
    .sc(ListGeneratedTemplates$)
    .build() {
}

class ListHookResultsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListHookResults", {})
    .n("CloudFormationClient", "ListHookResultsCommand")
    .sc(ListHookResults$)
    .build() {
}

class ListImportsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListImports", {})
    .n("CloudFormationClient", "ListImportsCommand")
    .sc(ListImports$)
    .build() {
}

class ListResourceScanRelatedResourcesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListResourceScanRelatedResources", {})
    .n("CloudFormationClient", "ListResourceScanRelatedResourcesCommand")
    .sc(ListResourceScanRelatedResources$)
    .build() {
}

class ListResourceScanResourcesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListResourceScanResources", {})
    .n("CloudFormationClient", "ListResourceScanResourcesCommand")
    .sc(ListResourceScanResources$)
    .build() {
}

class ListResourceScansCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListResourceScans", {})
    .n("CloudFormationClient", "ListResourceScansCommand")
    .sc(ListResourceScans$)
    .build() {
}

class ListStackInstanceResourceDriftsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackInstanceResourceDrifts", {})
    .n("CloudFormationClient", "ListStackInstanceResourceDriftsCommand")
    .sc(ListStackInstanceResourceDrifts$)
    .build() {
}

class ListStackInstancesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackInstances", {})
    .n("CloudFormationClient", "ListStackInstancesCommand")
    .sc(ListStackInstances$)
    .build() {
}

class ListStackRefactorActionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackRefactorActions", {})
    .n("CloudFormationClient", "ListStackRefactorActionsCommand")
    .sc(ListStackRefactorActions$)
    .build() {
}

class ListStackRefactorsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackRefactors", {})
    .n("CloudFormationClient", "ListStackRefactorsCommand")
    .sc(ListStackRefactors$)
    .build() {
}

class ListStackResourcesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackResources", {})
    .n("CloudFormationClient", "ListStackResourcesCommand")
    .sc(ListStackResources$)
    .build() {
}

class ListStacksCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStacks", {})
    .n("CloudFormationClient", "ListStacksCommand")
    .sc(ListStacks$)
    .build() {
}

class ListStackSetAutoDeploymentTargetsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackSetAutoDeploymentTargets", {})
    .n("CloudFormationClient", "ListStackSetAutoDeploymentTargetsCommand")
    .sc(ListStackSetAutoDeploymentTargets$)
    .build() {
}

class ListStackSetOperationResultsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackSetOperationResults", {})
    .n("CloudFormationClient", "ListStackSetOperationResultsCommand")
    .sc(ListStackSetOperationResults$)
    .build() {
}

class ListStackSetOperationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackSetOperations", {})
    .n("CloudFormationClient", "ListStackSetOperationsCommand")
    .sc(ListStackSetOperations$)
    .build() {
}

class ListStackSetsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListStackSets", {})
    .n("CloudFormationClient", "ListStackSetsCommand")
    .sc(ListStackSets$)
    .build() {
}

class ListTypeRegistrationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListTypeRegistrations", {})
    .n("CloudFormationClient", "ListTypeRegistrationsCommand")
    .sc(ListTypeRegistrations$)
    .build() {
}

class ListTypesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListTypes", {})
    .n("CloudFormationClient", "ListTypesCommand")
    .sc(ListTypes$)
    .build() {
}

class ListTypeVersionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ListTypeVersions", {})
    .n("CloudFormationClient", "ListTypeVersionsCommand")
    .sc(ListTypeVersions$)
    .build() {
}

class PublishTypeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "PublishType", {})
    .n("CloudFormationClient", "PublishTypeCommand")
    .sc(PublishType$)
    .build() {
}

class RecordHandlerProgressCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "RecordHandlerProgress", {})
    .n("CloudFormationClient", "RecordHandlerProgressCommand")
    .sc(RecordHandlerProgress$)
    .build() {
}

class RegisterPublisherCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "RegisterPublisher", {})
    .n("CloudFormationClient", "RegisterPublisherCommand")
    .sc(RegisterPublisher$)
    .build() {
}

class RegisterTypeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "RegisterType", {})
    .n("CloudFormationClient", "RegisterTypeCommand")
    .sc(RegisterType$)
    .build() {
}

class RollbackStackCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "RollbackStack", {})
    .n("CloudFormationClient", "RollbackStackCommand")
    .sc(RollbackStack$)
    .build() {
}

class SetStackPolicyCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "SetStackPolicy", {})
    .n("CloudFormationClient", "SetStackPolicyCommand")
    .sc(SetStackPolicy$)
    .build() {
}

class SetTypeConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "SetTypeConfiguration", {})
    .n("CloudFormationClient", "SetTypeConfigurationCommand")
    .sc(SetTypeConfiguration$)
    .build() {
}

class SetTypeDefaultVersionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "SetTypeDefaultVersion", {})
    .n("CloudFormationClient", "SetTypeDefaultVersionCommand")
    .sc(SetTypeDefaultVersion$)
    .build() {
}

class SignalResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "SignalResource", {})
    .n("CloudFormationClient", "SignalResourceCommand")
    .sc(SignalResource$)
    .build() {
}

class StartResourceScanCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "StartResourceScan", {})
    .n("CloudFormationClient", "StartResourceScanCommand")
    .sc(StartResourceScan$)
    .build() {
}

class StopStackSetOperationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "StopStackSetOperation", {})
    .n("CloudFormationClient", "StopStackSetOperationCommand")
    .sc(StopStackSetOperation$)
    .build() {
}

class TestTypeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "TestType", {})
    .n("CloudFormationClient", "TestTypeCommand")
    .sc(TestType$)
    .build() {
}

class UpdateGeneratedTemplateCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "UpdateGeneratedTemplate", {})
    .n("CloudFormationClient", "UpdateGeneratedTemplateCommand")
    .sc(UpdateGeneratedTemplate$)
    .build() {
}

class UpdateStackCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "UpdateStack", {})
    .n("CloudFormationClient", "UpdateStackCommand")
    .sc(UpdateStack$)
    .build() {
}

class UpdateStackInstancesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "UpdateStackInstances", {})
    .n("CloudFormationClient", "UpdateStackInstancesCommand")
    .sc(UpdateStackInstances$)
    .build() {
}

class UpdateStackSetCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "UpdateStackSet", {})
    .n("CloudFormationClient", "UpdateStackSetCommand")
    .sc(UpdateStackSet$)
    .build() {
}

class UpdateTerminationProtectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "UpdateTerminationProtection", {})
    .n("CloudFormationClient", "UpdateTerminationProtectionCommand")
    .sc(UpdateTerminationProtection$)
    .build() {
}

class ValidateTemplateCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CloudFormation", "ValidateTemplate", {})
    .n("CloudFormationClient", "ValidateTemplateCommand")
    .sc(ValidateTemplate$)
    .build() {
}

const paginateDescribeAccountLimits = createPaginator(CloudFormationClient, DescribeAccountLimitsCommand, "NextToken", "NextToken", "");

const paginateDescribeChangeSet = createPaginator(CloudFormationClient, DescribeChangeSetCommand, "NextToken", "NextToken", "");

const paginateDescribeEvents = createPaginator(CloudFormationClient, DescribeEventsCommand, "NextToken", "NextToken", "");

const paginateDescribeStackEvents = createPaginator(CloudFormationClient, DescribeStackEventsCommand, "NextToken", "NextToken", "");

const paginateDescribeStackResourceDrifts = createPaginator(CloudFormationClient, DescribeStackResourceDriftsCommand, "NextToken", "NextToken", "MaxResults");

const paginateDescribeStacks = createPaginator(CloudFormationClient, DescribeStacksCommand, "NextToken", "NextToken", "");

const paginateListChangeSets = createPaginator(CloudFormationClient, ListChangeSetsCommand, "NextToken", "NextToken", "");

const paginateListExports = createPaginator(CloudFormationClient, ListExportsCommand, "NextToken", "NextToken", "");

const paginateListGeneratedTemplates = createPaginator(CloudFormationClient, ListGeneratedTemplatesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListImports = createPaginator(CloudFormationClient, ListImportsCommand, "NextToken", "NextToken", "");

const paginateListResourceScanRelatedResources = createPaginator(CloudFormationClient, ListResourceScanRelatedResourcesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListResourceScanResources = createPaginator(CloudFormationClient, ListResourceScanResourcesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListResourceScans = createPaginator(CloudFormationClient, ListResourceScansCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStackInstances = createPaginator(CloudFormationClient, ListStackInstancesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStackRefactorActions = createPaginator(CloudFormationClient, ListStackRefactorActionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStackRefactors = createPaginator(CloudFormationClient, ListStackRefactorsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStackResources = createPaginator(CloudFormationClient, ListStackResourcesCommand, "NextToken", "NextToken", "");

const paginateListStackSetOperationResults = createPaginator(CloudFormationClient, ListStackSetOperationResultsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStackSetOperations = createPaginator(CloudFormationClient, ListStackSetOperationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStackSets = createPaginator(CloudFormationClient, ListStackSetsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStacks = createPaginator(CloudFormationClient, ListStacksCommand, "NextToken", "NextToken", "");

const paginateListTypeRegistrations = createPaginator(CloudFormationClient, ListTypeRegistrationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTypes = createPaginator(CloudFormationClient, ListTypesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTypeVersions = createPaginator(CloudFormationClient, ListTypeVersionsCommand, "NextToken", "NextToken", "MaxResults");

const checkState$9 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeChangeSetCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.Status;
            };
            if (returnComparator() === "CREATE_COMPLETE") {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.Status;
            };
            if (returnComparator() === "FAILED") {
                return { state: WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.FAILURE, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForChangeSetCreateComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$9);
};
const waitUntilChangeSetCreateComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$9);
    return checkExceptions(result);
};

const checkState$8 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeStacksCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "CREATE_COMPLETE");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_COMPLETE");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_IN_PROGRESS");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_FAILED");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_ROLLBACK_IN_PROGRESS");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_ROLLBACK_FAILED");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_ROLLBACK_COMPLETE");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "CREATE_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "DELETE_COMPLETE") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "DELETE_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "ROLLBACK_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "ROLLBACK_COMPLETE") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.FAILURE, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForStackCreateComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$8);
};
const waitUntilStackCreateComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$8);
    return checkExceptions(result);
};

const checkState$7 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeStacksCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "DELETE_COMPLETE");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "DELETE_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "CREATE_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "ROLLBACK_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_ROLLBACK_IN_PROGRESS") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_ROLLBACK_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_ROLLBACK_COMPLETE") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_COMPLETE") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.SUCCESS, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForStackDeleteComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$7);
};
const waitUntilStackDeleteComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$7);
    return checkExceptions(result);
};

const checkState$6 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeStacksCommand(input));
        reason = result;
        return { state: WaiterState.SUCCESS, reason };
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.RETRY, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForStackExists = async (params, input) => {
    const serviceDefaults = { minDelay: 5, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$6);
};
const waitUntilStackExists = async (params, input) => {
    const serviceDefaults = { minDelay: 5, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$6);
    return checkExceptions(result);
};

const checkState$5 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeStacksCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "IMPORT_COMPLETE");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "ROLLBACK_COMPLETE") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "ROLLBACK_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "IMPORT_ROLLBACK_IN_PROGRESS") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "IMPORT_ROLLBACK_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "IMPORT_ROLLBACK_COMPLETE") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.FAILURE, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForStackImportComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$5);
};
const waitUntilStackImportComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$5);
    return checkExceptions(result);
};

const checkState$4 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeStackRefactorCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.Status;
            };
            if (returnComparator() === "CREATE_COMPLETE") {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.Status;
            };
            if (returnComparator() === "CREATE_FAILED") {
                return { state: WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.FAILURE, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForStackRefactorCreateComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 5, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$4);
};
const waitUntilStackRefactorCreateComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 5, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$4);
    return checkExceptions(result);
};

const checkState$3 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeStackRefactorCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.ExecutionStatus;
            };
            if (returnComparator() === "EXECUTE_COMPLETE") {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.ExecutionStatus;
            };
            if (returnComparator() === "EXECUTE_FAILED") {
                return { state: WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.ExecutionStatus;
            };
            if (returnComparator() === "ROLLBACK_COMPLETE") {
                return { state: WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.FAILURE, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForStackRefactorExecuteComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$3);
};
const waitUntilStackRefactorExecuteComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$3);
    return checkExceptions(result);
};

const checkState$2 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeStacksCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_ROLLBACK_COMPLETE");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_ROLLBACK_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "DELETE_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.FAILURE, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForStackRollbackComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$2);
};
const waitUntilStackRollbackComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$2);
    return checkExceptions(result);
};

const checkState$1 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeStacksCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "UPDATE_COMPLETE");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_ROLLBACK_FAILED") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.Stacks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.StackStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "UPDATE_ROLLBACK_COMPLETE") {
                    return { state: WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "ValidationError") {
            return { state: WaiterState.FAILURE, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForStackUpdateComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
};
const waitUntilStackUpdateComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
    return checkExceptions(result);
};

const checkState = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeTypeRegistrationCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.ProgressStatus;
            };
            if (returnComparator() === "COMPLETE") {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.ProgressStatus;
            };
            if (returnComparator() === "FAILED") {
                return { state: WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForTypeRegistrationComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilTypeRegistrationComplete = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return checkExceptions(result);
};

const commands = {
    ActivateOrganizationsAccessCommand,
    ActivateTypeCommand,
    BatchDescribeTypeConfigurationsCommand,
    CancelUpdateStackCommand,
    ContinueUpdateRollbackCommand,
    CreateChangeSetCommand,
    CreateGeneratedTemplateCommand,
    CreateStackCommand,
    CreateStackInstancesCommand,
    CreateStackRefactorCommand,
    CreateStackSetCommand,
    DeactivateOrganizationsAccessCommand,
    DeactivateTypeCommand,
    DeleteChangeSetCommand,
    DeleteGeneratedTemplateCommand,
    DeleteStackCommand,
    DeleteStackInstancesCommand,
    DeleteStackSetCommand,
    DeregisterTypeCommand,
    DescribeAccountLimitsCommand,
    DescribeChangeSetCommand,
    DescribeChangeSetHooksCommand,
    DescribeEventsCommand,
    DescribeGeneratedTemplateCommand,
    DescribeOrganizationsAccessCommand,
    DescribePublisherCommand,
    DescribeResourceScanCommand,
    DescribeStackDriftDetectionStatusCommand,
    DescribeStackEventsCommand,
    DescribeStackInstanceCommand,
    DescribeStackRefactorCommand,
    DescribeStackResourceCommand,
    DescribeStackResourceDriftsCommand,
    DescribeStackResourcesCommand,
    DescribeStacksCommand,
    DescribeStackSetCommand,
    DescribeStackSetOperationCommand,
    DescribeTypeCommand,
    DescribeTypeRegistrationCommand,
    DetectStackDriftCommand,
    DetectStackResourceDriftCommand,
    DetectStackSetDriftCommand,
    EstimateTemplateCostCommand,
    ExecuteChangeSetCommand,
    ExecuteStackRefactorCommand,
    GetGeneratedTemplateCommand,
    GetHookResultCommand,
    GetStackPolicyCommand,
    GetTemplateCommand,
    GetTemplateSummaryCommand,
    ImportStacksToStackSetCommand,
    ListChangeSetsCommand,
    ListExportsCommand,
    ListGeneratedTemplatesCommand,
    ListHookResultsCommand,
    ListImportsCommand,
    ListResourceScanRelatedResourcesCommand,
    ListResourceScanResourcesCommand,
    ListResourceScansCommand,
    ListStackInstanceResourceDriftsCommand,
    ListStackInstancesCommand,
    ListStackRefactorActionsCommand,
    ListStackRefactorsCommand,
    ListStackResourcesCommand,
    ListStacksCommand,
    ListStackSetAutoDeploymentTargetsCommand,
    ListStackSetOperationResultsCommand,
    ListStackSetOperationsCommand,
    ListStackSetsCommand,
    ListTypeRegistrationsCommand,
    ListTypesCommand,
    ListTypeVersionsCommand,
    PublishTypeCommand,
    RecordHandlerProgressCommand,
    RegisterPublisherCommand,
    RegisterTypeCommand,
    RollbackStackCommand,
    SetStackPolicyCommand,
    SetTypeConfigurationCommand,
    SetTypeDefaultVersionCommand,
    SignalResourceCommand,
    StartResourceScanCommand,
    StopStackSetOperationCommand,
    TestTypeCommand,
    UpdateGeneratedTemplateCommand,
    UpdateStackCommand,
    UpdateStackInstancesCommand,
    UpdateStackSetCommand,
    UpdateTerminationProtectionCommand,
    ValidateTemplateCommand,
};
const paginators = {
    paginateDescribeAccountLimits,
    paginateDescribeChangeSet,
    paginateDescribeEvents,
    paginateDescribeStackEvents,
    paginateDescribeStackResourceDrifts,
    paginateDescribeStacks,
    paginateListChangeSets,
    paginateListExports,
    paginateListGeneratedTemplates,
    paginateListImports,
    paginateListResourceScanRelatedResources,
    paginateListResourceScanResources,
    paginateListResourceScans,
    paginateListStackInstances,
    paginateListStackRefactorActions,
    paginateListStackRefactors,
    paginateListStackResources,
    paginateListStacks,
    paginateListStackSetOperationResults,
    paginateListStackSetOperations,
    paginateListStackSets,
    paginateListTypeRegistrations,
    paginateListTypes,
    paginateListTypeVersions,
};
const waiters = {
    waitUntilChangeSetCreateComplete,
    waitUntilStackRefactorCreateComplete,
    waitUntilStackRefactorExecuteComplete,
    waitUntilStackCreateComplete,
    waitUntilStackDeleteComplete,
    waitUntilStackExists,
    waitUntilStackImportComplete,
    waitUntilStackRollbackComplete,
    waitUntilStackUpdateComplete,
    waitUntilTypeRegistrationComplete,
};
class CloudFormation extends CloudFormationClient {
}
createAggregatedClient(commands, CloudFormation, { paginators, waiters });

const AccountFilterType = {
    DIFFERENCE: "DIFFERENCE",
    INTERSECTION: "INTERSECTION",
    NONE: "NONE",
    UNION: "UNION",
};
const AccountGateStatus = {
    FAILED: "FAILED",
    SKIPPED: "SKIPPED",
    SUCCEEDED: "SUCCEEDED",
};
const ThirdPartyType = {
    HOOK: "HOOK",
    MODULE: "MODULE",
    RESOURCE: "RESOURCE",
};
const VersionBump = {
    MAJOR: "MAJOR",
    MINOR: "MINOR",
};
const AfterValueFrom = {
    TEMPLATE: "TEMPLATE",
};
const AnnotationSeverityLevel = {
    CRITICAL: "CRITICAL",
    HIGH: "HIGH",
    INFORMATIONAL: "INFORMATIONAL",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
};
const AnnotationStatus = {
    FAILED: "FAILED",
    PASSED: "PASSED",
    SKIPPED: "SKIPPED",
};
const AttributeChangeType = {
    Add: "Add",
    Modify: "Modify",
    Remove: "Remove",
    SyncWithActual: "SyncWithActual",
};
const BeaconStackOperationStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    SUCCEEDED: "SUCCEEDED",
};
const BeforeValueFrom = {
    ACTUAL_STATE: "ACTUAL_STATE",
    PREVIOUS_DEPLOYMENT_STATE: "PREVIOUS_DEPLOYMENT_STATE",
};
const CallAs = {
    DELEGATED_ADMIN: "DELEGATED_ADMIN",
    SELF: "SELF",
};
const Capability = {
    CAPABILITY_AUTO_EXPAND: "CAPABILITY_AUTO_EXPAND",
    CAPABILITY_IAM: "CAPABILITY_IAM",
    CAPABILITY_NAMED_IAM: "CAPABILITY_NAMED_IAM",
};
const Category = {
    ACTIVATED: "ACTIVATED",
    AWS_TYPES: "AWS_TYPES",
    REGISTERED: "REGISTERED",
    THIRD_PARTY: "THIRD_PARTY",
};
const ChangeAction = {
    Add: "Add",
    Dynamic: "Dynamic",
    Import: "Import",
    Modify: "Modify",
    Remove: "Remove",
    SyncWithActual: "SyncWithActual",
};
const ChangeSource = {
    Automatic: "Automatic",
    DirectModification: "DirectModification",
    NoModification: "NoModification",
    ParameterReference: "ParameterReference",
    ResourceAttribute: "ResourceAttribute",
    ResourceReference: "ResourceReference",
};
const EvaluationType = {
    Dynamic: "Dynamic",
    Static: "Static",
};
const ResourceAttribute = {
    CreationPolicy: "CreationPolicy",
    DeletionPolicy: "DeletionPolicy",
    Metadata: "Metadata",
    Properties: "Properties",
    Tags: "Tags",
    UpdatePolicy: "UpdatePolicy",
    UpdateReplacePolicy: "UpdateReplacePolicy",
};
const RequiresRecreation = {
    Always: "Always",
    Conditionally: "Conditionally",
    Never: "Never",
};
const PolicyAction = {
    Delete: "Delete",
    ReplaceAndDelete: "ReplaceAndDelete",
    ReplaceAndRetain: "ReplaceAndRetain",
    ReplaceAndSnapshot: "ReplaceAndSnapshot",
    Retain: "Retain",
    Snapshot: "Snapshot",
};
const Replacement = {
    Conditional: "Conditional",
    False: "False",
    True: "True",
};
const DriftIgnoredReason = {
    MANAGED_BY_AWS: "MANAGED_BY_AWS",
    WRITE_ONLY_PROPERTY: "WRITE_ONLY_PROPERTY",
};
const StackResourceDriftStatus = {
    DELETED: "DELETED",
    IN_SYNC: "IN_SYNC",
    MODIFIED: "MODIFIED",
    NOT_CHECKED: "NOT_CHECKED",
    UNKNOWN: "UNKNOWN",
    UNSUPPORTED: "UNSUPPORTED",
};
const ChangeType = {
    Resource: "Resource",
};
const HookFailureMode = {
    FAIL: "FAIL",
    WARN: "WARN",
};
const HookInvocationPoint = {
    PRE_PROVISION: "PRE_PROVISION",
};
const HookTargetType = {
    RESOURCE: "RESOURCE",
};
const ChangeSetHooksStatus = {
    PLANNED: "PLANNED",
    PLANNING: "PLANNING",
    UNAVAILABLE: "UNAVAILABLE",
};
const ChangeSetStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    CREATE_PENDING: "CREATE_PENDING",
    DELETE_COMPLETE: "DELETE_COMPLETE",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    DELETE_PENDING: "DELETE_PENDING",
    FAILED: "FAILED",
};
const ExecutionStatus = {
    AVAILABLE: "AVAILABLE",
    EXECUTE_COMPLETE: "EXECUTE_COMPLETE",
    EXECUTE_FAILED: "EXECUTE_FAILED",
    EXECUTE_IN_PROGRESS: "EXECUTE_IN_PROGRESS",
    OBSOLETE: "OBSOLETE",
    UNAVAILABLE: "UNAVAILABLE",
};
const ChangeSetType = {
    CREATE: "CREATE",
    IMPORT: "IMPORT",
    UPDATE: "UPDATE",
};
const DeploymentMode = {
    REVERT_DRIFT: "REVERT_DRIFT",
};
const OnStackFailure = {
    DELETE: "DELETE",
    DO_NOTHING: "DO_NOTHING",
    ROLLBACK: "ROLLBACK",
};
const GeneratedTemplateDeletionPolicy = {
    DELETE: "DELETE",
    RETAIN: "RETAIN",
};
const GeneratedTemplateUpdateReplacePolicy = {
    DELETE: "DELETE",
    RETAIN: "RETAIN",
};
const OnFailure = {
    DELETE: "DELETE",
    DO_NOTHING: "DO_NOTHING",
    ROLLBACK: "ROLLBACK",
};
const ConcurrencyMode = {
    SOFT_FAILURE_TOLERANCE: "SOFT_FAILURE_TOLERANCE",
    STRICT_FAILURE_TOLERANCE: "STRICT_FAILURE_TOLERANCE",
};
const RegionConcurrencyType = {
    PARALLEL: "PARALLEL",
    SEQUENTIAL: "SEQUENTIAL",
};
const PermissionModels = {
    SELF_MANAGED: "SELF_MANAGED",
    SERVICE_MANAGED: "SERVICE_MANAGED",
};
const DeletionMode = {
    FORCE_DELETE_STACK: "FORCE_DELETE_STACK",
    STANDARD: "STANDARD",
};
const RegistryType = {
    HOOK: "HOOK",
    MODULE: "MODULE",
    RESOURCE: "RESOURCE",
};
const StackDriftStatus = {
    DRIFTED: "DRIFTED",
    IN_SYNC: "IN_SYNC",
    NOT_CHECKED: "NOT_CHECKED",
    UNKNOWN: "UNKNOWN",
};
const DetailedStatus = {
    CONFIGURATION_COMPLETE: "CONFIGURATION_COMPLETE",
    VALIDATION_FAILED: "VALIDATION_FAILED",
};
const EventType = {
    HOOK_INVOCATION_ERROR: "HOOK_INVOCATION_ERROR",
    PROGRESS_EVENT: "PROGRESS_EVENT",
    PROVISIONING_ERROR: "PROVISIONING_ERROR",
    STACK_EVENT: "STACK_EVENT",
    VALIDATION_ERROR: "VALIDATION_ERROR",
};
const HookStatus = {
    HOOK_COMPLETE_FAILED: "HOOK_COMPLETE_FAILED",
    HOOK_COMPLETE_SUCCEEDED: "HOOK_COMPLETE_SUCCEEDED",
    HOOK_FAILED: "HOOK_FAILED",
    HOOK_IN_PROGRESS: "HOOK_IN_PROGRESS",
};
const OperationType = {
    CONTINUE_ROLLBACK: "CONTINUE_ROLLBACK",
    CREATE_CHANGESET: "CREATE_CHANGESET",
    CREATE_STACK: "CREATE_STACK",
    DELETE_STACK: "DELETE_STACK",
    ROLLBACK: "ROLLBACK",
    UPDATE_STACK: "UPDATE_STACK",
};
const ResourceStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_COMPLETE: "DELETE_COMPLETE",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    DELETE_SKIPPED: "DELETE_SKIPPED",
    EXPORT_COMPLETE: "EXPORT_COMPLETE",
    EXPORT_FAILED: "EXPORT_FAILED",
    EXPORT_IN_PROGRESS: "EXPORT_IN_PROGRESS",
    EXPORT_ROLLBACK_COMPLETE: "EXPORT_ROLLBACK_COMPLETE",
    EXPORT_ROLLBACK_FAILED: "EXPORT_ROLLBACK_FAILED",
    EXPORT_ROLLBACK_IN_PROGRESS: "EXPORT_ROLLBACK_IN_PROGRESS",
    IMPORT_COMPLETE: "IMPORT_COMPLETE",
    IMPORT_FAILED: "IMPORT_FAILED",
    IMPORT_IN_PROGRESS: "IMPORT_IN_PROGRESS",
    IMPORT_ROLLBACK_COMPLETE: "IMPORT_ROLLBACK_COMPLETE",
    IMPORT_ROLLBACK_FAILED: "IMPORT_ROLLBACK_FAILED",
    IMPORT_ROLLBACK_IN_PROGRESS: "IMPORT_ROLLBACK_IN_PROGRESS",
    ROLLBACK_COMPLETE: "ROLLBACK_COMPLETE",
    ROLLBACK_FAILED: "ROLLBACK_FAILED",
    ROLLBACK_IN_PROGRESS: "ROLLBACK_IN_PROGRESS",
    UPDATE_COMPLETE: "UPDATE_COMPLETE",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATE_IN_PROGRESS: "UPDATE_IN_PROGRESS",
    UPDATE_ROLLBACK_COMPLETE: "UPDATE_ROLLBACK_COMPLETE",
    UPDATE_ROLLBACK_FAILED: "UPDATE_ROLLBACK_FAILED",
    UPDATE_ROLLBACK_IN_PROGRESS: "UPDATE_ROLLBACK_IN_PROGRESS",
};
const ValidationStatus = {
    FAILED: "FAILED",
    SKIPPED: "SKIPPED",
};
const GeneratedTemplateResourceStatus = {
    COMPLETE: "COMPLETE",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PENDING: "PENDING",
};
const WarningType = {
    EXCLUDED_PROPERTIES: "EXCLUDED_PROPERTIES",
    EXCLUDED_RESOURCES: "EXCLUDED_RESOURCES",
    MUTUALLY_EXCLUSIVE_PROPERTIES: "MUTUALLY_EXCLUSIVE_PROPERTIES",
    MUTUALLY_EXCLUSIVE_TYPES: "MUTUALLY_EXCLUSIVE_TYPES",
    UNSUPPORTED_PROPERTIES: "UNSUPPORTED_PROPERTIES",
};
const GeneratedTemplateStatus = {
    COMPLETE: "COMPLETE",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    CREATE_PENDING: "CREATE_PENDING",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    DELETE_PENDING: "DELETE_PENDING",
    FAILED: "FAILED",
    UPDATE_IN_PROGRESS: "UPDATE_IN_PROGRESS",
    UPDATE_PENDING: "UPDATE_PENDING",
};
const OrganizationStatus = {
    DISABLED: "DISABLED",
    DISABLED_PERMANENTLY: "DISABLED_PERMANENTLY",
    ENABLED: "ENABLED",
};
const IdentityProvider = {
    AWS_Marketplace: "AWS_Marketplace",
    Bitbucket: "Bitbucket",
    GitHub: "GitHub",
};
const PublisherStatus = {
    UNVERIFIED: "UNVERIFIED",
    VERIFIED: "VERIFIED",
};
const ResourceScanStatus = {
    COMPLETE: "COMPLETE",
    EXPIRED: "EXPIRED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
const StackDriftDetectionStatus = {
    DETECTION_COMPLETE: "DETECTION_COMPLETE",
    DETECTION_FAILED: "DETECTION_FAILED",
    DETECTION_IN_PROGRESS: "DETECTION_IN_PROGRESS",
};
const StackInstanceDetailedStatus = {
    CANCELLED: "CANCELLED",
    FAILED: "FAILED",
    FAILED_IMPORT: "FAILED_IMPORT",
    INOPERABLE: "INOPERABLE",
    PENDING: "PENDING",
    RUNNING: "RUNNING",
    SKIPPED_SUSPENDED_ACCOUNT: "SKIPPED_SUSPENDED_ACCOUNT",
    SUCCEEDED: "SUCCEEDED",
};
const StackInstanceStatus = {
    CURRENT: "CURRENT",
    INOPERABLE: "INOPERABLE",
    OUTDATED: "OUTDATED",
};
const StackRefactorExecutionStatus = {
    AVAILABLE: "AVAILABLE",
    EXECUTE_COMPLETE: "EXECUTE_COMPLETE",
    EXECUTE_FAILED: "EXECUTE_FAILED",
    EXECUTE_IN_PROGRESS: "EXECUTE_IN_PROGRESS",
    OBSOLETE: "OBSOLETE",
    ROLLBACK_COMPLETE: "ROLLBACK_COMPLETE",
    ROLLBACK_FAILED: "ROLLBACK_FAILED",
    ROLLBACK_IN_PROGRESS: "ROLLBACK_IN_PROGRESS",
    UNAVAILABLE: "UNAVAILABLE",
};
const StackRefactorStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_COMPLETE: "DELETE_COMPLETE",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
};
const DifferenceType = {
    ADD: "ADD",
    NOT_EQUAL: "NOT_EQUAL",
    REMOVE: "REMOVE",
};
const StackStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_COMPLETE: "DELETE_COMPLETE",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    IMPORT_COMPLETE: "IMPORT_COMPLETE",
    IMPORT_IN_PROGRESS: "IMPORT_IN_PROGRESS",
    IMPORT_ROLLBACK_COMPLETE: "IMPORT_ROLLBACK_COMPLETE",
    IMPORT_ROLLBACK_FAILED: "IMPORT_ROLLBACK_FAILED",
    IMPORT_ROLLBACK_IN_PROGRESS: "IMPORT_ROLLBACK_IN_PROGRESS",
    REVIEW_IN_PROGRESS: "REVIEW_IN_PROGRESS",
    ROLLBACK_COMPLETE: "ROLLBACK_COMPLETE",
    ROLLBACK_FAILED: "ROLLBACK_FAILED",
    ROLLBACK_IN_PROGRESS: "ROLLBACK_IN_PROGRESS",
    UPDATE_COMPLETE: "UPDATE_COMPLETE",
    UPDATE_COMPLETE_CLEANUP_IN_PROGRESS: "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATE_IN_PROGRESS: "UPDATE_IN_PROGRESS",
    UPDATE_ROLLBACK_COMPLETE: "UPDATE_ROLLBACK_COMPLETE",
    UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS: "UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS",
    UPDATE_ROLLBACK_FAILED: "UPDATE_ROLLBACK_FAILED",
    UPDATE_ROLLBACK_IN_PROGRESS: "UPDATE_ROLLBACK_IN_PROGRESS",
};
const StackSetDriftDetectionStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PARTIAL_SUCCESS: "PARTIAL_SUCCESS",
    STOPPED: "STOPPED",
};
const StackSetDriftStatus = {
    DRIFTED: "DRIFTED",
    IN_SYNC: "IN_SYNC",
    NOT_CHECKED: "NOT_CHECKED",
};
const StackSetStatus = {
    ACTIVE: "ACTIVE",
    DELETED: "DELETED",
};
const StackSetOperationAction = {
    CREATE: "CREATE",
    DELETE: "DELETE",
    DETECT_DRIFT: "DETECT_DRIFT",
    UPDATE: "UPDATE",
};
const StackSetOperationStatus = {
    FAILED: "FAILED",
    QUEUED: "QUEUED",
    RUNNING: "RUNNING",
    STOPPED: "STOPPED",
    STOPPING: "STOPPING",
    SUCCEEDED: "SUCCEEDED",
};
const DeprecatedStatus = {
    DEPRECATED: "DEPRECATED",
    LIVE: "LIVE",
};
const ProvisioningType = {
    FULLY_MUTABLE: "FULLY_MUTABLE",
    IMMUTABLE: "IMMUTABLE",
    NON_PROVISIONABLE: "NON_PROVISIONABLE",
};
const TypeTestsStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    NOT_TESTED: "NOT_TESTED",
    PASSED: "PASSED",
};
const Visibility = {
    PRIVATE: "PRIVATE",
    PUBLIC: "PUBLIC",
};
const RegistrationStatus = {
    COMPLETE: "COMPLETE",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
const TemplateFormat = {
    JSON: "JSON",
    YAML: "YAML",
};
const HookTargetAction = {
    CREATE: "CREATE",
    DELETE: "DELETE",
    IMPORT: "IMPORT",
    UPDATE: "UPDATE",
};
const TemplateStage = {
    Original: "Original",
    Processed: "Processed",
};
const ListHookResultsTargetType = {
    CHANGE_SET: "CHANGE_SET",
    CLOUD_CONTROL: "CLOUD_CONTROL",
    RESOURCE: "RESOURCE",
    STACK: "STACK",
};
const ScanType = {
    FULL: "FULL",
    PARTIAL: "PARTIAL",
};
const StackInstanceFilterName = {
    DETAILED_STATUS: "DETAILED_STATUS",
    DRIFT_STATUS: "DRIFT_STATUS",
    LAST_OPERATION_ID: "LAST_OPERATION_ID",
};
const StackRefactorActionType = {
    CREATE: "CREATE",
    MOVE: "MOVE",
};
const StackRefactorDetection = {
    AUTO: "AUTO",
    MANUAL: "MANUAL",
};
const StackRefactorActionEntity = {
    RESOURCE: "RESOURCE",
    STACK: "STACK",
};
const OperationResultFilterName = {
    OPERATION_RESULT_STATUS: "OPERATION_RESULT_STATUS",
};
const StackSetOperationResultStatus = {
    CANCELLED: "CANCELLED",
    FAILED: "FAILED",
    PENDING: "PENDING",
    RUNNING: "RUNNING",
    SUCCEEDED: "SUCCEEDED",
};
const OperationStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PENDING: "PENDING",
    SUCCESS: "SUCCESS",
};
const HandlerErrorCode = {
    AccessDenied: "AccessDenied",
    AlreadyExists: "AlreadyExists",
    GeneralServiceException: "GeneralServiceException",
    HandlerInternalFailure: "HandlerInternalFailure",
    InternalFailure: "InternalFailure",
    InvalidCredentials: "InvalidCredentials",
    InvalidRequest: "InvalidRequest",
    InvalidTypeConfiguration: "InvalidTypeConfiguration",
    NetworkFailure: "NetworkFailure",
    NonCompliant: "NonCompliant",
    NotFound: "NotFound",
    NotUpdatable: "NotUpdatable",
    ResourceConflict: "ResourceConflict",
    ServiceInternalError: "ServiceInternalError",
    ServiceLimitExceeded: "ServiceLimitExceeded",
    ServiceTimeout: "NotStabilized",
    Throttling: "Throttling",
    Unknown: "Unknown",
    UnsupportedTarget: "UnsupportedTarget",
};
const ResourceSignalStatus = {
    FAILURE: "FAILURE",
    SUCCESS: "SUCCESS",
};

exports.AccountFilterType = AccountFilterType;
exports.AccountGateStatus = AccountGateStatus;
exports.ActivateOrganizationsAccessCommand = ActivateOrganizationsAccessCommand;
exports.ActivateTypeCommand = ActivateTypeCommand;
exports.AfterValueFrom = AfterValueFrom;
exports.AnnotationSeverityLevel = AnnotationSeverityLevel;
exports.AnnotationStatus = AnnotationStatus;
exports.AttributeChangeType = AttributeChangeType;
exports.BatchDescribeTypeConfigurationsCommand = BatchDescribeTypeConfigurationsCommand;
exports.BeaconStackOperationStatus = BeaconStackOperationStatus;
exports.BeforeValueFrom = BeforeValueFrom;
exports.CallAs = CallAs;
exports.CancelUpdateStackCommand = CancelUpdateStackCommand;
exports.Capability = Capability;
exports.Category = Category;
exports.ChangeAction = ChangeAction;
exports.ChangeSetHooksStatus = ChangeSetHooksStatus;
exports.ChangeSetStatus = ChangeSetStatus;
exports.ChangeSetType = ChangeSetType;
exports.ChangeSource = ChangeSource;
exports.ChangeType = ChangeType;
exports.CloudFormation = CloudFormation;
exports.CloudFormationClient = CloudFormationClient;
exports.ConcurrencyMode = ConcurrencyMode;
exports.ContinueUpdateRollbackCommand = ContinueUpdateRollbackCommand;
exports.CreateChangeSetCommand = CreateChangeSetCommand;
exports.CreateGeneratedTemplateCommand = CreateGeneratedTemplateCommand;
exports.CreateStackCommand = CreateStackCommand;
exports.CreateStackInstancesCommand = CreateStackInstancesCommand;
exports.CreateStackRefactorCommand = CreateStackRefactorCommand;
exports.CreateStackSetCommand = CreateStackSetCommand;
exports.DeactivateOrganizationsAccessCommand = DeactivateOrganizationsAccessCommand;
exports.DeactivateTypeCommand = DeactivateTypeCommand;
exports.DeleteChangeSetCommand = DeleteChangeSetCommand;
exports.DeleteGeneratedTemplateCommand = DeleteGeneratedTemplateCommand;
exports.DeleteStackCommand = DeleteStackCommand;
exports.DeleteStackInstancesCommand = DeleteStackInstancesCommand;
exports.DeleteStackSetCommand = DeleteStackSetCommand;
exports.DeletionMode = DeletionMode;
exports.DeploymentMode = DeploymentMode;
exports.DeprecatedStatus = DeprecatedStatus;
exports.DeregisterTypeCommand = DeregisterTypeCommand;
exports.DescribeAccountLimitsCommand = DescribeAccountLimitsCommand;
exports.DescribeChangeSetCommand = DescribeChangeSetCommand;
exports.DescribeChangeSetHooksCommand = DescribeChangeSetHooksCommand;
exports.DescribeEventsCommand = DescribeEventsCommand;
exports.DescribeGeneratedTemplateCommand = DescribeGeneratedTemplateCommand;
exports.DescribeOrganizationsAccessCommand = DescribeOrganizationsAccessCommand;
exports.DescribePublisherCommand = DescribePublisherCommand;
exports.DescribeResourceScanCommand = DescribeResourceScanCommand;
exports.DescribeStackDriftDetectionStatusCommand = DescribeStackDriftDetectionStatusCommand;
exports.DescribeStackEventsCommand = DescribeStackEventsCommand;
exports.DescribeStackInstanceCommand = DescribeStackInstanceCommand;
exports.DescribeStackRefactorCommand = DescribeStackRefactorCommand;
exports.DescribeStackResourceCommand = DescribeStackResourceCommand;
exports.DescribeStackResourceDriftsCommand = DescribeStackResourceDriftsCommand;
exports.DescribeStackResourcesCommand = DescribeStackResourcesCommand;
exports.DescribeStackSetCommand = DescribeStackSetCommand;
exports.DescribeStackSetOperationCommand = DescribeStackSetOperationCommand;
exports.DescribeStacksCommand = DescribeStacksCommand;
exports.DescribeTypeCommand = DescribeTypeCommand;
exports.DescribeTypeRegistrationCommand = DescribeTypeRegistrationCommand;
exports.DetailedStatus = DetailedStatus;
exports.DetectStackDriftCommand = DetectStackDriftCommand;
exports.DetectStackResourceDriftCommand = DetectStackResourceDriftCommand;
exports.DetectStackSetDriftCommand = DetectStackSetDriftCommand;
exports.DifferenceType = DifferenceType;
exports.DriftIgnoredReason = DriftIgnoredReason;
exports.EstimateTemplateCostCommand = EstimateTemplateCostCommand;
exports.EvaluationType = EvaluationType;
exports.EventType = EventType;
exports.ExecuteChangeSetCommand = ExecuteChangeSetCommand;
exports.ExecuteStackRefactorCommand = ExecuteStackRefactorCommand;
exports.ExecutionStatus = ExecutionStatus;
exports.GeneratedTemplateDeletionPolicy = GeneratedTemplateDeletionPolicy;
exports.GeneratedTemplateResourceStatus = GeneratedTemplateResourceStatus;
exports.GeneratedTemplateStatus = GeneratedTemplateStatus;
exports.GeneratedTemplateUpdateReplacePolicy = GeneratedTemplateUpdateReplacePolicy;
exports.GetGeneratedTemplateCommand = GetGeneratedTemplateCommand;
exports.GetHookResultCommand = GetHookResultCommand;
exports.GetStackPolicyCommand = GetStackPolicyCommand;
exports.GetTemplateCommand = GetTemplateCommand;
exports.GetTemplateSummaryCommand = GetTemplateSummaryCommand;
exports.HandlerErrorCode = HandlerErrorCode;
exports.HookFailureMode = HookFailureMode;
exports.HookInvocationPoint = HookInvocationPoint;
exports.HookStatus = HookStatus;
exports.HookTargetAction = HookTargetAction;
exports.HookTargetType = HookTargetType;
exports.IdentityProvider = IdentityProvider;
exports.ImportStacksToStackSetCommand = ImportStacksToStackSetCommand;
exports.ListChangeSetsCommand = ListChangeSetsCommand;
exports.ListExportsCommand = ListExportsCommand;
exports.ListGeneratedTemplatesCommand = ListGeneratedTemplatesCommand;
exports.ListHookResultsCommand = ListHookResultsCommand;
exports.ListHookResultsTargetType = ListHookResultsTargetType;
exports.ListImportsCommand = ListImportsCommand;
exports.ListResourceScanRelatedResourcesCommand = ListResourceScanRelatedResourcesCommand;
exports.ListResourceScanResourcesCommand = ListResourceScanResourcesCommand;
exports.ListResourceScansCommand = ListResourceScansCommand;
exports.ListStackInstanceResourceDriftsCommand = ListStackInstanceResourceDriftsCommand;
exports.ListStackInstancesCommand = ListStackInstancesCommand;
exports.ListStackRefactorActionsCommand = ListStackRefactorActionsCommand;
exports.ListStackRefactorsCommand = ListStackRefactorsCommand;
exports.ListStackResourcesCommand = ListStackResourcesCommand;
exports.ListStackSetAutoDeploymentTargetsCommand = ListStackSetAutoDeploymentTargetsCommand;
exports.ListStackSetOperationResultsCommand = ListStackSetOperationResultsCommand;
exports.ListStackSetOperationsCommand = ListStackSetOperationsCommand;
exports.ListStackSetsCommand = ListStackSetsCommand;
exports.ListStacksCommand = ListStacksCommand;
exports.ListTypeRegistrationsCommand = ListTypeRegistrationsCommand;
exports.ListTypeVersionsCommand = ListTypeVersionsCommand;
exports.ListTypesCommand = ListTypesCommand;
exports.OnFailure = OnFailure;
exports.OnStackFailure = OnStackFailure;
exports.OperationResultFilterName = OperationResultFilterName;
exports.OperationStatus = OperationStatus;
exports.OperationType = OperationType;
exports.OrganizationStatus = OrganizationStatus;
exports.PermissionModels = PermissionModels;
exports.PolicyAction = PolicyAction;
exports.ProvisioningType = ProvisioningType;
exports.PublishTypeCommand = PublishTypeCommand;
exports.PublisherStatus = PublisherStatus;
exports.RecordHandlerProgressCommand = RecordHandlerProgressCommand;
exports.RegionConcurrencyType = RegionConcurrencyType;
exports.RegisterPublisherCommand = RegisterPublisherCommand;
exports.RegisterTypeCommand = RegisterTypeCommand;
exports.RegistrationStatus = RegistrationStatus;
exports.RegistryType = RegistryType;
exports.Replacement = Replacement;
exports.RequiresRecreation = RequiresRecreation;
exports.ResourceAttribute = ResourceAttribute;
exports.ResourceScanStatus = ResourceScanStatus;
exports.ResourceSignalStatus = ResourceSignalStatus;
exports.ResourceStatus = ResourceStatus;
exports.RollbackStackCommand = RollbackStackCommand;
exports.ScanType = ScanType;
exports.SetStackPolicyCommand = SetStackPolicyCommand;
exports.SetTypeConfigurationCommand = SetTypeConfigurationCommand;
exports.SetTypeDefaultVersionCommand = SetTypeDefaultVersionCommand;
exports.SignalResourceCommand = SignalResourceCommand;
exports.StackDriftDetectionStatus = StackDriftDetectionStatus;
exports.StackDriftStatus = StackDriftStatus;
exports.StackInstanceDetailedStatus = StackInstanceDetailedStatus;
exports.StackInstanceFilterName = StackInstanceFilterName;
exports.StackInstanceStatus = StackInstanceStatus;
exports.StackRefactorActionEntity = StackRefactorActionEntity;
exports.StackRefactorActionType = StackRefactorActionType;
exports.StackRefactorDetection = StackRefactorDetection;
exports.StackRefactorExecutionStatus = StackRefactorExecutionStatus;
exports.StackRefactorStatus = StackRefactorStatus;
exports.StackResourceDriftStatus = StackResourceDriftStatus;
exports.StackSetDriftDetectionStatus = StackSetDriftDetectionStatus;
exports.StackSetDriftStatus = StackSetDriftStatus;
exports.StackSetOperationAction = StackSetOperationAction;
exports.StackSetOperationResultStatus = StackSetOperationResultStatus;
exports.StackSetOperationStatus = StackSetOperationStatus;
exports.StackSetStatus = StackSetStatus;
exports.StackStatus = StackStatus;
exports.StartResourceScanCommand = StartResourceScanCommand;
exports.StopStackSetOperationCommand = StopStackSetOperationCommand;
exports.TemplateFormat = TemplateFormat;
exports.TemplateStage = TemplateStage;
exports.TestTypeCommand = TestTypeCommand;
exports.ThirdPartyType = ThirdPartyType;
exports.TypeTestsStatus = TypeTestsStatus;
exports.UpdateGeneratedTemplateCommand = UpdateGeneratedTemplateCommand;
exports.UpdateStackCommand = UpdateStackCommand;
exports.UpdateStackInstancesCommand = UpdateStackInstancesCommand;
exports.UpdateStackSetCommand = UpdateStackSetCommand;
exports.UpdateTerminationProtectionCommand = UpdateTerminationProtectionCommand;
exports.ValidateTemplateCommand = ValidateTemplateCommand;
exports.ValidationStatus = ValidationStatus;
exports.VersionBump = VersionBump;
exports.Visibility = Visibility;
exports.WarningType = WarningType;
exports.paginateDescribeAccountLimits = paginateDescribeAccountLimits;
exports.paginateDescribeChangeSet = paginateDescribeChangeSet;
exports.paginateDescribeEvents = paginateDescribeEvents;
exports.paginateDescribeStackEvents = paginateDescribeStackEvents;
exports.paginateDescribeStackResourceDrifts = paginateDescribeStackResourceDrifts;
exports.paginateDescribeStacks = paginateDescribeStacks;
exports.paginateListChangeSets = paginateListChangeSets;
exports.paginateListExports = paginateListExports;
exports.paginateListGeneratedTemplates = paginateListGeneratedTemplates;
exports.paginateListImports = paginateListImports;
exports.paginateListResourceScanRelatedResources = paginateListResourceScanRelatedResources;
exports.paginateListResourceScanResources = paginateListResourceScanResources;
exports.paginateListResourceScans = paginateListResourceScans;
exports.paginateListStackInstances = paginateListStackInstances;
exports.paginateListStackRefactorActions = paginateListStackRefactorActions;
exports.paginateListStackRefactors = paginateListStackRefactors;
exports.paginateListStackResources = paginateListStackResources;
exports.paginateListStackSetOperationResults = paginateListStackSetOperationResults;
exports.paginateListStackSetOperations = paginateListStackSetOperations;
exports.paginateListStackSets = paginateListStackSets;
exports.paginateListStacks = paginateListStacks;
exports.paginateListTypeRegistrations = paginateListTypeRegistrations;
exports.paginateListTypeVersions = paginateListTypeVersions;
exports.paginateListTypes = paginateListTypes;
exports.waitForChangeSetCreateComplete = waitForChangeSetCreateComplete;
exports.waitForStackCreateComplete = waitForStackCreateComplete;
exports.waitForStackDeleteComplete = waitForStackDeleteComplete;
exports.waitForStackExists = waitForStackExists;
exports.waitForStackImportComplete = waitForStackImportComplete;
exports.waitForStackRefactorCreateComplete = waitForStackRefactorCreateComplete;
exports.waitForStackRefactorExecuteComplete = waitForStackRefactorExecuteComplete;
exports.waitForStackRollbackComplete = waitForStackRollbackComplete;
exports.waitForStackUpdateComplete = waitForStackUpdateComplete;
exports.waitForTypeRegistrationComplete = waitForTypeRegistrationComplete;
exports.waitUntilChangeSetCreateComplete = waitUntilChangeSetCreateComplete;
exports.waitUntilStackCreateComplete = waitUntilStackCreateComplete;
exports.waitUntilStackDeleteComplete = waitUntilStackDeleteComplete;
exports.waitUntilStackExists = waitUntilStackExists;
exports.waitUntilStackImportComplete = waitUntilStackImportComplete;
exports.waitUntilStackRefactorCreateComplete = waitUntilStackRefactorCreateComplete;
exports.waitUntilStackRefactorExecuteComplete = waitUntilStackRefactorExecuteComplete;
exports.waitUntilStackRollbackComplete = waitUntilStackRollbackComplete;
exports.waitUntilStackUpdateComplete = waitUntilStackUpdateComplete;
exports.waitUntilTypeRegistrationComplete = waitUntilTypeRegistrationComplete;
