// @aws-sdk/client-iam/dist-cjs/schemas/schemas_0.js
const _A = "Arn";
const _AA = "AccountAlias";
const _AAc = "AccountAliases";
const _ABO = "AllowedByOrganizations";
const _ABPB = "AllowedByPermissionsBoundary";
const _AC = "AuthenticationCode1";
const _ACIDTOIDCP = "AddClientIDToOpenIDConnectProvider";
const _ACIDTOIDCPR = "AddClientIDToOpenIDConnectProviderRequest";
const _ACt = "AttachmentCount";
const _ACu = "AuthenticationCode2";
const _AD = "AccessDetail";
const _ADR = "AcceptDelegationRequest";
const _ADRR = "AcceptDelegationRequestRequest";
const _ADRRs = "AssociateDelegationRequestRequest";
const _ADRs = "AssociateDelegationRequest";
const _ADc = "AccessDetails";
const _AEM = "AssertionEncryptionMode";
const _AGP = "AttachGroupPolicy";
const _AGPR = "AttachGroupPolicyRequest";
const _AI = "ApproverId";
const _AK = "AccessKey";
const _AKI = "AccessKeyId";
const _AKLU = "AccessKeyLastUsed";
const _AKM = "AccessKeyMetadata";
const _AMP = "AttachedManagedPolicies";
const _AN = "ActionNames";
const _ANMODAE = "AccountNotManagementOrDelegatedAdministratorException";
const _ANc = "ActionName";
const _AP = "AttachedPolicy";
const _APB = "AttachedPermissionsBoundary";
const _APK = "AddPrivateKey";
const _APt = "AttachedPolicies";
const _ARP = "AttachRolePolicy";
const _ARPD = "AssumeRolePolicyDocument";
const _ARPR = "AttachRolePolicyRequest";
const _ARTIP = "AddRoleToInstanceProfile";
const _ARTIPR = "AddRoleToInstanceProfileRequest";
const _AS = "AssignmentStatus";
const _AU = "AllUsers";
const _AUP = "AttachUserPolicy";
const _AUPR = "AttachUserPolicyRequest";
const _AUTCP = "AllowUsersToChangePassword";
const _AUTG = "AddUserToGroup";
const _AUTGR = "AddUserToGroupRequest";
const _AWSSN = "AWSServiceName";
const _BD = "BootstrapDatum";
const _BSS = "Base32StringSeed";
const _C = "Code";
const _CA = "CallerArn";
const _CAA = "CreateAccountAlias";
const _CAAR = "CreateAccountAliasRequest";
const _CAD = "CredentialAgeDays";
const _CAK = "CreateAccessKey";
const _CAKR = "CreateAccessKeyRequest";
const _CAKRr = "CreateAccessKeyResponse";
const _CB = "CertificateBody";
const _CC = "CertificateChain";
const _CD = "CreateDate";
const _CDL = "ConsoleDeepLink";
const _CDR = "CreateDelegationRequest";
const _CDRR = "CreateDelegationRequestRequest";
const _CDRRr = "CreateDelegationRequestResponse";
const _CE = "ContextEntry";
const _CELT = "ContextEntryListType";
const _CEo = "ContextEntries";
const _CG = "CreateGroup";
const _CGR = "CreateGroupRequest";
const _CGRr = "CreateGroupResponse";
const _CI = "CertificateId";
const _CID = "ClientID";
const _CIDL = "ClientIDList";
const _CINMAE = "CallerIsNotManagementAccountException";
const _CIP = "CreateInstanceProfile";
const _CIPR = "CreateInstanceProfileRequest";
const _CIPRr = "CreateInstanceProfileResponse";
const _CKN = "ContextKeyName";
const _CKNo = "ContextKeyNames";
const _CKT = "ContextKeyType";
const _CKV = "ContextKeyValues";
const _CLP = "CreateLoginProfile";
const _CLPR = "CreateLoginProfileRequest";
const _CLPRr = "CreateLoginProfileResponse";
const _CME = "ConcurrentModificationException";
const _COIDCP = "CreateOpenIDConnectProvider";
const _COIDCPR = "CreateOpenIDConnectProviderRequest";
const _COIDCPRr = "CreateOpenIDConnectProviderResponse";
const _CP = "ChangePassword";
const _CPR = "ChangePasswordRequest";
const _CPRr = "CreatePolicyRequest";
const _CPRre = "CreatePolicyResponse";
const _CPV = "CreatePolicyVersion";
const _CPVR = "CreatePolicyVersionRequest";
const _CPVRr = "CreatePolicyVersionResponse";
const _CPr = "CreatePolicy";
const _CR = "CreateRole";
const _CREE = "CredentialReportExpiredException";
const _CRNPE = "CredentialReportNotPresentException";
const _CRNRE = "CredentialReportNotReadyException";
const _CRR = "CreateRoleRequest";
const _CRRr = "CreateRoleResponse";
const _CS = "CustomSuffix";
const _CSAMLP = "CreateSAMLProvider";
const _CSAMLPR = "CreateSAMLProviderRequest";
const _CSAMLPRr = "CreateSAMLProviderResponse";
const _CSLR = "CreateServiceLinkedRole";
const _CSLRR = "CreateServiceLinkedRoleRequest";
const _CSLRRr = "CreateServiceLinkedRoleResponse";
const _CSSC = "CreateServiceSpecificCredential";
const _CSSCR = "CreateServiceSpecificCredentialRequest";
const _CSSCRr = "CreateServiceSpecificCredentialResponse";
const _CU = "CreateUser";
const _CUR = "CreateUserRequest";
const _CURr = "CreateUserResponse";
const _CVMFAD = "CreateVirtualMFADevice";
const _CVMFADR = "CreateVirtualMFADeviceRequest";
const _CVMFADRr = "CreateVirtualMFADeviceResponse";
const _Ce = "Certifications";
const _Cer = "Certificates";
const _Cert = "Certificate";
const _Co = "Content";
const _Col = "Column";
const _D = "Description";
const _DAA = "DeleteAccountAlias";
const _DAAR = "DeleteAccountAliasRequest";
const _DAK = "DeleteAccessKey";
const _DAKR = "DeleteAccessKeyRequest";
const _DAPP = "DeleteAccountPasswordPolicy";
const _DCE = "DeleteConflictException";
const _DCEu = "DuplicateCertificateException";
const _DG = "DeleteGroup";
const _DGP = "DeleteGroupPolicy";
const _DGPR = "DeleteGroupPolicyRequest";
const _DGPRe = "DetachGroupPolicyRequest";
const _DGPe = "DetachGroupPolicy";
const _DGR = "DeleteGroupRequest";
const _DIP = "DeleteInstanceProfile";
const _DIPR = "DeleteInstanceProfileRequest";
const _DLP = "DeleteLoginProfile";
const _DLPR = "DeleteLoginProfileRequest";
const _DMFAD = "DeactivateMFADevice";
const _DMFADR = "DeactivateMFADeviceRequest";
const _DOIDCP = "DeleteOpenIDConnectProvider";
const _DOIDCPR = "DeleteOpenIDConnectProviderRequest";
const _DORCM = "DisableOrganizationsRootCredentialsManagement";
const _DORCMR = "DisableOrganizationsRootCredentialsManagementRequest";
const _DORCMRi = "DisableOrganizationsRootCredentialsManagementResponse";
const _DORS = "DisableOrganizationsRootSessions";
const _DORSR = "DisableOrganizationsRootSessionsRequest";
const _DORSRi = "DisableOrganizationsRootSessionsResponse";
const _DOWIF = "DisableOutboundWebIdentityFederation";
const _DP = "DelegationPermission";
const _DPC = "DelegationPermissionCheck";
const _DPR = "DeletePolicyRequest";
const _DPV = "DeletePolicyVersion";
const _DPVR = "DeletePolicyVersionRequest";
const _DPe = "DeletePolicy";
const _DR = "DelegationRequest";
const _DRI = "DelegationRequestId";
const _DRP = "DeleteRolePolicy";
const _DRPB = "DeleteRolePermissionsBoundary";
const _DRPBR = "DeleteRolePermissionsBoundaryRequest";
const _DRPR = "DeleteRolePolicyRequest";
const _DRPRe = "DetachRolePolicyRequest";
const _DRPe = "DetachRolePolicy";
const _DRR = "DeleteRoleRequest";
const _DRe = "DelegationRequests";
const _DRel = "DeleteRole";
const _DSAMLP = "DeleteSAMLProvider";
const _DSAMLPR = "DeleteSAMLProviderRequest";
const _DSC = "DeleteServerCertificate";
const _DSCR = "DeleteServerCertificateRequest";
const _DSCRe = "DeleteSigningCertificateRequest";
const _DSCe = "DeleteSigningCertificate";
const _DSLR = "DeleteServiceLinkedRole";
const _DSLRR = "DeleteServiceLinkedRoleRequest";
const _DSLRRe = "DeleteServiceLinkedRoleResponse";
const _DSSC = "DeleteServiceSpecificCredential";
const _DSSCR = "DeleteServiceSpecificCredentialRequest";
const _DSSHPK = "DeleteSSHPublicKey";
const _DSSHPKE = "DuplicateSSHPublicKeyException";
const _DSSHPKR = "DeleteSSHPublicKeyRequest";
const _DTFRT = "DeletionTaskFailureReasonType";
const _DTI = "DeletionTaskId";
const _DU = "DeleteUser";
const _DUP = "DeleteUserPolicy";
const _DUPB = "DeleteUserPermissionsBoundary";
const _DUPBR = "DeleteUserPermissionsBoundaryRequest";
const _DUPR = "DeleteUserPolicyRequest";
const _DUPRe = "DetachUserPolicyRequest";
const _DUPe = "DetachUserPolicy";
const _DUR = "DeleteUserRequest";
const _DVI = "DefaultVersionId";
const _DVMFAD = "DeleteVirtualMFADevice";
const _DVMFADR = "DeleteVirtualMFADeviceRequest";
const _Do = "Document";
const _E = "Error";
const _EA = "EntityArn";
const _EAEE = "EntityAlreadyExistsException";
const _EAN = "EvalActionName";
const _ED = "EntityDetails";
const _EDD = "EvalDecisionDetails";
const _EDL = "EntityDetailsList";
const _EDn = "EnableDate";
const _EDr = "ErrorDetails";
const _EDv = "EvalDecision";
const _EDx = "ExpirationDate";
const _EF = "EnabledFeatures";
const _EFn = "EntityFilter";
const _EI = "EntityInfo";
const _EMFAD = "EnableMFADevice";
const _EMFADR = "EnableMFADeviceRequest";
const _EN = "EntityName";
const _EORCM = "EnableOrganizationsRootCredentialsManagement";
const _EORCMR = "EnableOrganizationsRootCredentialsManagementRequest";
const _EORCMRn = "EnableOrganizationsRootCredentialsManagementResponse";
const _EORS = "EnableOrganizationsRootSessions";
const _EORSR = "EnableOrganizationsRootSessionsRequest";
const _EORSRn = "EnableOrganizationsRootSessionsResponse";
const _EOWIF = "EnableOutboundWebIdentityFederation";
const _EOWIFR = "EnableOutboundWebIdentityFederationResponse";
const _EP = "EntityPath";
const _EPn = "EndPosition";
const _EPx = "ExpirePasswords";
const _ER = "EvaluationResult";
const _ERD = "EvalResourceDecision";
const _ERLT = "EvaluationResultsListType";
const _ERN = "EvalResourceName";
const _ERv = "EvaluationResults";
const _ET = "ExpirationTime";
const _ETUE = "EntityTemporarilyUnmodifiableException";
const _ETn = "EntityType";
const _En = "Encoding";
const _Ex = "Expiration";
const _F = "Filter";
const _FDE = "FeatureDisabledException";
const _FEE = "FeatureEnabledException";
const _Fi = "Fingerprint";
const _G = "Group";
const _GAAD = "GetAccountAuthorizationDetails";
const _GAADR = "GetAccountAuthorizationDetailsRequest";
const _GAADRe = "GetAccountAuthorizationDetailsResponse";
const _GAKLU = "GetAccessKeyLastUsed";
const _GAKLUR = "GetAccessKeyLastUsedRequest";
const _GAKLURe = "GetAccessKeyLastUsedResponse";
const _GAPP = "GetAccountPasswordPolicy";
const _GAPPR = "GetAccountPasswordPolicyResponse";
const _GAS = "GetAccountSummary";
const _GASR = "GetAccountSummaryResponse";
const _GCKFCP = "GetContextKeysForCustomPolicy";
const _GCKFCPR = "GetContextKeysForCustomPolicyRequest";
const _GCKFPP = "GetContextKeysForPrincipalPolicy";
const _GCKFPPR = "GetContextKeysForPrincipalPolicyRequest";
const _GCKFPR = "GetContextKeysForPolicyResponse";
const _GCR = "GenerateCredentialReport";
const _GCRR = "GenerateCredentialReportResponse";
const _GCRRe = "GetCredentialReportResponse";
const _GCRe = "GetCredentialReport";
const _GD = "GroupDetail";
const _GDL = "GroupDetailList";
const _GDR = "GetDelegationRequest";
const _GDRR = "GetDelegationRequestRequest";
const _GDRRe = "GetDelegationRequestResponse";
const _GETV = "GlobalEndpointTokenVersion";
const _GG = "GetGroup";
const _GGP = "GetGroupPolicy";
const _GGPR = "GetGroupPolicyRequest";
const _GGPRe = "GetGroupPolicyResponse";
const _GGR = "GetGroupRequest";
const _GGRe = "GetGroupResponse";
const _GHRS = "GetHumanReadableSummary";
const _GHRSR = "GetHumanReadableSummaryRequest";
const _GHRSRe = "GetHumanReadableSummaryResponse";
const _GI = "GroupId";
const _GIP = "GetInstanceProfile";
const _GIPR = "GetInstanceProfileRequest";
const _GIPRe = "GetInstanceProfileResponse";
const _GL = "GroupList";
const _GLP = "GetLoginProfile";
const _GLPR = "GetLoginProfileRequest";
const _GLPRe = "GetLoginProfileResponse";
const _GMFAD = "GetMFADevice";
const _GMFADR = "GetMFADeviceRequest";
const _GMFADRe = "GetMFADeviceResponse";
const _GN = "GroupName";
const _GOAR = "GenerateOrganizationsAccessReport";
const _GOARR = "GenerateOrganizationsAccessReportRequest";
const _GOARRe = "GenerateOrganizationsAccessReportResponse";
const _GOARRet = "GetOrganizationsAccessReportRequest";
const _GOARRetr = "GetOrganizationsAccessReportResponse";
const _GOARe = "GetOrganizationsAccessReport";
const _GOIDCP = "GetOpenIDConnectProvider";
const _GOIDCPR = "GetOpenIDConnectProviderRequest";
const _GOIDCPRe = "GetOpenIDConnectProviderResponse";
const _GOWIFI = "GetOutboundWebIdentityFederationInfo";
const _GOWIFIR = "GetOutboundWebIdentityFederationInfoResponse";
const _GP = "GetPolicy";
const _GPL = "GroupPolicyList";
const _GPR = "GetPolicyRequest";
const _GPRe = "GetPolicyResponse";
const _GPV = "GetPolicyVersion";
const _GPVR = "GetPolicyVersionRequest";
const _GPVRe = "GetPolicyVersionResponse";
const _GR = "GetRole";
const _GRP = "GetRolePolicy";
const _GRPR = "GetRolePolicyRequest";
const _GRPRe = "GetRolePolicyResponse";
const _GRR = "GetRoleRequest";
const _GRRe = "GetRoleResponse";
const _GSAMLP = "GetSAMLProvider";
const _GSAMLPR = "GetSAMLProviderRequest";
const _GSAMLPRe = "GetSAMLProviderResponse";
const _GSC = "GetServerCertificate";
const _GSCR = "GetServerCertificateRequest";
const _GSCRe = "GetServerCertificateResponse";
const _GSLAD = "GenerateServiceLastAccessedDetails";
const _GSLADR = "GenerateServiceLastAccessedDetailsRequest";
const _GSLADRe = "GenerateServiceLastAccessedDetailsResponse";
const _GSLADRet = "GetServiceLastAccessedDetailsRequest";
const _GSLADRete = "GetServiceLastAccessedDetailsResponse";
const _GSLADWE = "GetServiceLastAccessedDetailsWithEntities";
const _GSLADWER = "GetServiceLastAccessedDetailsWithEntitiesRequest";
const _GSLADWERe = "GetServiceLastAccessedDetailsWithEntitiesResponse";
const _GSLADe = "GetServiceLastAccessedDetails";
const _GSLRDS = "GetServiceLinkedRoleDeletionStatus";
const _GSLRDSR = "GetServiceLinkedRoleDeletionStatusRequest";
const _GSLRDSRe = "GetServiceLinkedRoleDeletionStatusResponse";
const _GSSHPK = "GetSSHPublicKey";
const _GSSHPKR = "GetSSHPublicKeyRequest";
const _GSSHPKRe = "GetSSHPublicKeyResponse";
const _GT = "GeneratedTime";
const _GU = "GetUser";
const _GUP = "GetUserPolicy";
const _GUPR = "GetUserPolicyRequest";
const _GUPRe = "GetUserPolicyResponse";
const _GUR = "GetUserRequest";
const _GURe = "GetUserResponse";
const _Gr = "Granularity";
const _Gro = "Groups";
const _HE = "HardExpiry";
const _I = "Id";
const _IA = "IsAttachable";
const _IACE = "InvalidAuthenticationCodeException";
const _ICE = "InvalidCertificateException";
const _IDV = "IsDefaultVersion";
const _II = "IssuerIdentifier";
const _IIE = "InvalidInputException";
const _IP = "InstanceProfile";
const _IPI = "InstanceProfileId";
const _IPKE = "InvalidPublicKeyException";
const _IPL = "InstanceProfileList";
const _IPN = "InstanceProfileName";
const _IPn = "InstanceProfiles";
const _IT = "IsTruncated";
const _IUTE = "InvalidUserTypeException";
const _JCD = "JobCreationDate";
const _JCDo = "JobCompletionDate";
const _JI = "JobId";
const _JS = "JobStatus";
const _JT = "JobType";
const _JVE = "JwtVendingEnabled";
const _K = "Key";
const _KI = "KeyId";
const _KPME = "KeyPairMismatchException";
const _L = "Locale";
const _LA = "LastAuthenticated";
const _LAA = "ListAccountAliases";
const _LAAR = "ListAccountAliasesRequest";
const _LAARi = "ListAccountAliasesResponse";
const _LAE = "LastAuthenticatedEntity";
const _LAEa = "LastAccessedEntity";
const _LAGP = "ListAttachedGroupPolicies";
const _LAGPR = "ListAttachedGroupPoliciesRequest";
const _LAGPRi = "ListAttachedGroupPoliciesResponse";
const _LAK = "ListAccessKeys";
const _LAKR = "ListAccessKeysRequest";
const _LAKRi = "ListAccessKeysResponse";
const _LAR = "LastAuthenticatedRegion";
const _LARP = "ListAttachedRolePolicies";
const _LARPR = "ListAttachedRolePoliciesRequest";
const _LARPRi = "ListAttachedRolePoliciesResponse";
const _LARa = "LastAccessedRegion";
const _LAT = "LastAuthenticatedTime";
const _LATa = "LastAccessedTime";
const _LAUP = "ListAttachedUserPolicies";
const _LAUPR = "ListAttachedUserPoliciesRequest";
const _LAUPRi = "ListAttachedUserPoliciesResponse";
const _LDR = "ListDelegationRequests";
const _LDRR = "ListDelegationRequestsRequest";
const _LDRRi = "ListDelegationRequestsResponse";
const _LEE = "LimitExceededException";
const _LEFP = "ListEntitiesForPolicy";
const _LEFPR = "ListEntitiesForPolicyRequest";
const _LEFPRi = "ListEntitiesForPolicyResponse";
const _LG = "ListGroups";
const _LGFU = "ListGroupsForUser";
const _LGFUR = "ListGroupsForUserRequest";
const _LGFURi = "ListGroupsForUserResponse";
const _LGP = "ListGroupPolicies";
const _LGPR = "ListGroupPoliciesRequest";
const _LGPRi = "ListGroupPoliciesResponse";
const _LGR = "ListGroupsRequest";
const _LGRi = "ListGroupsResponse";
const _LIP = "ListInstanceProfiles";
const _LIPFR = "ListInstanceProfilesForRole";
const _LIPFRR = "ListInstanceProfilesForRoleRequest";
const _LIPFRRi = "ListInstanceProfilesForRoleResponse";
const _LIPR = "ListInstanceProfilesRequest";
const _LIPRi = "ListInstanceProfilesResponse";
const _LIPT = "ListInstanceProfileTags";
const _LIPTR = "ListInstanceProfileTagsRequest";
const _LIPTRi = "ListInstanceProfileTagsResponse";
const _LMFAD = "ListMFADevices";
const _LMFADR = "ListMFADevicesRequest";
const _LMFADRi = "ListMFADevicesResponse";
const _LMFADT = "ListMFADeviceTags";
const _LMFADTR = "ListMFADeviceTagsRequest";
const _LMFADTRi = "ListMFADeviceTagsResponse";
const _LOF = "ListOrganizationsFeatures";
const _LOFR = "ListOrganizationsFeaturesRequest";
const _LOFRi = "ListOrganizationsFeaturesResponse";
const _LOIDCP = "ListOpenIDConnectProviders";
const _LOIDCPR = "ListOpenIDConnectProvidersRequest";
const _LOIDCPRi = "ListOpenIDConnectProvidersResponse";
const _LOIDCPT = "ListOpenIDConnectProviderTags";
const _LOIDCPTR = "ListOpenIDConnectProviderTagsRequest";
const _LOIDCPTRi = "ListOpenIDConnectProviderTagsResponse";
const _LP = "LoginProfile";
const _LPGSA = "ListPoliciesGrantingServiceAccess";
const _LPGSAE = "ListPoliciesGrantingServiceAccessEntry";
const _LPGSAR = "ListPoliciesGrantingServiceAccessRequest";
const _LPGSARi = "ListPoliciesGrantingServiceAccessResponse";
const _LPR = "ListPoliciesRequest";
const _LPRi = "ListPoliciesResponse";
const _LPT = "ListPolicyTags";
const _LPTR = "ListPolicyTagsRequest";
const _LPTRi = "ListPolicyTagsResponse";
const _LPV = "ListPolicyVersions";
const _LPVR = "ListPolicyVersionsRequest";
const _LPVRi = "ListPolicyVersionsResponse";
const _LPi = "ListPolicies";
const _LR = "ListRoles";
const _LRP = "ListRolePolicies";
const _LRPR = "ListRolePoliciesRequest";
const _LRPRi = "ListRolePoliciesResponse";
const _LRR = "ListRolesRequest";
const _LRRi = "ListRolesResponse";
const _LRT = "ListRoleTags";
const _LRTR = "ListRoleTagsRequest";
const _LRTRi = "ListRoleTagsResponse";
const _LSAMLP = "ListSAMLProviders";
const _LSAMLPR = "ListSAMLProvidersRequest";
const _LSAMLPRi = "ListSAMLProvidersResponse";
const _LSAMLPT = "ListSAMLProviderTags";
const _LSAMLPTR = "ListSAMLProviderTagsRequest";
const _LSAMLPTRi = "ListSAMLProviderTagsResponse";
const _LSC = "ListServerCertificates";
const _LSCR = "ListServerCertificatesRequest";
const _LSCRi = "ListServerCertificatesResponse";
const _LSCRis = "ListSigningCertificatesRequest";
const _LSCRist = "ListSigningCertificatesResponse";
const _LSCT = "ListServerCertificateTags";
const _LSCTR = "ListServerCertificateTagsRequest";
const _LSCTRi = "ListServerCertificateTagsResponse";
const _LSCi = "ListSigningCertificates";
const _LSSC = "ListServiceSpecificCredentials";
const _LSSCR = "ListServiceSpecificCredentialsRequest";
const _LSSCRi = "ListServiceSpecificCredentialsResponse";
const _LSSHPK = "ListSSHPublicKeys";
const _LSSHPKR = "ListSSHPublicKeysRequest";
const _LSSHPKRi = "ListSSHPublicKeysResponse";
const _LU = "ListUsers";
const _LUD = "LastUsedDate";
const _LUP = "ListUserPolicies";
const _LUPR = "ListUserPoliciesRequest";
const _LUPRi = "ListUserPoliciesResponse";
const _LUR = "ListUsersRequest";
const _LURi = "ListUsersResponse";
const _LUT = "ListUserTags";
const _LUTR = "ListUserTagsRequest";
const _LUTRi = "ListUserTagsResponse";
const _LVMFAD = "ListVirtualMFADevices";
const _LVMFADR = "ListVirtualMFADevicesRequest";
const _LVMFADRi = "ListVirtualMFADevicesResponse";
const _Li = "Line";
const _M = "Message";
const _MCE = "MalformedCertificateException";
const _MCV = "MissingContextValues";
const _MFAD = "MFADevices";
const _MFADe = "MFADevice";
const _MI = "MaxItems";
const _MPA = "MaxPasswordAge";
const _MPD = "ManagedPolicyDetail";
const _MPDE = "MalformedPolicyDocumentException";
const _MPDLT = "ManagedPolicyDetailListType";
const _MPL = "MinimumPasswordLength";
const _MS = "MatchedStatements";
const _MSD = "MaxSessionDuration";
const _Ma = "Marker";
const _N = "Name";
const _NC = "NotificationChannel";
const _NGN = "NewGroupName";
const _NOSA = "NumberOfServicesAccessible";
const _NOSNA = "NumberOfServicesNotAccessed";
const _NP = "NewPassword";
const _NPe = "NewPath";
const _NSCN = "NewServerCertificateName";
const _NSEE = "NoSuchEntityException";
const _NUN = "NewUserName";
const _No = "Notes";
const _OA = "OnlyAttached";
const _OAI = "OwnerAccountId";
const _ODD = "OrganizationsDecisionDetail";
const _OI = "OwnerId";
const _OIDCPA = "OpenIDConnectProviderArn";
const _OIDCPL = "OpenIDConnectProviderList";
const _OIDCPLE = "OpenIDConnectProviderListEntry";
const _OIDCPLT = "OpenIDConnectProviderListType";
const _OIICEE = "OpenIdIdpCommunicationErrorException";
const _OIr = "OrganizationId";
const _ONFE = "OrganizationNotFoundException";
const _ONIAFME = "OrganizationNotInAllFeaturesModeException";
const _OP = "OldPassword";
const _OPI = "OrganizationsPolicyId";
const _OSBO = "OnlySendByOwner";
const _P = "Permissions";
const _PA = "PolicyArn";
const _PB = "PermissionsBoundary";
const _PBA = "PermissionsBoundaryArn";
const _PBDD = "PermissionsBoundaryDecisionDetail";
const _PBPIL = "PermissionsBoundaryPolicyInputList";
const _PBT = "PermissionsBoundaryType";
const _PBUC = "PermissionsBoundaryUsageCount";
const _PCR = "PermissionCheckResult";
const _PCS = "PermissionCheckStatus";
const _PD = "PolicyDocument";
const _PDo = "PolicyDetail";
const _PEE = "PolicyEvaluationException";
const _PG = "PolicyGroups";
const _PGLT = "PolicyGroupListType";
const _PGP = "PutGroupPolicy";
const _PGPR = "PutGroupPolicyRequest";
const _PGSA = "PoliciesGrantingServiceAccess";
const _PGSAo = "PolicyGrantingServiceAccess";
const _PGo = "PolicyGroup";
const _PI = "PolicyId";
const _PIL = "PolicyInputList";
const _PK = "PrivateKey";
const _PKL = "PrivateKeyList";
const _PLU = "PasswordLastUsed";
const _PN = "PolicyName";
const _PNAE = "PolicyNotAttachableException";
const _PNo = "PolicyNames";
const _PP = "PermissionPolicy";
const _PPVE = "PasswordPolicyViolationException";
const _PPa = "PasswordPolicy";
const _PPat = "PathPrefix";
const _PPo = "PolicyParameter";
const _PR = "PolicyRoles";
const _PRLT = "PolicyRoleListType";
const _PRP = "PasswordReusePrevention";
const _PRPB = "PutRolePermissionsBoundary";
const _PRPBR = "PutRolePermissionsBoundaryRequest";
const _PRPR = "PutRolePolicyRequest";
const _PRPu = "PutRolePolicy";
const _PRR = "PasswordResetRequired";
const _PRo = "PolicyRole";
const _PSA = "PolicySourceArn";
const _PT = "PolicyType";
const _PTA = "PolicyTemplateArn";
const _PU = "PolicyUsers";
const _PUF = "PolicyUsageFilter";
const _PULT = "PolicyUserListType";
const _PUP = "PutUserPolicy";
const _PUPB = "PutUserPermissionsBoundary";
const _PUPBR = "PutUserPermissionsBoundaryRequest";
const _PUPR = "PutUserPolicyRequest";
const _PUo = "PolicyUser";
const _PV = "PolicyVersion";
const _PVL = "PolicyVersionList";
const _Pa = "Path";
const _Par = "Parameters";
const _Pas = "Password";
const _Po = "Policy";
const _Pol = "Policies";
const _Pos = "Position";
const _QRCPNG = "QRCodePNG";
const _R = "Region";
const _RA = "ResourceArns";
const _RCIDFOIDCP = "RemoveClientIDFromOpenIDConnectProvider";
const _RCIDFOIDCPR = "RemoveClientIDFromOpenIDConnectProviderRequest";
const _RD = "RoleDetail";
const _RDL = "RoleDetailList";
const _RDR = "RejectDelegationRequest";
const _RDRR = "RejectDelegationRequestRequest";
const _RF = "ReportFormat";
const _RGLEE = "ReportGenerationLimitExceededException";
const _RHO = "ResourceHandlingOption";
const _RI = "RequestorId";
const _RIo = "RoleId";
const _RLC = "RequireLowercaseCharacters";
const _RLU = "RoleLastUsed";
const _RM = "RequestMessage";
const _RMFAD = "ResyncMFADevice";
const _RMFADR = "ResyncMFADeviceRequest";
const _RN = "RoleName";
const _RNe = "RequestorName";
const _RNeq = "RequireNumbers";
const _RO = "ResourceOwner";
const _RP = "ResourcePolicy";
const _RPK = "RemovePrivateKey";
const _RPL = "RolePolicyList";
const _RPRA = "RolePermissionRestrictionArns";
const _RR = "RejectionReason";
const _RRFIP = "RemoveRoleFromInstanceProfile";
const _RRFIPR = "RemoveRoleFromInstanceProfileRequest";
const _RS = "RequireSymbols";
const _RSR = "ResourceSpecificResults";
const _RSRLT = "ResourceSpecificResultListType";
const _RSRe = "ResourceSpecificResult";
const _RSSC = "ResetServiceSpecificCredential";
const _RSSCR = "ResetServiceSpecificCredentialRequest";
const _RSSCRe = "ResetServiceSpecificCredentialResponse";
const _RU = "RedirectUrl";
const _RUC = "RequireUppercaseCharacters";
const _RUFG = "RemoveUserFromGroup";
const _RUFGR = "RemoveUserFromGroupRequest";
const _RUL = "RoleUsageList";
const _RULT = "RoleUsageListType";
const _RUT = "RoleUsageType";
const _RWI = "RequestorWorkflowId";
const _Re = "Reason";
const _Res = "Resources";
const _Ro = "Role";
const _Rol = "Roles";
const _S = "Status";
const _SAD = "SetAsDefault";
const _SAK = "SecretAccessKey";
const _SAMLMD = "SAMLMetadataDocument";
const _SAMLPA = "SAMLProviderArn";
const _SAMLPK = "SAMLPrivateKey";
const _SAMLPL = "SAMLProviderList";
const _SAMLPLE = "SAMLProviderListEntry";
const _SAMLPLT = "SAMLProviderListType";
const _SAMLPUUID = "SAMLProviderUUID";
const _SANEE = "ServiceAccessNotEnabledException";
const _SC = "SummaryContent";
const _SCA = "ServiceCredentialAlias";
const _SCI = "ServerCertificateId";
const _SCM = "ServerCertificateMetadata";
const _SCML = "ServerCertificateMetadataList";
const _SCN = "ServerCertificateName";
const _SCP = "SimulateCustomPolicy";
const _SCPR = "SimulateCustomPolicyRequest";
const _SCS = "ServiceCredentialSecret";
const _SCe = "ServerCertificate";
const _SCi = "SigningCertificate";
const _SD = "SessionDuration";
const _SDPV = "SetDefaultPolicyVersion";
const _SDPVR = "SetDefaultPolicyVersionRequest";
const _SDT = "SendDelegationToken";
const _SDTR = "SendDelegationTokenRequest";
const _SFE = "ServiceFailureException";
const _SK = "SortKey";
const _SLA = "ServicesLastAccessed";
const _SLAe = "ServiceLastAccessed";
const _SLT = "StatementListType";
const _SM = "SummaryMap";
const _SN = "ServiceName";
const _SNSE = "ServiceNotSupportedException";
const _SNe = "ServiceNamespace";
const _SNer = "SerialNumber";
const _SNerv = "ServiceNamespaces";
const _SP = "ServicePassword";
const _SPI = "SourcePolicyId";
const _SPP = "SimulatePrincipalPolicy";
const _SPPR = "SimulatePrincipalPolicyRequest";
const _SPR = "SimulatePolicyResponse";
const _SPT = "SourcePolicyType";
const _SPt = "StartPosition";
const _SS = "SummaryState";
const _SSC = "ServiceSpecificCredential";
const _SSCI = "ServiceSpecificCredentialId";
const _SSCLT = "ServiceSpecificCredentialsListType";
const _SSCM = "ServiceSpecificCredentialMetadata";
const _SSCe = "ServiceSpecificCredentials";
const _SSHPK = "SSHPublicKey";
const _SSHPKB = "SSHPublicKeyBody";
const _SSHPKI = "SSHPublicKeyId";
const _SSHPKLT = "SSHPublicKeyListType";
const _SSHPKM = "SSHPublicKeyMetadata";
const _SSHPKu = "SSHPublicKeys";
const _SSTSP = "SetSecurityTokenServicePreferences";
const _SSTSPR = "SetSecurityTokenServicePreferencesRequest";
const _SUN = "ServiceUserName";
const _Sc = "Scope";
const _St = "State";
const _Sta = "Statement";
const _T = "Tags";
const _TAE = "TotalAuthenticatedEntities";
const _TALA = "TrackedActionsLastAccessed";
const _TALAr = "TrackedActionLastAccessed";
const _TIP = "TagInstanceProfile";
const _TIPR = "TagInstanceProfileRequest";
const _TK = "TagKeys";
const _TL = "ThumbprintList";
const _TMFAD = "TagMFADevice";
const _TMFADR = "TagMFADeviceRequest";
const _TOIDCP = "TagOpenIDConnectProvider";
const _TOIDCPR = "TagOpenIDConnectProviderRequest";
const _TP = "TagPolicy";
const _TPR = "TagPolicyRequest";
const _TR = "TagRole";
const _TRR = "TagRoleRequest";
const _TSAMLP = "TagSAMLProvider";
const _TSAMLPR = "TagSAMLProviderRequest";
const _TSC = "TagServerCertificate";
const _TSCR = "TagServerCertificateRequest";
const _TU = "TagUser";
const _TUR = "TagUserRequest";
const _Ta = "Tag";
const _Ti = "Timestamp";
const _Ty = "Type";
const _U = "Url";
const _UAK = "UpdateAccessKey";
const _UAKR = "UpdateAccessKeyRequest";
const _UAPP = "UpdateAccountPasswordPolicy";
const _UAPPR = "UpdateAccountPasswordPolicyRequest";
const _UARP = "UpdateAssumeRolePolicy";
const _UARPR = "UpdateAssumeRolePolicyRequest";
const _UD = "UpdateDate";
const _UDL = "UserDetailList";
const _UDR = "UpdateDelegationRequest";
const _UDRR = "UpdateDelegationRequestRequest";
const _UDp = "UploadDate";
const _UDs = "UserDetail";
const _UEE = "UnmodifiableEntityException";
const _UG = "UpdateGroup";
const _UGR = "UpdateGroupRequest";
const _UI = "UserId";
const _UIP = "UntagInstanceProfile";
const _UIPR = "UntagInstanceProfileRequest";
const _ULP = "UpdateLoginProfile";
const _ULPR = "UpdateLoginProfileRequest";
const _UMFAD = "UntagMFADevice";
const _UMFADR = "UntagMFADeviceRequest";
const _UN = "UserName";
const _UOIDCP = "UntagOpenIDConnectProvider";
const _UOIDCPR = "UntagOpenIDConnectProviderRequest";
const _UOIDCPT = "UpdateOpenIDConnectProviderThumbprint";
const _UOIDCPTR = "UpdateOpenIDConnectProviderThumbprintRequest";
const _UP = "UntagPolicy";
const _UPKEE = "UnrecognizedPublicKeyEncodingException";
const _UPL = "UserPolicyList";
const _UPR = "UntagPolicyRequest";
const _UR = "UntagRole";
const _URD = "UpdateRoleDescription";
const _URDR = "UpdateRoleDescriptionRequest";
const _URDRp = "UpdateRoleDescriptionResponse";
const _URR = "UntagRoleRequest";
const _URRp = "UpdateRoleRequest";
const _URRpd = "UpdateRoleResponse";
const _URp = "UpdateRole";
const _USAMLP = "UntagSAMLProvider";
const _USAMLPR = "UntagSAMLProviderRequest";
const _USAMLPRp = "UpdateSAMLProviderRequest";
const _USAMLPRpd = "UpdateSAMLProviderResponse";
const _USAMLPp = "UpdateSAMLProvider";
const _USC = "UntagServerCertificate";
const _USCR = "UntagServerCertificateRequest";
const _USCRp = "UpdateServerCertificateRequest";
const _USCRpd = "UpdateSigningCertificateRequest";
const _USCRpl = "UploadServerCertificateRequest";
const _USCRplo = "UploadServerCertificateResponse";
const _USCRploa = "UploadSigningCertificateRequest";
const _USCRpload = "UploadSigningCertificateResponse";
const _USCp = "UpdateServerCertificate";
const _USCpd = "UpdateSigningCertificate";
const _USCpl = "UploadServerCertificate";
const _USCplo = "UploadSigningCertificate";
const _USSC = "UpdateServiceSpecificCredential";
const _USSCR = "UpdateServiceSpecificCredentialRequest";
const _USSHPK = "UpdateSSHPublicKey";
const _USSHPKR = "UpdateSSHPublicKeyRequest";
const _USSHPKRp = "UploadSSHPublicKeyRequest";
const _USSHPKRpl = "UploadSSHPublicKeyResponse";
const _USSHPKp = "UploadSSHPublicKey";
const _UT = "UpdatedTime";
const _UU = "UntagUser";
const _UUR = "UntagUserRequest";
const _UURp = "UpdateUserRequest";
const _UUp = "UpdateUser";
const _Us = "User";
const _Use = "Users";
const _V = "Versions";
const _VI = "VersionId";
const _VMFAD = "VirtualMFADevice";
const _VMFADN = "VirtualMFADeviceName";
const _VMFADi = "VirtualMFADevices";
const _VU = "ValidUntil";
const _Va = "Values";
const _Val = "Value";
const _aKMLT = "accessKeyMetadataListType";
const _aKST = "accessKeySecretType";
const _aPLT = "attachedPoliciesListType";
const _aQE = "awsQueryError";
const _c = "client";
const _cLT = "certificateListType";
const _dRLT = "delegationRequestsListType";
const _e = "error";
const _eDLT = "entityDetailsListType";
const _gDLT = "groupDetailListType";
const _gLT = "groupListType";
const _hE = "httpError";
const _iPLT = "instanceProfileListType";
const _iT = "isTruncated";
const _lPGSARLT = "listPolicyGrantingServiceAccessResponseListType";
const _m = "message";
const _mDLT = "mfaDeviceListType";
const _pDLT = "policyDetailListType";
const _pDVLT = "policyDocumentVersionListType";
const _pGSALT = "policyGrantingServiceAccessListType";
const _pKL = "privateKeyList";
const _pKT = "privateKeyType";
const _pLT = "policyListType";
const _pPLT = "policyParameterListType";
const _pT = "passwordType";
const _rDLT = "roleDetailListType";
const _rLT = "roleListType";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.iam";
const _sCMLT = "serverCertificateMetadataListType";
const _sCS = "serviceCredentialSecret";
const _sP = "servicePassword";
const _se = "server";
const _tLT = "tagListType";
const _uDLT = "userDetailListType";
const _uLT = "userListType";
const _vMFADLT = "virtualMFADeviceListType";
const n0 = "com.amazonaws.iam";
const { TypeRegistry } = require("@smithy/core/schema");
const { AccountNotManagementOrDelegatedAdministratorException, CallerIsNotManagementAccountException, ConcurrentModificationException, CredentialReportExpiredException, CredentialReportNotPresentException, CredentialReportNotReadyException, DeleteConflictException, DuplicateCertificateException, DuplicateSSHPublicKeyException, EntityAlreadyExistsException, EntityTemporarilyUnmodifiableException, FeatureDisabledException, FeatureEnabledException, InvalidAuthenticationCodeException, InvalidCertificateException, InvalidInputException, InvalidPublicKeyException, InvalidUserTypeException, KeyPairMismatchException, LimitExceededException, MalformedCertificateException, MalformedPolicyDocumentException, NoSuchEntityException, OpenIdIdpCommunicationErrorException, OrganizationNotFoundException, OrganizationNotInAllFeaturesModeException, PasswordPolicyViolationException, PolicyEvaluationException, PolicyNotAttachableException, ReportGenerationLimitExceededException, ServiceAccessNotEnabledException, ServiceFailureException, ServiceNotSupportedException, UnmodifiableEntityException, UnrecognizedPublicKeyEncodingException } = require("../models/errors");
const { IAMServiceException } = require("../models/IAMServiceException");
const _s_registry = TypeRegistry.for(_s);
const IAMServiceException$ = [-3, _s, "IAMServiceException", 0, [], []];
exports.IAMServiceException$ = IAMServiceException$;
_s_registry.registerError(IAMServiceException$, IAMServiceException);
const n0_registry = TypeRegistry.for(n0);
const AccountNotManagementOrDelegatedAdministratorException$ = [-3, n0, _ANMODAE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
exports.AccountNotManagementOrDelegatedAdministratorException$ = AccountNotManagementOrDelegatedAdministratorException$;
n0_registry.registerError(AccountNotManagementOrDelegatedAdministratorException$, AccountNotManagementOrDelegatedAdministratorException);
const CallerIsNotManagementAccountException$ = [-3, n0, _CINMAE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
exports.CallerIsNotManagementAccountException$ = CallerIsNotManagementAccountException$;
n0_registry.registerError(CallerIsNotManagementAccountException$, CallerIsNotManagementAccountException);
const ConcurrentModificationException$ = [-3, n0, _CME,
    { [_aQE]: [`ConcurrentModification`, 409], [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.ConcurrentModificationException$ = ConcurrentModificationException$;
n0_registry.registerError(ConcurrentModificationException$, ConcurrentModificationException);
const CredentialReportExpiredException$ = [-3, n0, _CREE,
    { [_aQE]: [`ReportExpired`, 410], [_e]: _c, [_hE]: 410 },
    [_m],
    [0]
];
exports.CredentialReportExpiredException$ = CredentialReportExpiredException$;
n0_registry.registerError(CredentialReportExpiredException$, CredentialReportExpiredException);
const CredentialReportNotPresentException$ = [-3, n0, _CRNPE,
    { [_aQE]: [`ReportNotPresent`, 410], [_e]: _c, [_hE]: 410 },
    [_m],
    [0]
];
exports.CredentialReportNotPresentException$ = CredentialReportNotPresentException$;
n0_registry.registerError(CredentialReportNotPresentException$, CredentialReportNotPresentException);
const CredentialReportNotReadyException$ = [-3, n0, _CRNRE,
    { [_aQE]: [`ReportInProgress`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
exports.CredentialReportNotReadyException$ = CredentialReportNotReadyException$;
n0_registry.registerError(CredentialReportNotReadyException$, CredentialReportNotReadyException);
const DeleteConflictException$ = [-3, n0, _DCE,
    { [_aQE]: [`DeleteConflict`, 409], [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.DeleteConflictException$ = DeleteConflictException$;
n0_registry.registerError(DeleteConflictException$, DeleteConflictException);
const DuplicateCertificateException$ = [-3, n0, _DCEu,
    { [_aQE]: [`DuplicateCertificate`, 409], [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.DuplicateCertificateException$ = DuplicateCertificateException$;
n0_registry.registerError(DuplicateCertificateException$, DuplicateCertificateException);
const DuplicateSSHPublicKeyException$ = [-3, n0, _DSSHPKE,
    { [_aQE]: [`DuplicateSSHPublicKey`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.DuplicateSSHPublicKeyException$ = DuplicateSSHPublicKeyException$;
n0_registry.registerError(DuplicateSSHPublicKeyException$, DuplicateSSHPublicKeyException);
const EntityAlreadyExistsException$ = [-3, n0, _EAEE,
    { [_aQE]: [`EntityAlreadyExists`, 409], [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.EntityAlreadyExistsException$ = EntityAlreadyExistsException$;
n0_registry.registerError(EntityAlreadyExistsException$, EntityAlreadyExistsException);
const EntityTemporarilyUnmodifiableException$ = [-3, n0, _ETUE,
    { [_aQE]: [`EntityTemporarilyUnmodifiable`, 409], [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.EntityTemporarilyUnmodifiableException$ = EntityTemporarilyUnmodifiableException$;
n0_registry.registerError(EntityTemporarilyUnmodifiableException$, EntityTemporarilyUnmodifiableException);
const FeatureDisabledException$ = [-3, n0, _FDE,
    { [_aQE]: [`FeatureDisabled`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
exports.FeatureDisabledException$ = FeatureDisabledException$;
n0_registry.registerError(FeatureDisabledException$, FeatureDisabledException);
const FeatureEnabledException$ = [-3, n0, _FEE,
    { [_aQE]: [`FeatureEnabled`, 409], [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.FeatureEnabledException$ = FeatureEnabledException$;
n0_registry.registerError(FeatureEnabledException$, FeatureEnabledException);
const InvalidAuthenticationCodeException$ = [-3, n0, _IACE,
    { [_aQE]: [`InvalidAuthenticationCode`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
exports.InvalidAuthenticationCodeException$ = InvalidAuthenticationCodeException$;
n0_registry.registerError(InvalidAuthenticationCodeException$, InvalidAuthenticationCodeException);
const InvalidCertificateException$ = [-3, n0, _ICE,
    { [_aQE]: [`InvalidCertificate`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidCertificateException$ = InvalidCertificateException$;
n0_registry.registerError(InvalidCertificateException$, InvalidCertificateException);
const InvalidInputException$ = [-3, n0, _IIE,
    { [_aQE]: [`InvalidInput`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidInputException$ = InvalidInputException$;
n0_registry.registerError(InvalidInputException$, InvalidInputException);
const InvalidPublicKeyException$ = [-3, n0, _IPKE,
    { [_aQE]: [`InvalidPublicKey`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidPublicKeyException$ = InvalidPublicKeyException$;
n0_registry.registerError(InvalidPublicKeyException$, InvalidPublicKeyException);
const InvalidUserTypeException$ = [-3, n0, _IUTE,
    { [_aQE]: [`InvalidUserType`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidUserTypeException$ = InvalidUserTypeException$;
n0_registry.registerError(InvalidUserTypeException$, InvalidUserTypeException);
const KeyPairMismatchException$ = [-3, n0, _KPME,
    { [_aQE]: [`KeyPairMismatch`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.KeyPairMismatchException$ = KeyPairMismatchException$;
n0_registry.registerError(KeyPairMismatchException$, KeyPairMismatchException);
const LimitExceededException$ = [-3, n0, _LEE,
    { [_aQE]: [`LimitExceeded`, 409], [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.LimitExceededException$ = LimitExceededException$;
n0_registry.registerError(LimitExceededException$, LimitExceededException);
const MalformedCertificateException$ = [-3, n0, _MCE,
    { [_aQE]: [`MalformedCertificate`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.MalformedCertificateException$ = MalformedCertificateException$;
n0_registry.registerError(MalformedCertificateException$, MalformedCertificateException);
const MalformedPolicyDocumentException$ = [-3, n0, _MPDE,
    { [_aQE]: [`MalformedPolicyDocument`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.MalformedPolicyDocumentException$ = MalformedPolicyDocumentException$;
n0_registry.registerError(MalformedPolicyDocumentException$, MalformedPolicyDocumentException);
const NoSuchEntityException$ = [-3, n0, _NSEE,
    { [_aQE]: [`NoSuchEntity`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
exports.NoSuchEntityException$ = NoSuchEntityException$;
n0_registry.registerError(NoSuchEntityException$, NoSuchEntityException);
const OpenIdIdpCommunicationErrorException$ = [-3, n0, _OIICEE,
    { [_aQE]: [`OpenIdIdpCommunicationError`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.OpenIdIdpCommunicationErrorException$ = OpenIdIdpCommunicationErrorException$;
n0_registry.registerError(OpenIdIdpCommunicationErrorException$, OpenIdIdpCommunicationErrorException);
const OrganizationNotFoundException$ = [-3, n0, _ONFE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
exports.OrganizationNotFoundException$ = OrganizationNotFoundException$;
n0_registry.registerError(OrganizationNotFoundException$, OrganizationNotFoundException);
const OrganizationNotInAllFeaturesModeException$ = [-3, n0, _ONIAFME,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
exports.OrganizationNotInAllFeaturesModeException$ = OrganizationNotInAllFeaturesModeException$;
n0_registry.registerError(OrganizationNotInAllFeaturesModeException$, OrganizationNotInAllFeaturesModeException);
const PasswordPolicyViolationException$ = [-3, n0, _PPVE,
    { [_aQE]: [`PasswordPolicyViolation`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.PasswordPolicyViolationException$ = PasswordPolicyViolationException$;
n0_registry.registerError(PasswordPolicyViolationException$, PasswordPolicyViolationException);
const PolicyEvaluationException$ = [-3, n0, _PEE,
    { [_aQE]: [`PolicyEvaluation`, 500], [_e]: _se, [_hE]: 500 },
    [_m],
    [0]
];
exports.PolicyEvaluationException$ = PolicyEvaluationException$;
n0_registry.registerError(PolicyEvaluationException$, PolicyEvaluationException);
const PolicyNotAttachableException$ = [-3, n0, _PNAE,
    { [_aQE]: [`PolicyNotAttachable`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.PolicyNotAttachableException$ = PolicyNotAttachableException$;
n0_registry.registerError(PolicyNotAttachableException$, PolicyNotAttachableException);
const ReportGenerationLimitExceededException$ = [-3, n0, _RGLEE,
    { [_aQE]: [`ReportGenerationLimitExceeded`, 409], [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.ReportGenerationLimitExceededException$ = ReportGenerationLimitExceededException$;
n0_registry.registerError(ReportGenerationLimitExceededException$, ReportGenerationLimitExceededException);
const ServiceAccessNotEnabledException$ = [-3, n0, _SANEE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
exports.ServiceAccessNotEnabledException$ = ServiceAccessNotEnabledException$;
n0_registry.registerError(ServiceAccessNotEnabledException$, ServiceAccessNotEnabledException);
const ServiceFailureException$ = [-3, n0, _SFE,
    { [_aQE]: [`ServiceFailure`, 500], [_e]: _se, [_hE]: 500 },
    [_m],
    [0]
];
exports.ServiceFailureException$ = ServiceFailureException$;
n0_registry.registerError(ServiceFailureException$, ServiceFailureException);
const ServiceNotSupportedException$ = [-3, n0, _SNSE,
    { [_aQE]: [`NotSupportedService`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
exports.ServiceNotSupportedException$ = ServiceNotSupportedException$;
n0_registry.registerError(ServiceNotSupportedException$, ServiceNotSupportedException);
const UnmodifiableEntityException$ = [-3, n0, _UEE,
    { [_aQE]: [`UnmodifiableEntity`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UnmodifiableEntityException$ = UnmodifiableEntityException$;
n0_registry.registerError(UnmodifiableEntityException$, UnmodifiableEntityException);
const UnrecognizedPublicKeyEncodingException$ = [-3, n0, _UPKEE,
    { [_aQE]: [`UnrecognizedPublicKeyEncoding`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UnrecognizedPublicKeyEncodingException$ = UnrecognizedPublicKeyEncodingException$;
n0_registry.registerError(UnrecognizedPublicKeyEncodingException$, UnrecognizedPublicKeyEncodingException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var accessKeySecretType = [0, n0, _aKST, 8, 0];
var BootstrapDatum = [0, n0, _BD, 8, 21];
var passwordType = [0, n0, _pT, 8, 0];
var privateKeyType = [0, n0, _pKT, 8, 0];
var serviceCredentialSecret = [0, n0, _sCS, 8, 0];
var servicePassword = [0, n0, _sP, 8, 0];
const AcceptDelegationRequestRequest$ = [3, n0, _ADRR,
    0,
    [_DRI],
    [0], 1
];
exports.AcceptDelegationRequestRequest$ = AcceptDelegationRequestRequest$;
const AccessDetail$ = [3, n0, _AD,
    0,
    [_SN, _SNe, _R, _EP, _LAT, _TAE],
    [0, 0, 0, 0, 4, 1], 2
];
exports.AccessDetail$ = AccessDetail$;
const AccessKey$ = [3, n0, _AK,
    0,
    [_UN, _AKI, _S, _SAK, _CD],
    [0, 0, 0, [() => accessKeySecretType, 0], 4], 4
];
exports.AccessKey$ = AccessKey$;
const AccessKeyLastUsed$ = [3, n0, _AKLU,
    0,
    [_SN, _R, _LUD],
    [0, 0, 4], 2
];
exports.AccessKeyLastUsed$ = AccessKeyLastUsed$;
const AccessKeyMetadata$ = [3, n0, _AKM,
    0,
    [_UN, _AKI, _S, _CD],
    [0, 0, 0, 4]
];
exports.AccessKeyMetadata$ = AccessKeyMetadata$;
const AddClientIDToOpenIDConnectProviderRequest$ = [3, n0, _ACIDTOIDCPR,
    0,
    [_OIDCPA, _CID],
    [0, 0], 2
];
exports.AddClientIDToOpenIDConnectProviderRequest$ = AddClientIDToOpenIDConnectProviderRequest$;
const AddRoleToInstanceProfileRequest$ = [3, n0, _ARTIPR,
    0,
    [_IPN, _RN],
    [0, 0], 2
];
exports.AddRoleToInstanceProfileRequest$ = AddRoleToInstanceProfileRequest$;
const AddUserToGroupRequest$ = [3, n0, _AUTGR,
    0,
    [_GN, _UN],
    [0, 0], 2
];
exports.AddUserToGroupRequest$ = AddUserToGroupRequest$;
const AssociateDelegationRequestRequest$ = [3, n0, _ADRRs,
    0,
    [_DRI],
    [0], 1
];
exports.AssociateDelegationRequestRequest$ = AssociateDelegationRequestRequest$;
const AttachedPermissionsBoundary$ = [3, n0, _APB,
    0,
    [_PBT, _PBA],
    [0, 0]
];
exports.AttachedPermissionsBoundary$ = AttachedPermissionsBoundary$;
const AttachedPolicy$ = [3, n0, _AP,
    0,
    [_PN, _PA],
    [0, 0]
];
exports.AttachedPolicy$ = AttachedPolicy$;
const AttachGroupPolicyRequest$ = [3, n0, _AGPR,
    0,
    [_GN, _PA],
    [0, 0], 2
];
exports.AttachGroupPolicyRequest$ = AttachGroupPolicyRequest$;
const AttachRolePolicyRequest$ = [3, n0, _ARPR,
    0,
    [_RN, _PA],
    [0, 0], 2
];
exports.AttachRolePolicyRequest$ = AttachRolePolicyRequest$;
const AttachUserPolicyRequest$ = [3, n0, _AUPR,
    0,
    [_UN, _PA],
    [0, 0], 2
];
exports.AttachUserPolicyRequest$ = AttachUserPolicyRequest$;
const ChangePasswordRequest$ = [3, n0, _CPR,
    0,
    [_OP, _NP],
    [[() => passwordType, 0], [() => passwordType, 0]], 2
];
exports.ChangePasswordRequest$ = ChangePasswordRequest$;
const ContextEntry$ = [3, n0, _CE,
    0,
    [_CKN, _CKV, _CKT],
    [0, 64 | 0, 0]
];
exports.ContextEntry$ = ContextEntry$;
const CreateAccessKeyRequest$ = [3, n0, _CAKR,
    0,
    [_UN],
    [0]
];
exports.CreateAccessKeyRequest$ = CreateAccessKeyRequest$;
const CreateAccessKeyResponse$ = [3, n0, _CAKRr,
    0,
    [_AK],
    [[() => AccessKey$, 0]], 1
];
exports.CreateAccessKeyResponse$ = CreateAccessKeyResponse$;
const CreateAccountAliasRequest$ = [3, n0, _CAAR,
    0,
    [_AA],
    [0], 1
];
exports.CreateAccountAliasRequest$ = CreateAccountAliasRequest$;
const CreateDelegationRequestRequest$ = [3, n0, _CDRR,
    0,
    [_D, _P, _RWI, _NC, _SD, _OAI, _RM, _RU, _OSBO],
    [0, () => DelegationPermission$, 0, 0, 1, 0, 0, 0, 2], 5
];
exports.CreateDelegationRequestRequest$ = CreateDelegationRequestRequest$;
const CreateDelegationRequestResponse$ = [3, n0, _CDRRr,
    0,
    [_CDL, _DRI],
    [0, 0]
];
exports.CreateDelegationRequestResponse$ = CreateDelegationRequestResponse$;
const CreateGroupRequest$ = [3, n0, _CGR,
    0,
    [_GN, _Pa],
    [0, 0], 1
];
exports.CreateGroupRequest$ = CreateGroupRequest$;
const CreateGroupResponse$ = [3, n0, _CGRr,
    0,
    [_G],
    [() => Group$], 1
];
exports.CreateGroupResponse$ = CreateGroupResponse$;
const CreateInstanceProfileRequest$ = [3, n0, _CIPR,
    0,
    [_IPN, _Pa, _T],
    [0, 0, () => tagListType], 1
];
exports.CreateInstanceProfileRequest$ = CreateInstanceProfileRequest$;
const CreateInstanceProfileResponse$ = [3, n0, _CIPRr,
    0,
    [_IP],
    [() => InstanceProfile$], 1
];
exports.CreateInstanceProfileResponse$ = CreateInstanceProfileResponse$;
const CreateLoginProfileRequest$ = [3, n0, _CLPR,
    0,
    [_UN, _Pas, _PRR],
    [0, [() => passwordType, 0], 2]
];
exports.CreateLoginProfileRequest$ = CreateLoginProfileRequest$;
const CreateLoginProfileResponse$ = [3, n0, _CLPRr,
    0,
    [_LP],
    [() => LoginProfile$], 1
];
exports.CreateLoginProfileResponse$ = CreateLoginProfileResponse$;
const CreateOpenIDConnectProviderRequest$ = [3, n0, _COIDCPR,
    0,
    [_U, _CIDL, _TL, _T],
    [0, 64 | 0, 64 | 0, () => tagListType], 1
];
exports.CreateOpenIDConnectProviderRequest$ = CreateOpenIDConnectProviderRequest$;
const CreateOpenIDConnectProviderResponse$ = [3, n0, _COIDCPRr,
    0,
    [_OIDCPA, _T],
    [0, () => tagListType]
];
exports.CreateOpenIDConnectProviderResponse$ = CreateOpenIDConnectProviderResponse$;
const CreatePolicyRequest$ = [3, n0, _CPRr,
    0,
    [_PN, _PD, _Pa, _D, _T],
    [0, 0, 0, 0, () => tagListType], 2
];
exports.CreatePolicyRequest$ = CreatePolicyRequest$;
const CreatePolicyResponse$ = [3, n0, _CPRre,
    0,
    [_Po],
    [() => Policy$]
];
exports.CreatePolicyResponse$ = CreatePolicyResponse$;
const CreatePolicyVersionRequest$ = [3, n0, _CPVR,
    0,
    [_PA, _PD, _SAD],
    [0, 0, 2], 2
];
exports.CreatePolicyVersionRequest$ = CreatePolicyVersionRequest$;
const CreatePolicyVersionResponse$ = [3, n0, _CPVRr,
    0,
    [_PV],
    [() => PolicyVersion$]
];
exports.CreatePolicyVersionResponse$ = CreatePolicyVersionResponse$;
const CreateRoleRequest$ = [3, n0, _CRR,
    0,
    [_RN, _ARPD, _Pa, _D, _MSD, _PB, _T],
    [0, 0, 0, 0, 1, 0, () => tagListType], 2
];
exports.CreateRoleRequest$ = CreateRoleRequest$;
const CreateRoleResponse$ = [3, n0, _CRRr,
    0,
    [_Ro],
    [() => Role$], 1
];
exports.CreateRoleResponse$ = CreateRoleResponse$;
const CreateSAMLProviderRequest$ = [3, n0, _CSAMLPR,
    0,
    [_SAMLMD, _N, _T, _AEM, _APK],
    [0, 0, () => tagListType, 0, [() => privateKeyType, 0]], 2
];
exports.CreateSAMLProviderRequest$ = CreateSAMLProviderRequest$;
const CreateSAMLProviderResponse$ = [3, n0, _CSAMLPRr,
    0,
    [_SAMLPA, _T],
    [0, () => tagListType]
];
exports.CreateSAMLProviderResponse$ = CreateSAMLProviderResponse$;
const CreateServiceLinkedRoleRequest$ = [3, n0, _CSLRR,
    0,
    [_AWSSN, _D, _CS],
    [0, 0, 0], 1
];
exports.CreateServiceLinkedRoleRequest$ = CreateServiceLinkedRoleRequest$;
const CreateServiceLinkedRoleResponse$ = [3, n0, _CSLRRr,
    0,
    [_Ro],
    [() => Role$]
];
exports.CreateServiceLinkedRoleResponse$ = CreateServiceLinkedRoleResponse$;
const CreateServiceSpecificCredentialRequest$ = [3, n0, _CSSCR,
    0,
    [_UN, _SN, _CAD],
    [0, 0, 1], 2
];
exports.CreateServiceSpecificCredentialRequest$ = CreateServiceSpecificCredentialRequest$;
const CreateServiceSpecificCredentialResponse$ = [3, n0, _CSSCRr,
    0,
    [_SSC],
    [[() => ServiceSpecificCredential$, 0]]
];
exports.CreateServiceSpecificCredentialResponse$ = CreateServiceSpecificCredentialResponse$;
const CreateUserRequest$ = [3, n0, _CUR,
    0,
    [_UN, _Pa, _PB, _T],
    [0, 0, 0, () => tagListType], 1
];
exports.CreateUserRequest$ = CreateUserRequest$;
const CreateUserResponse$ = [3, n0, _CURr,
    0,
    [_Us],
    [() => User$]
];
exports.CreateUserResponse$ = CreateUserResponse$;
const CreateVirtualMFADeviceRequest$ = [3, n0, _CVMFADR,
    0,
    [_VMFADN, _Pa, _T],
    [0, 0, () => tagListType], 1
];
exports.CreateVirtualMFADeviceRequest$ = CreateVirtualMFADeviceRequest$;
const CreateVirtualMFADeviceResponse$ = [3, n0, _CVMFADRr,
    0,
    [_VMFAD],
    [[() => VirtualMFADevice$, 0]], 1
];
exports.CreateVirtualMFADeviceResponse$ = CreateVirtualMFADeviceResponse$;
const DeactivateMFADeviceRequest$ = [3, n0, _DMFADR,
    0,
    [_SNer, _UN],
    [0, 0], 1
];
exports.DeactivateMFADeviceRequest$ = DeactivateMFADeviceRequest$;
const DelegationPermission$ = [3, n0, _DP,
    0,
    [_PTA, _Par],
    [0, () => policyParameterListType]
];
exports.DelegationPermission$ = DelegationPermission$;
const DelegationRequest$ = [3, n0, _DR,
    0,
    [_DRI, _OAI, _D, _RM, _P, _PP, _RPRA, _OI, _AI, _St, _ET, _RI, _RNe, _CD, _SD, _RU, _No, _RR, _OSBO, _UT],
    [0, 0, 0, 0, () => DelegationPermission$, 0, 64 | 0, 0, 0, 0, 4, 0, 0, 4, 1, 0, 0, 0, 2, 4]
];
exports.DelegationRequest$ = DelegationRequest$;
const DeleteAccessKeyRequest$ = [3, n0, _DAKR,
    0,
    [_AKI, _UN],
    [0, 0], 1
];
exports.DeleteAccessKeyRequest$ = DeleteAccessKeyRequest$;
const DeleteAccountAliasRequest$ = [3, n0, _DAAR,
    0,
    [_AA],
    [0], 1
];
exports.DeleteAccountAliasRequest$ = DeleteAccountAliasRequest$;
const DeleteGroupPolicyRequest$ = [3, n0, _DGPR,
    0,
    [_GN, _PN],
    [0, 0], 2
];
exports.DeleteGroupPolicyRequest$ = DeleteGroupPolicyRequest$;
const DeleteGroupRequest$ = [3, n0, _DGR,
    0,
    [_GN],
    [0], 1
];
exports.DeleteGroupRequest$ = DeleteGroupRequest$;
const DeleteInstanceProfileRequest$ = [3, n0, _DIPR,
    0,
    [_IPN],
    [0], 1
];
exports.DeleteInstanceProfileRequest$ = DeleteInstanceProfileRequest$;
const DeleteLoginProfileRequest$ = [3, n0, _DLPR,
    0,
    [_UN],
    [0]
];
exports.DeleteLoginProfileRequest$ = DeleteLoginProfileRequest$;
const DeleteOpenIDConnectProviderRequest$ = [3, n0, _DOIDCPR,
    0,
    [_OIDCPA],
    [0], 1
];
exports.DeleteOpenIDConnectProviderRequest$ = DeleteOpenIDConnectProviderRequest$;
const DeletePolicyRequest$ = [3, n0, _DPR,
    0,
    [_PA],
    [0], 1
];
exports.DeletePolicyRequest$ = DeletePolicyRequest$;
const DeletePolicyVersionRequest$ = [3, n0, _DPVR,
    0,
    [_PA, _VI],
    [0, 0], 2
];
exports.DeletePolicyVersionRequest$ = DeletePolicyVersionRequest$;
const DeleteRolePermissionsBoundaryRequest$ = [3, n0, _DRPBR,
    0,
    [_RN],
    [0], 1
];
exports.DeleteRolePermissionsBoundaryRequest$ = DeleteRolePermissionsBoundaryRequest$;
const DeleteRolePolicyRequest$ = [3, n0, _DRPR,
    0,
    [_RN, _PN],
    [0, 0], 2
];
exports.DeleteRolePolicyRequest$ = DeleteRolePolicyRequest$;
const DeleteRoleRequest$ = [3, n0, _DRR,
    0,
    [_RN],
    [0], 1
];
exports.DeleteRoleRequest$ = DeleteRoleRequest$;
const DeleteSAMLProviderRequest$ = [3, n0, _DSAMLPR,
    0,
    [_SAMLPA],
    [0], 1
];
exports.DeleteSAMLProviderRequest$ = DeleteSAMLProviderRequest$;
const DeleteServerCertificateRequest$ = [3, n0, _DSCR,
    0,
    [_SCN],
    [0], 1
];
exports.DeleteServerCertificateRequest$ = DeleteServerCertificateRequest$;
const DeleteServiceLinkedRoleRequest$ = [3, n0, _DSLRR,
    0,
    [_RN],
    [0], 1
];
exports.DeleteServiceLinkedRoleRequest$ = DeleteServiceLinkedRoleRequest$;
const DeleteServiceLinkedRoleResponse$ = [3, n0, _DSLRRe,
    0,
    [_DTI],
    [0], 1
];
exports.DeleteServiceLinkedRoleResponse$ = DeleteServiceLinkedRoleResponse$;
const DeleteServiceSpecificCredentialRequest$ = [3, n0, _DSSCR,
    0,
    [_SSCI, _UN],
    [0, 0], 1
];
exports.DeleteServiceSpecificCredentialRequest$ = DeleteServiceSpecificCredentialRequest$;
const DeleteSigningCertificateRequest$ = [3, n0, _DSCRe,
    0,
    [_CI, _UN],
    [0, 0], 1
];
exports.DeleteSigningCertificateRequest$ = DeleteSigningCertificateRequest$;
const DeleteSSHPublicKeyRequest$ = [3, n0, _DSSHPKR,
    0,
    [_UN, _SSHPKI],
    [0, 0], 2
];
exports.DeleteSSHPublicKeyRequest$ = DeleteSSHPublicKeyRequest$;
const DeleteUserPermissionsBoundaryRequest$ = [3, n0, _DUPBR,
    0,
    [_UN],
    [0], 1
];
exports.DeleteUserPermissionsBoundaryRequest$ = DeleteUserPermissionsBoundaryRequest$;
const DeleteUserPolicyRequest$ = [3, n0, _DUPR,
    0,
    [_UN, _PN],
    [0, 0], 2
];
exports.DeleteUserPolicyRequest$ = DeleteUserPolicyRequest$;
const DeleteUserRequest$ = [3, n0, _DUR,
    0,
    [_UN],
    [0], 1
];
exports.DeleteUserRequest$ = DeleteUserRequest$;
const DeleteVirtualMFADeviceRequest$ = [3, n0, _DVMFADR,
    0,
    [_SNer],
    [0], 1
];
exports.DeleteVirtualMFADeviceRequest$ = DeleteVirtualMFADeviceRequest$;
const DeletionTaskFailureReasonType$ = [3, n0, _DTFRT,
    0,
    [_Re, _RUL],
    [0, () => RoleUsageListType]
];
exports.DeletionTaskFailureReasonType$ = DeletionTaskFailureReasonType$;
const DetachGroupPolicyRequest$ = [3, n0, _DGPRe,
    0,
    [_GN, _PA],
    [0, 0], 2
];
exports.DetachGroupPolicyRequest$ = DetachGroupPolicyRequest$;
const DetachRolePolicyRequest$ = [3, n0, _DRPRe,
    0,
    [_RN, _PA],
    [0, 0], 2
];
exports.DetachRolePolicyRequest$ = DetachRolePolicyRequest$;
const DetachUserPolicyRequest$ = [3, n0, _DUPRe,
    0,
    [_UN, _PA],
    [0, 0], 2
];
exports.DetachUserPolicyRequest$ = DetachUserPolicyRequest$;
const DisableOrganizationsRootCredentialsManagementRequest$ = [3, n0, _DORCMR,
    0,
    [],
    []
];
exports.DisableOrganizationsRootCredentialsManagementRequest$ = DisableOrganizationsRootCredentialsManagementRequest$;
const DisableOrganizationsRootCredentialsManagementResponse$ = [3, n0, _DORCMRi,
    0,
    [_OIr, _EF],
    [0, 64 | 0]
];
exports.DisableOrganizationsRootCredentialsManagementResponse$ = DisableOrganizationsRootCredentialsManagementResponse$;
const DisableOrganizationsRootSessionsRequest$ = [3, n0, _DORSR,
    0,
    [],
    []
];
exports.DisableOrganizationsRootSessionsRequest$ = DisableOrganizationsRootSessionsRequest$;
const DisableOrganizationsRootSessionsResponse$ = [3, n0, _DORSRi,
    0,
    [_OIr, _EF],
    [0, 64 | 0]
];
exports.DisableOrganizationsRootSessionsResponse$ = DisableOrganizationsRootSessionsResponse$;
const EnableMFADeviceRequest$ = [3, n0, _EMFADR,
    0,
    [_UN, _SNer, _AC, _ACu],
    [0, 0, 0, 0], 4
];
exports.EnableMFADeviceRequest$ = EnableMFADeviceRequest$;
const EnableOrganizationsRootCredentialsManagementRequest$ = [3, n0, _EORCMR,
    0,
    [],
    []
];
exports.EnableOrganizationsRootCredentialsManagementRequest$ = EnableOrganizationsRootCredentialsManagementRequest$;
const EnableOrganizationsRootCredentialsManagementResponse$ = [3, n0, _EORCMRn,
    0,
    [_OIr, _EF],
    [0, 64 | 0]
];
exports.EnableOrganizationsRootCredentialsManagementResponse$ = EnableOrganizationsRootCredentialsManagementResponse$;
const EnableOrganizationsRootSessionsRequest$ = [3, n0, _EORSR,
    0,
    [],
    []
];
exports.EnableOrganizationsRootSessionsRequest$ = EnableOrganizationsRootSessionsRequest$;
const EnableOrganizationsRootSessionsResponse$ = [3, n0, _EORSRn,
    0,
    [_OIr, _EF],
    [0, 64 | 0]
];
exports.EnableOrganizationsRootSessionsResponse$ = EnableOrganizationsRootSessionsResponse$;
const EnableOutboundWebIdentityFederationResponse$ = [3, n0, _EOWIFR,
    0,
    [_II],
    [0]
];
exports.EnableOutboundWebIdentityFederationResponse$ = EnableOutboundWebIdentityFederationResponse$;
const EntityDetails$ = [3, n0, _ED,
    0,
    [_EI, _LA],
    [() => EntityInfo$, 4], 1
];
exports.EntityDetails$ = EntityDetails$;
const EntityInfo$ = [3, n0, _EI,
    0,
    [_A, _N, _Ty, _I, _Pa],
    [0, 0, 0, 0, 0], 4
];
exports.EntityInfo$ = EntityInfo$;
const ErrorDetails$ = [3, n0, _EDr,
    0,
    [_M, _C],
    [0, 0], 2
];
exports.ErrorDetails$ = ErrorDetails$;
const EvaluationResult$ = [3, n0, _ER,
    0,
    [_EAN, _EDv, _ERN, _MS, _MCV, _ODD, _PBDD, _EDD, _RSR],
    [0, 0, 0, () => StatementListType, 64 | 0, () => OrganizationsDecisionDetail$, () => PermissionsBoundaryDecisionDetail$, 128 | 0, () => ResourceSpecificResultListType], 2
];
exports.EvaluationResult$ = EvaluationResult$;
const GenerateCredentialReportResponse$ = [3, n0, _GCRR,
    0,
    [_St, _D],
    [0, 0]
];
exports.GenerateCredentialReportResponse$ = GenerateCredentialReportResponse$;
const GenerateOrganizationsAccessReportRequest$ = [3, n0, _GOARR,
    0,
    [_EP, _OPI],
    [0, 0], 1
];
exports.GenerateOrganizationsAccessReportRequest$ = GenerateOrganizationsAccessReportRequest$;
const GenerateOrganizationsAccessReportResponse$ = [3, n0, _GOARRe,
    0,
    [_JI],
    [0]
];
exports.GenerateOrganizationsAccessReportResponse$ = GenerateOrganizationsAccessReportResponse$;
const GenerateServiceLastAccessedDetailsRequest$ = [3, n0, _GSLADR,
    0,
    [_A, _Gr],
    [0, 0], 1
];
exports.GenerateServiceLastAccessedDetailsRequest$ = GenerateServiceLastAccessedDetailsRequest$;
const GenerateServiceLastAccessedDetailsResponse$ = [3, n0, _GSLADRe,
    0,
    [_JI],
    [0]
];
exports.GenerateServiceLastAccessedDetailsResponse$ = GenerateServiceLastAccessedDetailsResponse$;
const GetAccessKeyLastUsedRequest$ = [3, n0, _GAKLUR,
    0,
    [_AKI],
    [0], 1
];
exports.GetAccessKeyLastUsedRequest$ = GetAccessKeyLastUsedRequest$;
const GetAccessKeyLastUsedResponse$ = [3, n0, _GAKLURe,
    0,
    [_UN, _AKLU],
    [0, () => AccessKeyLastUsed$]
];
exports.GetAccessKeyLastUsedResponse$ = GetAccessKeyLastUsedResponse$;
const GetAccountAuthorizationDetailsRequest$ = [3, n0, _GAADR,
    0,
    [_F, _MI, _Ma],
    [64 | 0, 1, 0]
];
exports.GetAccountAuthorizationDetailsRequest$ = GetAccountAuthorizationDetailsRequest$;
const GetAccountAuthorizationDetailsResponse$ = [3, n0, _GAADRe,
    0,
    [_UDL, _GDL, _RDL, _Pol, _IT, _Ma],
    [() => userDetailListType, () => groupDetailListType, () => roleDetailListType, () => ManagedPolicyDetailListType, 2, 0]
];
exports.GetAccountAuthorizationDetailsResponse$ = GetAccountAuthorizationDetailsResponse$;
const GetAccountPasswordPolicyResponse$ = [3, n0, _GAPPR,
    0,
    [_PPa],
    [() => PasswordPolicy$], 1
];
exports.GetAccountPasswordPolicyResponse$ = GetAccountPasswordPolicyResponse$;
const GetAccountSummaryResponse$ = [3, n0, _GASR,
    0,
    [_SM],
    [128 | 1]
];
exports.GetAccountSummaryResponse$ = GetAccountSummaryResponse$;
const GetContextKeysForCustomPolicyRequest$ = [3, n0, _GCKFCPR,
    0,
    [_PIL],
    [64 | 0], 1
];
exports.GetContextKeysForCustomPolicyRequest$ = GetContextKeysForCustomPolicyRequest$;
const GetContextKeysForPolicyResponse$ = [3, n0, _GCKFPR,
    0,
    [_CKNo],
    [64 | 0]
];
exports.GetContextKeysForPolicyResponse$ = GetContextKeysForPolicyResponse$;
const GetContextKeysForPrincipalPolicyRequest$ = [3, n0, _GCKFPPR,
    0,
    [_PSA, _PIL],
    [0, 64 | 0], 1
];
exports.GetContextKeysForPrincipalPolicyRequest$ = GetContextKeysForPrincipalPolicyRequest$;
const GetCredentialReportResponse$ = [3, n0, _GCRRe,
    0,
    [_Co, _RF, _GT],
    [21, 0, 4]
];
exports.GetCredentialReportResponse$ = GetCredentialReportResponse$;
const GetDelegationRequestRequest$ = [3, n0, _GDRR,
    0,
    [_DRI, _DPC],
    [0, 2], 1
];
exports.GetDelegationRequestRequest$ = GetDelegationRequestRequest$;
const GetDelegationRequestResponse$ = [3, n0, _GDRRe,
    0,
    [_DR, _PCS, _PCR],
    [() => DelegationRequest$, 0, 0]
];
exports.GetDelegationRequestResponse$ = GetDelegationRequestResponse$;
const GetGroupPolicyRequest$ = [3, n0, _GGPR,
    0,
    [_GN, _PN],
    [0, 0], 2
];
exports.GetGroupPolicyRequest$ = GetGroupPolicyRequest$;
const GetGroupPolicyResponse$ = [3, n0, _GGPRe,
    0,
    [_GN, _PN, _PD],
    [0, 0, 0], 3
];
exports.GetGroupPolicyResponse$ = GetGroupPolicyResponse$;
const GetGroupRequest$ = [3, n0, _GGR,
    0,
    [_GN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.GetGroupRequest$ = GetGroupRequest$;
const GetGroupResponse$ = [3, n0, _GGRe,
    0,
    [_G, _Use, _IT, _Ma],
    [() => Group$, () => userListType, 2, 0], 2
];
exports.GetGroupResponse$ = GetGroupResponse$;
const GetHumanReadableSummaryRequest$ = [3, n0, _GHRSR,
    0,
    [_EA, _L],
    [0, 0], 1
];
exports.GetHumanReadableSummaryRequest$ = GetHumanReadableSummaryRequest$;
const GetHumanReadableSummaryResponse$ = [3, n0, _GHRSRe,
    0,
    [_SC, _L, _SS],
    [0, 0, 0]
];
exports.GetHumanReadableSummaryResponse$ = GetHumanReadableSummaryResponse$;
const GetInstanceProfileRequest$ = [3, n0, _GIPR,
    0,
    [_IPN],
    [0], 1
];
exports.GetInstanceProfileRequest$ = GetInstanceProfileRequest$;
const GetInstanceProfileResponse$ = [3, n0, _GIPRe,
    0,
    [_IP],
    [() => InstanceProfile$], 1
];
exports.GetInstanceProfileResponse$ = GetInstanceProfileResponse$;
const GetLoginProfileRequest$ = [3, n0, _GLPR,
    0,
    [_UN],
    [0]
];
exports.GetLoginProfileRequest$ = GetLoginProfileRequest$;
const GetLoginProfileResponse$ = [3, n0, _GLPRe,
    0,
    [_LP],
    [() => LoginProfile$], 1
];
exports.GetLoginProfileResponse$ = GetLoginProfileResponse$;
const GetMFADeviceRequest$ = [3, n0, _GMFADR,
    0,
    [_SNer, _UN],
    [0, 0], 1
];
exports.GetMFADeviceRequest$ = GetMFADeviceRequest$;
const GetMFADeviceResponse$ = [3, n0, _GMFADRe,
    0,
    [_SNer, _UN, _EDn, _Ce],
    [0, 0, 4, 128 | 0], 1
];
exports.GetMFADeviceResponse$ = GetMFADeviceResponse$;
const GetOpenIDConnectProviderRequest$ = [3, n0, _GOIDCPR,
    0,
    [_OIDCPA],
    [0], 1
];
exports.GetOpenIDConnectProviderRequest$ = GetOpenIDConnectProviderRequest$;
const GetOpenIDConnectProviderResponse$ = [3, n0, _GOIDCPRe,
    0,
    [_U, _CIDL, _TL, _CD, _T],
    [0, 64 | 0, 64 | 0, 4, () => tagListType]
];
exports.GetOpenIDConnectProviderResponse$ = GetOpenIDConnectProviderResponse$;
const GetOrganizationsAccessReportRequest$ = [3, n0, _GOARRet,
    0,
    [_JI, _MI, _Ma, _SK],
    [0, 1, 0, 0], 1
];
exports.GetOrganizationsAccessReportRequest$ = GetOrganizationsAccessReportRequest$;
const GetOrganizationsAccessReportResponse$ = [3, n0, _GOARRetr,
    0,
    [_JS, _JCD, _JCDo, _NOSA, _NOSNA, _ADc, _IT, _Ma, _EDr],
    [0, 4, 4, 1, 1, () => AccessDetails, 2, 0, () => ErrorDetails$], 2
];
exports.GetOrganizationsAccessReportResponse$ = GetOrganizationsAccessReportResponse$;
const GetOutboundWebIdentityFederationInfoResponse$ = [3, n0, _GOWIFIR,
    0,
    [_II, _JVE],
    [0, 2]
];
exports.GetOutboundWebIdentityFederationInfoResponse$ = GetOutboundWebIdentityFederationInfoResponse$;
const GetPolicyRequest$ = [3, n0, _GPR,
    0,
    [_PA],
    [0], 1
];
exports.GetPolicyRequest$ = GetPolicyRequest$;
const GetPolicyResponse$ = [3, n0, _GPRe,
    0,
    [_Po],
    [() => Policy$]
];
exports.GetPolicyResponse$ = GetPolicyResponse$;
const GetPolicyVersionRequest$ = [3, n0, _GPVR,
    0,
    [_PA, _VI],
    [0, 0], 2
];
exports.GetPolicyVersionRequest$ = GetPolicyVersionRequest$;
const GetPolicyVersionResponse$ = [3, n0, _GPVRe,
    0,
    [_PV],
    [() => PolicyVersion$]
];
exports.GetPolicyVersionResponse$ = GetPolicyVersionResponse$;
const GetRolePolicyRequest$ = [3, n0, _GRPR,
    0,
    [_RN, _PN],
    [0, 0], 2
];
exports.GetRolePolicyRequest$ = GetRolePolicyRequest$;
const GetRolePolicyResponse$ = [3, n0, _GRPRe,
    0,
    [_RN, _PN, _PD],
    [0, 0, 0], 3
];
exports.GetRolePolicyResponse$ = GetRolePolicyResponse$;
const GetRoleRequest$ = [3, n0, _GRR,
    0,
    [_RN],
    [0], 1
];
exports.GetRoleRequest$ = GetRoleRequest$;
const GetRoleResponse$ = [3, n0, _GRRe,
    0,
    [_Ro],
    [() => Role$], 1
];
exports.GetRoleResponse$ = GetRoleResponse$;
const GetSAMLProviderRequest$ = [3, n0, _GSAMLPR,
    0,
    [_SAMLPA],
    [0], 1
];
exports.GetSAMLProviderRequest$ = GetSAMLProviderRequest$;
const GetSAMLProviderResponse$ = [3, n0, _GSAMLPRe,
    0,
    [_SAMLPUUID, _SAMLMD, _CD, _VU, _T, _AEM, _PKL],
    [0, 0, 4, 4, () => tagListType, 0, () => privateKeyList]
];
exports.GetSAMLProviderResponse$ = GetSAMLProviderResponse$;
const GetServerCertificateRequest$ = [3, n0, _GSCR,
    0,
    [_SCN],
    [0], 1
];
exports.GetServerCertificateRequest$ = GetServerCertificateRequest$;
const GetServerCertificateResponse$ = [3, n0, _GSCRe,
    0,
    [_SCe],
    [() => ServerCertificate$], 1
];
exports.GetServerCertificateResponse$ = GetServerCertificateResponse$;
const GetServiceLastAccessedDetailsRequest$ = [3, n0, _GSLADRet,
    0,
    [_JI, _MI, _Ma],
    [0, 1, 0], 1
];
exports.GetServiceLastAccessedDetailsRequest$ = GetServiceLastAccessedDetailsRequest$;
const GetServiceLastAccessedDetailsResponse$ = [3, n0, _GSLADRete,
    0,
    [_JS, _JCD, _SLA, _JCDo, _JT, _IT, _Ma, _E],
    [0, 4, () => ServicesLastAccessed, 4, 0, 2, 0, () => ErrorDetails$], 4
];
exports.GetServiceLastAccessedDetailsResponse$ = GetServiceLastAccessedDetailsResponse$;
const GetServiceLastAccessedDetailsWithEntitiesRequest$ = [3, n0, _GSLADWER,
    0,
    [_JI, _SNe, _MI, _Ma],
    [0, 0, 1, 0], 2
];
exports.GetServiceLastAccessedDetailsWithEntitiesRequest$ = GetServiceLastAccessedDetailsWithEntitiesRequest$;
const GetServiceLastAccessedDetailsWithEntitiesResponse$ = [3, n0, _GSLADWERe,
    0,
    [_JS, _JCD, _JCDo, _EDL, _IT, _Ma, _E],
    [0, 4, 4, () => entityDetailsListType, 2, 0, () => ErrorDetails$], 4
];
exports.GetServiceLastAccessedDetailsWithEntitiesResponse$ = GetServiceLastAccessedDetailsWithEntitiesResponse$;
const GetServiceLinkedRoleDeletionStatusRequest$ = [3, n0, _GSLRDSR,
    0,
    [_DTI],
    [0], 1
];
exports.GetServiceLinkedRoleDeletionStatusRequest$ = GetServiceLinkedRoleDeletionStatusRequest$;
const GetServiceLinkedRoleDeletionStatusResponse$ = [3, n0, _GSLRDSRe,
    0,
    [_S, _Re],
    [0, () => DeletionTaskFailureReasonType$], 1
];
exports.GetServiceLinkedRoleDeletionStatusResponse$ = GetServiceLinkedRoleDeletionStatusResponse$;
const GetSSHPublicKeyRequest$ = [3, n0, _GSSHPKR,
    0,
    [_UN, _SSHPKI, _En],
    [0, 0, 0], 3
];
exports.GetSSHPublicKeyRequest$ = GetSSHPublicKeyRequest$;
const GetSSHPublicKeyResponse$ = [3, n0, _GSSHPKRe,
    0,
    [_SSHPK],
    [() => SSHPublicKey$]
];
exports.GetSSHPublicKeyResponse$ = GetSSHPublicKeyResponse$;
const GetUserPolicyRequest$ = [3, n0, _GUPR,
    0,
    [_UN, _PN],
    [0, 0], 2
];
exports.GetUserPolicyRequest$ = GetUserPolicyRequest$;
const GetUserPolicyResponse$ = [3, n0, _GUPRe,
    0,
    [_UN, _PN, _PD],
    [0, 0, 0], 3
];
exports.GetUserPolicyResponse$ = GetUserPolicyResponse$;
const GetUserRequest$ = [3, n0, _GUR,
    0,
    [_UN],
    [0]
];
exports.GetUserRequest$ = GetUserRequest$;
const GetUserResponse$ = [3, n0, _GURe,
    0,
    [_Us],
    [() => User$], 1
];
exports.GetUserResponse$ = GetUserResponse$;
const Group$ = [3, n0, _G,
    0,
    [_Pa, _GN, _GI, _A, _CD],
    [0, 0, 0, 0, 4], 5
];
exports.Group$ = Group$;
const GroupDetail$ = [3, n0, _GD,
    0,
    [_Pa, _GN, _GI, _A, _CD, _GPL, _AMP],
    [0, 0, 0, 0, 4, () => policyDetailListType, () => attachedPoliciesListType]
];
exports.GroupDetail$ = GroupDetail$;
const InstanceProfile$ = [3, n0, _IP,
    0,
    [_Pa, _IPN, _IPI, _A, _CD, _Rol, _T],
    [0, 0, 0, 0, 4, () => roleListType, () => tagListType], 6
];
exports.InstanceProfile$ = InstanceProfile$;
const ListAccessKeysRequest$ = [3, n0, _LAKR,
    0,
    [_UN, _Ma, _MI],
    [0, 0, 1]
];
exports.ListAccessKeysRequest$ = ListAccessKeysRequest$;
const ListAccessKeysResponse$ = [3, n0, _LAKRi,
    0,
    [_AKM, _IT, _Ma],
    [() => accessKeyMetadataListType, 2, 0], 1
];
exports.ListAccessKeysResponse$ = ListAccessKeysResponse$;
const ListAccountAliasesRequest$ = [3, n0, _LAAR,
    0,
    [_Ma, _MI],
    [0, 1]
];
exports.ListAccountAliasesRequest$ = ListAccountAliasesRequest$;
const ListAccountAliasesResponse$ = [3, n0, _LAARi,
    0,
    [_AAc, _IT, _Ma],
    [64 | 0, 2, 0], 1
];
exports.ListAccountAliasesResponse$ = ListAccountAliasesResponse$;
const ListAttachedGroupPoliciesRequest$ = [3, n0, _LAGPR,
    0,
    [_GN, _PPat, _Ma, _MI],
    [0, 0, 0, 1], 1
];
exports.ListAttachedGroupPoliciesRequest$ = ListAttachedGroupPoliciesRequest$;
const ListAttachedGroupPoliciesResponse$ = [3, n0, _LAGPRi,
    0,
    [_APt, _IT, _Ma],
    [() => attachedPoliciesListType, 2, 0]
];
exports.ListAttachedGroupPoliciesResponse$ = ListAttachedGroupPoliciesResponse$;
const ListAttachedRolePoliciesRequest$ = [3, n0, _LARPR,
    0,
    [_RN, _PPat, _Ma, _MI],
    [0, 0, 0, 1], 1
];
exports.ListAttachedRolePoliciesRequest$ = ListAttachedRolePoliciesRequest$;
const ListAttachedRolePoliciesResponse$ = [3, n0, _LARPRi,
    0,
    [_APt, _IT, _Ma],
    [() => attachedPoliciesListType, 2, 0]
];
exports.ListAttachedRolePoliciesResponse$ = ListAttachedRolePoliciesResponse$;
const ListAttachedUserPoliciesRequest$ = [3, n0, _LAUPR,
    0,
    [_UN, _PPat, _Ma, _MI],
    [0, 0, 0, 1], 1
];
exports.ListAttachedUserPoliciesRequest$ = ListAttachedUserPoliciesRequest$;
const ListAttachedUserPoliciesResponse$ = [3, n0, _LAUPRi,
    0,
    [_APt, _IT, _Ma],
    [() => attachedPoliciesListType, 2, 0]
];
exports.ListAttachedUserPoliciesResponse$ = ListAttachedUserPoliciesResponse$;
const ListDelegationRequestsRequest$ = [3, n0, _LDRR,
    0,
    [_OI, _Ma, _MI],
    [0, 0, 1]
];
exports.ListDelegationRequestsRequest$ = ListDelegationRequestsRequest$;
const ListDelegationRequestsResponse$ = [3, n0, _LDRRi,
    0,
    [_DRe, _Ma, _iT],
    [() => delegationRequestsListType, 0, 2]
];
exports.ListDelegationRequestsResponse$ = ListDelegationRequestsResponse$;
const ListEntitiesForPolicyRequest$ = [3, n0, _LEFPR,
    0,
    [_PA, _EFn, _PPat, _PUF, _Ma, _MI],
    [0, 0, 0, 0, 0, 1], 1
];
exports.ListEntitiesForPolicyRequest$ = ListEntitiesForPolicyRequest$;
const ListEntitiesForPolicyResponse$ = [3, n0, _LEFPRi,
    0,
    [_PG, _PU, _PR, _IT, _Ma],
    [() => PolicyGroupListType, () => PolicyUserListType, () => PolicyRoleListType, 2, 0]
];
exports.ListEntitiesForPolicyResponse$ = ListEntitiesForPolicyResponse$;
const ListGroupPoliciesRequest$ = [3, n0, _LGPR,
    0,
    [_GN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListGroupPoliciesRequest$ = ListGroupPoliciesRequest$;
const ListGroupPoliciesResponse$ = [3, n0, _LGPRi,
    0,
    [_PNo, _IT, _Ma],
    [64 | 0, 2, 0], 1
];
exports.ListGroupPoliciesResponse$ = ListGroupPoliciesResponse$;
const ListGroupsForUserRequest$ = [3, n0, _LGFUR,
    0,
    [_UN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListGroupsForUserRequest$ = ListGroupsForUserRequest$;
const ListGroupsForUserResponse$ = [3, n0, _LGFURi,
    0,
    [_Gro, _IT, _Ma],
    [() => groupListType, 2, 0], 1
];
exports.ListGroupsForUserResponse$ = ListGroupsForUserResponse$;
const ListGroupsRequest$ = [3, n0, _LGR,
    0,
    [_PPat, _Ma, _MI],
    [0, 0, 1]
];
exports.ListGroupsRequest$ = ListGroupsRequest$;
const ListGroupsResponse$ = [3, n0, _LGRi,
    0,
    [_Gro, _IT, _Ma],
    [() => groupListType, 2, 0], 1
];
exports.ListGroupsResponse$ = ListGroupsResponse$;
const ListInstanceProfilesForRoleRequest$ = [3, n0, _LIPFRR,
    0,
    [_RN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListInstanceProfilesForRoleRequest$ = ListInstanceProfilesForRoleRequest$;
const ListInstanceProfilesForRoleResponse$ = [3, n0, _LIPFRRi,
    0,
    [_IPn, _IT, _Ma],
    [() => instanceProfileListType, 2, 0], 1
];
exports.ListInstanceProfilesForRoleResponse$ = ListInstanceProfilesForRoleResponse$;
const ListInstanceProfilesRequest$ = [3, n0, _LIPR,
    0,
    [_PPat, _Ma, _MI],
    [0, 0, 1]
];
exports.ListInstanceProfilesRequest$ = ListInstanceProfilesRequest$;
const ListInstanceProfilesResponse$ = [3, n0, _LIPRi,
    0,
    [_IPn, _IT, _Ma],
    [() => instanceProfileListType, 2, 0], 1
];
exports.ListInstanceProfilesResponse$ = ListInstanceProfilesResponse$;
const ListInstanceProfileTagsRequest$ = [3, n0, _LIPTR,
    0,
    [_IPN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListInstanceProfileTagsRequest$ = ListInstanceProfileTagsRequest$;
const ListInstanceProfileTagsResponse$ = [3, n0, _LIPTRi,
    0,
    [_T, _IT, _Ma],
    [() => tagListType, 2, 0], 1
];
exports.ListInstanceProfileTagsResponse$ = ListInstanceProfileTagsResponse$;
const ListMFADevicesRequest$ = [3, n0, _LMFADR,
    0,
    [_UN, _Ma, _MI],
    [0, 0, 1]
];
exports.ListMFADevicesRequest$ = ListMFADevicesRequest$;
const ListMFADevicesResponse$ = [3, n0, _LMFADRi,
    0,
    [_MFAD, _IT, _Ma],
    [() => mfaDeviceListType, 2, 0], 1
];
exports.ListMFADevicesResponse$ = ListMFADevicesResponse$;
const ListMFADeviceTagsRequest$ = [3, n0, _LMFADTR,
    0,
    [_SNer, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListMFADeviceTagsRequest$ = ListMFADeviceTagsRequest$;
const ListMFADeviceTagsResponse$ = [3, n0, _LMFADTRi,
    0,
    [_T, _IT, _Ma],
    [() => tagListType, 2, 0], 1
];
exports.ListMFADeviceTagsResponse$ = ListMFADeviceTagsResponse$;
const ListOpenIDConnectProvidersRequest$ = [3, n0, _LOIDCPR,
    0,
    [],
    []
];
exports.ListOpenIDConnectProvidersRequest$ = ListOpenIDConnectProvidersRequest$;
const ListOpenIDConnectProvidersResponse$ = [3, n0, _LOIDCPRi,
    0,
    [_OIDCPL],
    [() => OpenIDConnectProviderListType]
];
exports.ListOpenIDConnectProvidersResponse$ = ListOpenIDConnectProvidersResponse$;
const ListOpenIDConnectProviderTagsRequest$ = [3, n0, _LOIDCPTR,
    0,
    [_OIDCPA, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListOpenIDConnectProviderTagsRequest$ = ListOpenIDConnectProviderTagsRequest$;
const ListOpenIDConnectProviderTagsResponse$ = [3, n0, _LOIDCPTRi,
    0,
    [_T, _IT, _Ma],
    [() => tagListType, 2, 0], 1
];
exports.ListOpenIDConnectProviderTagsResponse$ = ListOpenIDConnectProviderTagsResponse$;
const ListOrganizationsFeaturesRequest$ = [3, n0, _LOFR,
    0,
    [],
    []
];
exports.ListOrganizationsFeaturesRequest$ = ListOrganizationsFeaturesRequest$;
const ListOrganizationsFeaturesResponse$ = [3, n0, _LOFRi,
    0,
    [_OIr, _EF],
    [0, 64 | 0]
];
exports.ListOrganizationsFeaturesResponse$ = ListOrganizationsFeaturesResponse$;
const ListPoliciesGrantingServiceAccessEntry$ = [3, n0, _LPGSAE,
    0,
    [_SNe, _Pol],
    [0, () => policyGrantingServiceAccessListType]
];
exports.ListPoliciesGrantingServiceAccessEntry$ = ListPoliciesGrantingServiceAccessEntry$;
const ListPoliciesGrantingServiceAccessRequest$ = [3, n0, _LPGSAR,
    0,
    [_A, _SNerv, _Ma],
    [0, 64 | 0, 0], 2
];
exports.ListPoliciesGrantingServiceAccessRequest$ = ListPoliciesGrantingServiceAccessRequest$;
const ListPoliciesGrantingServiceAccessResponse$ = [3, n0, _LPGSARi,
    0,
    [_PGSA, _IT, _Ma],
    [() => listPolicyGrantingServiceAccessResponseListType, 2, 0], 1
];
exports.ListPoliciesGrantingServiceAccessResponse$ = ListPoliciesGrantingServiceAccessResponse$;
const ListPoliciesRequest$ = [3, n0, _LPR,
    0,
    [_Sc, _OA, _PPat, _PUF, _Ma, _MI],
    [0, 2, 0, 0, 0, 1]
];
exports.ListPoliciesRequest$ = ListPoliciesRequest$;
const ListPoliciesResponse$ = [3, n0, _LPRi,
    0,
    [_Pol, _IT, _Ma],
    [() => policyListType, 2, 0]
];
exports.ListPoliciesResponse$ = ListPoliciesResponse$;
const ListPolicyTagsRequest$ = [3, n0, _LPTR,
    0,
    [_PA, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListPolicyTagsRequest$ = ListPolicyTagsRequest$;
const ListPolicyTagsResponse$ = [3, n0, _LPTRi,
    0,
    [_T, _IT, _Ma],
    [() => tagListType, 2, 0], 1
];
exports.ListPolicyTagsResponse$ = ListPolicyTagsResponse$;
const ListPolicyVersionsRequest$ = [3, n0, _LPVR,
    0,
    [_PA, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListPolicyVersionsRequest$ = ListPolicyVersionsRequest$;
const ListPolicyVersionsResponse$ = [3, n0, _LPVRi,
    0,
    [_V, _IT, _Ma],
    [() => policyDocumentVersionListType, 2, 0]
];
exports.ListPolicyVersionsResponse$ = ListPolicyVersionsResponse$;
const ListRolePoliciesRequest$ = [3, n0, _LRPR,
    0,
    [_RN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListRolePoliciesRequest$ = ListRolePoliciesRequest$;
const ListRolePoliciesResponse$ = [3, n0, _LRPRi,
    0,
    [_PNo, _IT, _Ma],
    [64 | 0, 2, 0], 1
];
exports.ListRolePoliciesResponse$ = ListRolePoliciesResponse$;
const ListRolesRequest$ = [3, n0, _LRR,
    0,
    [_PPat, _Ma, _MI],
    [0, 0, 1]
];
exports.ListRolesRequest$ = ListRolesRequest$;
const ListRolesResponse$ = [3, n0, _LRRi,
    0,
    [_Rol, _IT, _Ma],
    [() => roleListType, 2, 0], 1
];
exports.ListRolesResponse$ = ListRolesResponse$;
const ListRoleTagsRequest$ = [3, n0, _LRTR,
    0,
    [_RN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListRoleTagsRequest$ = ListRoleTagsRequest$;
const ListRoleTagsResponse$ = [3, n0, _LRTRi,
    0,
    [_T, _IT, _Ma],
    [() => tagListType, 2, 0], 1
];
exports.ListRoleTagsResponse$ = ListRoleTagsResponse$;
const ListSAMLProvidersRequest$ = [3, n0, _LSAMLPR,
    0,
    [],
    []
];
exports.ListSAMLProvidersRequest$ = ListSAMLProvidersRequest$;
const ListSAMLProvidersResponse$ = [3, n0, _LSAMLPRi,
    0,
    [_SAMLPL],
    [() => SAMLProviderListType]
];
exports.ListSAMLProvidersResponse$ = ListSAMLProvidersResponse$;
const ListSAMLProviderTagsRequest$ = [3, n0, _LSAMLPTR,
    0,
    [_SAMLPA, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListSAMLProviderTagsRequest$ = ListSAMLProviderTagsRequest$;
const ListSAMLProviderTagsResponse$ = [3, n0, _LSAMLPTRi,
    0,
    [_T, _IT, _Ma],
    [() => tagListType, 2, 0], 1
];
exports.ListSAMLProviderTagsResponse$ = ListSAMLProviderTagsResponse$;
const ListServerCertificatesRequest$ = [3, n0, _LSCR,
    0,
    [_PPat, _Ma, _MI],
    [0, 0, 1]
];
exports.ListServerCertificatesRequest$ = ListServerCertificatesRequest$;
const ListServerCertificatesResponse$ = [3, n0, _LSCRi,
    0,
    [_SCML, _IT, _Ma],
    [() => serverCertificateMetadataListType, 2, 0], 1
];
exports.ListServerCertificatesResponse$ = ListServerCertificatesResponse$;
const ListServerCertificateTagsRequest$ = [3, n0, _LSCTR,
    0,
    [_SCN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListServerCertificateTagsRequest$ = ListServerCertificateTagsRequest$;
const ListServerCertificateTagsResponse$ = [3, n0, _LSCTRi,
    0,
    [_T, _IT, _Ma],
    [() => tagListType, 2, 0], 1
];
exports.ListServerCertificateTagsResponse$ = ListServerCertificateTagsResponse$;
const ListServiceSpecificCredentialsRequest$ = [3, n0, _LSSCR,
    0,
    [_UN, _SN, _AU, _Ma, _MI],
    [0, 0, 2, 0, 1]
];
exports.ListServiceSpecificCredentialsRequest$ = ListServiceSpecificCredentialsRequest$;
const ListServiceSpecificCredentialsResponse$ = [3, n0, _LSSCRi,
    0,
    [_SSCe, _Ma, _IT],
    [() => ServiceSpecificCredentialsListType, 0, 2]
];
exports.ListServiceSpecificCredentialsResponse$ = ListServiceSpecificCredentialsResponse$;
const ListSigningCertificatesRequest$ = [3, n0, _LSCRis,
    0,
    [_UN, _Ma, _MI],
    [0, 0, 1]
];
exports.ListSigningCertificatesRequest$ = ListSigningCertificatesRequest$;
const ListSigningCertificatesResponse$ = [3, n0, _LSCRist,
    0,
    [_Cer, _IT, _Ma],
    [() => certificateListType, 2, 0], 1
];
exports.ListSigningCertificatesResponse$ = ListSigningCertificatesResponse$;
const ListSSHPublicKeysRequest$ = [3, n0, _LSSHPKR,
    0,
    [_UN, _Ma, _MI],
    [0, 0, 1]
];
exports.ListSSHPublicKeysRequest$ = ListSSHPublicKeysRequest$;
const ListSSHPublicKeysResponse$ = [3, n0, _LSSHPKRi,
    0,
    [_SSHPKu, _IT, _Ma],
    [() => SSHPublicKeyListType, 2, 0]
];
exports.ListSSHPublicKeysResponse$ = ListSSHPublicKeysResponse$;
const ListUserPoliciesRequest$ = [3, n0, _LUPR,
    0,
    [_UN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListUserPoliciesRequest$ = ListUserPoliciesRequest$;
const ListUserPoliciesResponse$ = [3, n0, _LUPRi,
    0,
    [_PNo, _IT, _Ma],
    [64 | 0, 2, 0], 1
];
exports.ListUserPoliciesResponse$ = ListUserPoliciesResponse$;
const ListUsersRequest$ = [3, n0, _LUR,
    0,
    [_PPat, _Ma, _MI],
    [0, 0, 1]
];
exports.ListUsersRequest$ = ListUsersRequest$;
const ListUsersResponse$ = [3, n0, _LURi,
    0,
    [_Use, _IT, _Ma],
    [() => userListType, 2, 0], 1
];
exports.ListUsersResponse$ = ListUsersResponse$;
const ListUserTagsRequest$ = [3, n0, _LUTR,
    0,
    [_UN, _Ma, _MI],
    [0, 0, 1], 1
];
exports.ListUserTagsRequest$ = ListUserTagsRequest$;
const ListUserTagsResponse$ = [3, n0, _LUTRi,
    0,
    [_T, _IT, _Ma],
    [() => tagListType, 2, 0], 1
];
exports.ListUserTagsResponse$ = ListUserTagsResponse$;
const ListVirtualMFADevicesRequest$ = [3, n0, _LVMFADR,
    0,
    [_AS, _Ma, _MI],
    [0, 0, 1]
];
exports.ListVirtualMFADevicesRequest$ = ListVirtualMFADevicesRequest$;
const ListVirtualMFADevicesResponse$ = [3, n0, _LVMFADRi,
    0,
    [_VMFADi, _IT, _Ma],
    [[() => virtualMFADeviceListType, 0], 2, 0], 1
];
exports.ListVirtualMFADevicesResponse$ = ListVirtualMFADevicesResponse$;
const LoginProfile$ = [3, n0, _LP,
    0,
    [_UN, _CD, _PRR],
    [0, 4, 2], 2
];
exports.LoginProfile$ = LoginProfile$;
const ManagedPolicyDetail$ = [3, n0, _MPD,
    0,
    [_PN, _PI, _A, _Pa, _DVI, _ACt, _PBUC, _IA, _D, _CD, _UD, _PVL],
    [0, 0, 0, 0, 0, 1, 1, 2, 0, 4, 4, () => policyDocumentVersionListType]
];
exports.ManagedPolicyDetail$ = ManagedPolicyDetail$;
const MFADevice$ = [3, n0, _MFADe,
    0,
    [_UN, _SNer, _EDn],
    [0, 0, 4], 3
];
exports.MFADevice$ = MFADevice$;
const OpenIDConnectProviderListEntry$ = [3, n0, _OIDCPLE,
    0,
    [_A],
    [0]
];
exports.OpenIDConnectProviderListEntry$ = OpenIDConnectProviderListEntry$;
const OrganizationsDecisionDetail$ = [3, n0, _ODD,
    0,
    [_ABO],
    [2]
];
exports.OrganizationsDecisionDetail$ = OrganizationsDecisionDetail$;
const PasswordPolicy$ = [3, n0, _PPa,
    0,
    [_MPL, _RS, _RNeq, _RUC, _RLC, _AUTCP, _EPx, _MPA, _PRP, _HE],
    [1, 2, 2, 2, 2, 2, 2, 1, 1, 2]
];
exports.PasswordPolicy$ = PasswordPolicy$;
const PermissionsBoundaryDecisionDetail$ = [3, n0, _PBDD,
    0,
    [_ABPB],
    [2]
];
exports.PermissionsBoundaryDecisionDetail$ = PermissionsBoundaryDecisionDetail$;
const Policy$ = [3, n0, _Po,
    0,
    [_PN, _PI, _A, _Pa, _DVI, _ACt, _PBUC, _IA, _D, _CD, _UD, _T],
    [0, 0, 0, 0, 0, 1, 1, 2, 0, 4, 4, () => tagListType]
];
exports.Policy$ = Policy$;
const PolicyDetail$ = [3, n0, _PDo,
    0,
    [_PN, _PD],
    [0, 0]
];
exports.PolicyDetail$ = PolicyDetail$;
const PolicyGrantingServiceAccess$ = [3, n0, _PGSAo,
    0,
    [_PN, _PT, _PA, _ETn, _EN],
    [0, 0, 0, 0, 0], 2
];
exports.PolicyGrantingServiceAccess$ = PolicyGrantingServiceAccess$;
const PolicyGroup$ = [3, n0, _PGo,
    0,
    [_GN, _GI],
    [0, 0]
];
exports.PolicyGroup$ = PolicyGroup$;
const PolicyParameter$ = [3, n0, _PPo,
    0,
    [_N, _Va, _Ty],
    [0, 64 | 0, 0]
];
exports.PolicyParameter$ = PolicyParameter$;
const PolicyRole$ = [3, n0, _PRo,
    0,
    [_RN, _RIo],
    [0, 0]
];
exports.PolicyRole$ = PolicyRole$;
const PolicyUser$ = [3, n0, _PUo,
    0,
    [_UN, _UI],
    [0, 0]
];
exports.PolicyUser$ = PolicyUser$;
const PolicyVersion$ = [3, n0, _PV,
    0,
    [_Do, _VI, _IDV, _CD],
    [0, 0, 2, 4]
];
exports.PolicyVersion$ = PolicyVersion$;
const Position$ = [3, n0, _Pos,
    0,
    [_Li, _Col],
    [1, 1]
];
exports.Position$ = Position$;
const PutGroupPolicyRequest$ = [3, n0, _PGPR,
    0,
    [_GN, _PN, _PD],
    [0, 0, 0], 3
];
exports.PutGroupPolicyRequest$ = PutGroupPolicyRequest$;
const PutRolePermissionsBoundaryRequest$ = [3, n0, _PRPBR,
    0,
    [_RN, _PB],
    [0, 0], 2
];
exports.PutRolePermissionsBoundaryRequest$ = PutRolePermissionsBoundaryRequest$;
const PutRolePolicyRequest$ = [3, n0, _PRPR,
    0,
    [_RN, _PN, _PD],
    [0, 0, 0], 3
];
exports.PutRolePolicyRequest$ = PutRolePolicyRequest$;
const PutUserPermissionsBoundaryRequest$ = [3, n0, _PUPBR,
    0,
    [_UN, _PB],
    [0, 0], 2
];
exports.PutUserPermissionsBoundaryRequest$ = PutUserPermissionsBoundaryRequest$;
const PutUserPolicyRequest$ = [3, n0, _PUPR,
    0,
    [_UN, _PN, _PD],
    [0, 0, 0], 3
];
exports.PutUserPolicyRequest$ = PutUserPolicyRequest$;
const RejectDelegationRequestRequest$ = [3, n0, _RDRR,
    0,
    [_DRI, _No],
    [0, 0], 1
];
exports.RejectDelegationRequestRequest$ = RejectDelegationRequestRequest$;
const RemoveClientIDFromOpenIDConnectProviderRequest$ = [3, n0, _RCIDFOIDCPR,
    0,
    [_OIDCPA, _CID],
    [0, 0], 2
];
exports.RemoveClientIDFromOpenIDConnectProviderRequest$ = RemoveClientIDFromOpenIDConnectProviderRequest$;
const RemoveRoleFromInstanceProfileRequest$ = [3, n0, _RRFIPR,
    0,
    [_IPN, _RN],
    [0, 0], 2
];
exports.RemoveRoleFromInstanceProfileRequest$ = RemoveRoleFromInstanceProfileRequest$;
const RemoveUserFromGroupRequest$ = [3, n0, _RUFGR,
    0,
    [_GN, _UN],
    [0, 0], 2
];
exports.RemoveUserFromGroupRequest$ = RemoveUserFromGroupRequest$;
const ResetServiceSpecificCredentialRequest$ = [3, n0, _RSSCR,
    0,
    [_SSCI, _UN],
    [0, 0], 1
];
exports.ResetServiceSpecificCredentialRequest$ = ResetServiceSpecificCredentialRequest$;
const ResetServiceSpecificCredentialResponse$ = [3, n0, _RSSCRe,
    0,
    [_SSC],
    [[() => ServiceSpecificCredential$, 0]]
];
exports.ResetServiceSpecificCredentialResponse$ = ResetServiceSpecificCredentialResponse$;
const ResourceSpecificResult$ = [3, n0, _RSRe,
    0,
    [_ERN, _ERD, _MS, _MCV, _EDD, _PBDD],
    [0, 0, () => StatementListType, 64 | 0, 128 | 0, () => PermissionsBoundaryDecisionDetail$], 2
];
exports.ResourceSpecificResult$ = ResourceSpecificResult$;
const ResyncMFADeviceRequest$ = [3, n0, _RMFADR,
    0,
    [_UN, _SNer, _AC, _ACu],
    [0, 0, 0, 0], 4
];
exports.ResyncMFADeviceRequest$ = ResyncMFADeviceRequest$;
const Role$ = [3, n0, _Ro,
    0,
    [_Pa, _RN, _RIo, _A, _CD, _ARPD, _D, _MSD, _PB, _T, _RLU],
    [0, 0, 0, 0, 4, 0, 0, 1, () => AttachedPermissionsBoundary$, () => tagListType, () => RoleLastUsed$], 5
];
exports.Role$ = Role$;
const RoleDetail$ = [3, n0, _RD,
    0,
    [_Pa, _RN, _RIo, _A, _CD, _ARPD, _IPL, _RPL, _AMP, _PB, _T, _RLU],
    [0, 0, 0, 0, 4, 0, () => instanceProfileListType, () => policyDetailListType, () => attachedPoliciesListType, () => AttachedPermissionsBoundary$, () => tagListType, () => RoleLastUsed$]
];
exports.RoleDetail$ = RoleDetail$;
const RoleLastUsed$ = [3, n0, _RLU,
    0,
    [_LUD, _R],
    [4, 0]
];
exports.RoleLastUsed$ = RoleLastUsed$;
const RoleUsageType$ = [3, n0, _RUT,
    0,
    [_R, _Res],
    [0, 64 | 0]
];
exports.RoleUsageType$ = RoleUsageType$;
const SAMLPrivateKey$ = [3, n0, _SAMLPK,
    0,
    [_KI, _Ti],
    [0, 4]
];
exports.SAMLPrivateKey$ = SAMLPrivateKey$;
const SAMLProviderListEntry$ = [3, n0, _SAMLPLE,
    0,
    [_A, _VU, _CD],
    [0, 4, 4]
];
exports.SAMLProviderListEntry$ = SAMLProviderListEntry$;
const SendDelegationTokenRequest$ = [3, n0, _SDTR,
    0,
    [_DRI],
    [0], 1
];
exports.SendDelegationTokenRequest$ = SendDelegationTokenRequest$;
const ServerCertificate$ = [3, n0, _SCe,
    0,
    [_SCM, _CB, _CC, _T],
    [() => ServerCertificateMetadata$, 0, 0, () => tagListType], 2
];
exports.ServerCertificate$ = ServerCertificate$;
const ServerCertificateMetadata$ = [3, n0, _SCM,
    0,
    [_Pa, _SCN, _SCI, _A, _UDp, _Ex],
    [0, 0, 0, 0, 4, 4], 4
];
exports.ServerCertificateMetadata$ = ServerCertificateMetadata$;
const ServiceLastAccessed$ = [3, n0, _SLAe,
    0,
    [_SN, _SNe, _LA, _LAE, _LAR, _TAE, _TALA],
    [0, 0, 4, 0, 0, 1, () => TrackedActionsLastAccessed], 2
];
exports.ServiceLastAccessed$ = ServiceLastAccessed$;
const ServiceSpecificCredential$ = [3, n0, _SSC,
    0,
    [_CD, _SN, _SSCI, _UN, _S, _EDx, _SUN, _SP, _SCA, _SCS],
    [4, 0, 0, 0, 0, 4, 0, [() => servicePassword, 0], 0, [() => serviceCredentialSecret, 0]], 5
];
exports.ServiceSpecificCredential$ = ServiceSpecificCredential$;
const ServiceSpecificCredentialMetadata$ = [3, n0, _SSCM,
    0,
    [_UN, _S, _CD, _SSCI, _SN, _SUN, _SCA, _EDx],
    [0, 0, 4, 0, 0, 0, 0, 4], 5
];
exports.ServiceSpecificCredentialMetadata$ = ServiceSpecificCredentialMetadata$;
const SetDefaultPolicyVersionRequest$ = [3, n0, _SDPVR,
    0,
    [_PA, _VI],
    [0, 0], 2
];
exports.SetDefaultPolicyVersionRequest$ = SetDefaultPolicyVersionRequest$;
const SetSecurityTokenServicePreferencesRequest$ = [3, n0, _SSTSPR,
    0,
    [_GETV],
    [0], 1
];
exports.SetSecurityTokenServicePreferencesRequest$ = SetSecurityTokenServicePreferencesRequest$;
const SigningCertificate$ = [3, n0, _SCi,
    0,
    [_UN, _CI, _CB, _S, _UDp],
    [0, 0, 0, 0, 4], 4
];
exports.SigningCertificate$ = SigningCertificate$;
const SimulateCustomPolicyRequest$ = [3, n0, _SCPR,
    0,
    [_PIL, _AN, _PBPIL, _RA, _RP, _RO, _CA, _CEo, _RHO, _MI, _Ma],
    [64 | 0, 64 | 0, 64 | 0, 64 | 0, 0, 0, 0, () => ContextEntryListType, 0, 1, 0], 2
];
exports.SimulateCustomPolicyRequest$ = SimulateCustomPolicyRequest$;
const SimulatePolicyResponse$ = [3, n0, _SPR,
    0,
    [_ERv, _IT, _Ma],
    [() => EvaluationResultsListType, 2, 0]
];
exports.SimulatePolicyResponse$ = SimulatePolicyResponse$;
const SimulatePrincipalPolicyRequest$ = [3, n0, _SPPR,
    0,
    [_PSA, _AN, _PIL, _PBPIL, _RA, _RP, _RO, _CA, _CEo, _RHO, _MI, _Ma],
    [0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 0, 0, 0, () => ContextEntryListType, 0, 1, 0], 2
];
exports.SimulatePrincipalPolicyRequest$ = SimulatePrincipalPolicyRequest$;
const SSHPublicKey$ = [3, n0, _SSHPK,
    0,
    [_UN, _SSHPKI, _Fi, _SSHPKB, _S, _UDp],
    [0, 0, 0, 0, 0, 4], 5
];
exports.SSHPublicKey$ = SSHPublicKey$;
const SSHPublicKeyMetadata$ = [3, n0, _SSHPKM,
    0,
    [_UN, _SSHPKI, _S, _UDp],
    [0, 0, 0, 4], 4
];
exports.SSHPublicKeyMetadata$ = SSHPublicKeyMetadata$;
const Statement$ = [3, n0, _Sta,
    0,
    [_SPI, _SPT, _SPt, _EPn],
    [0, 0, () => Position$, () => Position$]
];
exports.Statement$ = Statement$;
const Tag$ = [3, n0, _Ta,
    0,
    [_K, _Val],
    [0, 0], 2
];
exports.Tag$ = Tag$;
const TagInstanceProfileRequest$ = [3, n0, _TIPR,
    0,
    [_IPN, _T],
    [0, () => tagListType], 2
];
exports.TagInstanceProfileRequest$ = TagInstanceProfileRequest$;
const TagMFADeviceRequest$ = [3, n0, _TMFADR,
    0,
    [_SNer, _T],
    [0, () => tagListType], 2
];
exports.TagMFADeviceRequest$ = TagMFADeviceRequest$;
const TagOpenIDConnectProviderRequest$ = [3, n0, _TOIDCPR,
    0,
    [_OIDCPA, _T],
    [0, () => tagListType], 2
];
exports.TagOpenIDConnectProviderRequest$ = TagOpenIDConnectProviderRequest$;
const TagPolicyRequest$ = [3, n0, _TPR,
    0,
    [_PA, _T],
    [0, () => tagListType], 2
];
exports.TagPolicyRequest$ = TagPolicyRequest$;
const TagRoleRequest$ = [3, n0, _TRR,
    0,
    [_RN, _T],
    [0, () => tagListType], 2
];
exports.TagRoleRequest$ = TagRoleRequest$;
const TagSAMLProviderRequest$ = [3, n0, _TSAMLPR,
    0,
    [_SAMLPA, _T],
    [0, () => tagListType], 2
];
exports.TagSAMLProviderRequest$ = TagSAMLProviderRequest$;
const TagServerCertificateRequest$ = [3, n0, _TSCR,
    0,
    [_SCN, _T],
    [0, () => tagListType], 2
];
exports.TagServerCertificateRequest$ = TagServerCertificateRequest$;
const TagUserRequest$ = [3, n0, _TUR,
    0,
    [_UN, _T],
    [0, () => tagListType], 2
];
exports.TagUserRequest$ = TagUserRequest$;
const TrackedActionLastAccessed$ = [3, n0, _TALAr,
    0,
    [_ANc, _LAEa, _LATa, _LARa],
    [0, 0, 4, 0]
];
exports.TrackedActionLastAccessed$ = TrackedActionLastAccessed$;
const UntagInstanceProfileRequest$ = [3, n0, _UIPR,
    0,
    [_IPN, _TK],
    [0, 64 | 0], 2
];
exports.UntagInstanceProfileRequest$ = UntagInstanceProfileRequest$;
const UntagMFADeviceRequest$ = [3, n0, _UMFADR,
    0,
    [_SNer, _TK],
    [0, 64 | 0], 2
];
exports.UntagMFADeviceRequest$ = UntagMFADeviceRequest$;
const UntagOpenIDConnectProviderRequest$ = [3, n0, _UOIDCPR,
    0,
    [_OIDCPA, _TK],
    [0, 64 | 0], 2
];
exports.UntagOpenIDConnectProviderRequest$ = UntagOpenIDConnectProviderRequest$;
const UntagPolicyRequest$ = [3, n0, _UPR,
    0,
    [_PA, _TK],
    [0, 64 | 0], 2
];
exports.UntagPolicyRequest$ = UntagPolicyRequest$;
const UntagRoleRequest$ = [3, n0, _URR,
    0,
    [_RN, _TK],
    [0, 64 | 0], 2
];
exports.UntagRoleRequest$ = UntagRoleRequest$;
const UntagSAMLProviderRequest$ = [3, n0, _USAMLPR,
    0,
    [_SAMLPA, _TK],
    [0, 64 | 0], 2
];
exports.UntagSAMLProviderRequest$ = UntagSAMLProviderRequest$;
const UntagServerCertificateRequest$ = [3, n0, _USCR,
    0,
    [_SCN, _TK],
    [0, 64 | 0], 2
];
exports.UntagServerCertificateRequest$ = UntagServerCertificateRequest$;
const UntagUserRequest$ = [3, n0, _UUR,
    0,
    [_UN, _TK],
    [0, 64 | 0], 2
];
exports.UntagUserRequest$ = UntagUserRequest$;
const UpdateAccessKeyRequest$ = [3, n0, _UAKR,
    0,
    [_AKI, _S, _UN],
    [0, 0, 0], 2
];
exports.UpdateAccessKeyRequest$ = UpdateAccessKeyRequest$;
const UpdateAccountPasswordPolicyRequest$ = [3, n0, _UAPPR,
    0,
    [_MPL, _RS, _RNeq, _RUC, _RLC, _AUTCP, _MPA, _PRP, _HE],
    [1, 2, 2, 2, 2, 2, 1, 1, 2]
];
exports.UpdateAccountPasswordPolicyRequest$ = UpdateAccountPasswordPolicyRequest$;
const UpdateAssumeRolePolicyRequest$ = [3, n0, _UARPR,
    0,
    [_RN, _PD],
    [0, 0], 2
];
exports.UpdateAssumeRolePolicyRequest$ = UpdateAssumeRolePolicyRequest$;
const UpdateDelegationRequestRequest$ = [3, n0, _UDRR,
    0,
    [_DRI, _No],
    [0, 0], 1
];
exports.UpdateDelegationRequestRequest$ = UpdateDelegationRequestRequest$;
const UpdateGroupRequest$ = [3, n0, _UGR,
    0,
    [_GN, _NPe, _NGN],
    [0, 0, 0], 1
];
exports.UpdateGroupRequest$ = UpdateGroupRequest$;
const UpdateLoginProfileRequest$ = [3, n0, _ULPR,
    0,
    [_UN, _Pas, _PRR],
    [0, [() => passwordType, 0], 2], 1
];
exports.UpdateLoginProfileRequest$ = UpdateLoginProfileRequest$;
const UpdateOpenIDConnectProviderThumbprintRequest$ = [3, n0, _UOIDCPTR,
    0,
    [_OIDCPA, _TL],
    [0, 64 | 0], 2
];
exports.UpdateOpenIDConnectProviderThumbprintRequest$ = UpdateOpenIDConnectProviderThumbprintRequest$;
const UpdateRoleDescriptionRequest$ = [3, n0, _URDR,
    0,
    [_RN, _D],
    [0, 0], 2
];
exports.UpdateRoleDescriptionRequest$ = UpdateRoleDescriptionRequest$;
const UpdateRoleDescriptionResponse$ = [3, n0, _URDRp,
    0,
    [_Ro],
    [() => Role$]
];
exports.UpdateRoleDescriptionResponse$ = UpdateRoleDescriptionResponse$;
const UpdateRoleRequest$ = [3, n0, _URRp,
    0,
    [_RN, _D, _MSD],
    [0, 0, 1], 1
];
exports.UpdateRoleRequest$ = UpdateRoleRequest$;
const UpdateRoleResponse$ = [3, n0, _URRpd,
    0,
    [],
    []
];
exports.UpdateRoleResponse$ = UpdateRoleResponse$;
const UpdateSAMLProviderRequest$ = [3, n0, _USAMLPRp,
    0,
    [_SAMLPA, _SAMLMD, _AEM, _APK, _RPK],
    [0, 0, 0, [() => privateKeyType, 0], 0], 1
];
exports.UpdateSAMLProviderRequest$ = UpdateSAMLProviderRequest$;
const UpdateSAMLProviderResponse$ = [3, n0, _USAMLPRpd,
    0,
    [_SAMLPA],
    [0]
];
exports.UpdateSAMLProviderResponse$ = UpdateSAMLProviderResponse$;
const UpdateServerCertificateRequest$ = [3, n0, _USCRp,
    0,
    [_SCN, _NPe, _NSCN],
    [0, 0, 0], 1
];
exports.UpdateServerCertificateRequest$ = UpdateServerCertificateRequest$;
const UpdateServiceSpecificCredentialRequest$ = [3, n0, _USSCR,
    0,
    [_SSCI, _S, _UN],
    [0, 0, 0], 2
];
exports.UpdateServiceSpecificCredentialRequest$ = UpdateServiceSpecificCredentialRequest$;
const UpdateSigningCertificateRequest$ = [3, n0, _USCRpd,
    0,
    [_CI, _S, _UN],
    [0, 0, 0], 2
];
exports.UpdateSigningCertificateRequest$ = UpdateSigningCertificateRequest$;
const UpdateSSHPublicKeyRequest$ = [3, n0, _USSHPKR,
    0,
    [_UN, _SSHPKI, _S],
    [0, 0, 0], 3
];
exports.UpdateSSHPublicKeyRequest$ = UpdateSSHPublicKeyRequest$;
const UpdateUserRequest$ = [3, n0, _UURp,
    0,
    [_UN, _NPe, _NUN],
    [0, 0, 0], 1
];
exports.UpdateUserRequest$ = UpdateUserRequest$;
const UploadServerCertificateRequest$ = [3, n0, _USCRpl,
    0,
    [_SCN, _CB, _PK, _Pa, _CC, _T],
    [0, 0, [() => privateKeyType, 0], 0, 0, () => tagListType], 3
];
exports.UploadServerCertificateRequest$ = UploadServerCertificateRequest$;
const UploadServerCertificateResponse$ = [3, n0, _USCRplo,
    0,
    [_SCM, _T],
    [() => ServerCertificateMetadata$, () => tagListType]
];
exports.UploadServerCertificateResponse$ = UploadServerCertificateResponse$;
const UploadSigningCertificateRequest$ = [3, n0, _USCRploa,
    0,
    [_CB, _UN],
    [0, 0], 1
];
exports.UploadSigningCertificateRequest$ = UploadSigningCertificateRequest$;
const UploadSigningCertificateResponse$ = [3, n0, _USCRpload,
    0,
    [_Cert],
    [() => SigningCertificate$], 1
];
exports.UploadSigningCertificateResponse$ = UploadSigningCertificateResponse$;
const UploadSSHPublicKeyRequest$ = [3, n0, _USSHPKRp,
    0,
    [_UN, _SSHPKB],
    [0, 0], 2
];
exports.UploadSSHPublicKeyRequest$ = UploadSSHPublicKeyRequest$;
const UploadSSHPublicKeyResponse$ = [3, n0, _USSHPKRpl,
    0,
    [_SSHPK],
    [() => SSHPublicKey$]
];
exports.UploadSSHPublicKeyResponse$ = UploadSSHPublicKeyResponse$;
const User$ = [3, n0, _Us,
    0,
    [_Pa, _UN, _UI, _A, _CD, _PLU, _PB, _T],
    [0, 0, 0, 0, 4, 4, () => AttachedPermissionsBoundary$, () => tagListType], 5
];
exports.User$ = User$;
const UserDetail$ = [3, n0, _UDs,
    0,
    [_Pa, _UN, _UI, _A, _CD, _UPL, _GL, _AMP, _PB, _T],
    [0, 0, 0, 0, 4, () => policyDetailListType, 64 | 0, () => attachedPoliciesListType, () => AttachedPermissionsBoundary$, () => tagListType]
];
exports.UserDetail$ = UserDetail$;
const VirtualMFADevice$ = [3, n0, _VMFAD,
    0,
    [_SNer, _BSS, _QRCPNG, _Us, _EDn, _T],
    [0, [() => BootstrapDatum, 0], [() => BootstrapDatum, 0], () => User$, 4, () => tagListType], 1
];
exports.VirtualMFADevice$ = VirtualMFADevice$;
var __Unit = "unit";
var AccessDetails = [1, n0, _ADc,
    0, () => AccessDetail$
];
var accessKeyMetadataListType = [1, n0, _aKMLT,
    0, () => AccessKeyMetadata$
];
var accountAliasListType = 64 | 0;
var ActionNameListType = 64 | 0;
var ArnListType = 64 | 0;
var attachedPoliciesListType = [1, n0, _aPLT,
    0, () => AttachedPolicy$
];
var certificateListType = [1, n0, _cLT,
    0, () => SigningCertificate$
];
var clientIDListType = 64 | 0;
var ContextEntryListType = [1, n0, _CELT,
    0, () => ContextEntry$
];
var ContextKeyNamesResultListType = 64 | 0;
var ContextKeyValueListType = 64 | 0;
var delegationRequestsListType = [1, n0, _dRLT,
    0, () => DelegationRequest$
];
var entityDetailsListType = [1, n0, _eDLT,
    0, () => EntityDetails$
];
var entityListType = 64 | 0;
var EvaluationResultsListType = [1, n0, _ERLT,
    0, () => EvaluationResult$
];
var FeaturesListType = 64 | 0;
var groupDetailListType = [1, n0, _gDLT,
    0, () => GroupDetail$
];
var groupListType = [1, n0, _gLT,
    0, () => Group$
];
var groupNameListType = 64 | 0;
var instanceProfileListType = [1, n0, _iPLT,
    0, () => InstanceProfile$
];
var listPolicyGrantingServiceAccessResponseListType = [1, n0, _lPGSARLT,
    0, () => ListPoliciesGrantingServiceAccessEntry$
];
var ManagedPolicyDetailListType = [1, n0, _MPDLT,
    0, () => ManagedPolicyDetail$
];
var mfaDeviceListType = [1, n0, _mDLT,
    0, () => MFADevice$
];
var OpenIDConnectProviderListType = [1, n0, _OIDCPLT,
    0, () => OpenIDConnectProviderListEntry$
];
var policyDetailListType = [1, n0, _pDLT,
    0, () => PolicyDetail$
];
var policyDocumentVersionListType = [1, n0, _pDVLT,
    0, () => PolicyVersion$
];
var policyGrantingServiceAccessListType = [1, n0, _pGSALT,
    0, () => PolicyGrantingServiceAccess$
];
var PolicyGroupListType = [1, n0, _PGLT,
    0, () => PolicyGroup$
];
var policyListType = [1, n0, _pLT,
    0, () => Policy$
];
var policyNameListType = 64 | 0;
var policyParameterListType = [1, n0, _pPLT,
    0, () => PolicyParameter$
];
var policyParameterValuesListType = 64 | 0;
var PolicyRoleListType = [1, n0, _PRLT,
    0, () => PolicyRole$
];
var PolicyUserListType = [1, n0, _PULT,
    0, () => PolicyUser$
];
var privateKeyList = [1, n0, _pKL,
    0, () => SAMLPrivateKey$
];
var ResourceNameListType = 64 | 0;
var ResourceSpecificResultListType = [1, n0, _RSRLT,
    0, () => ResourceSpecificResult$
];
var roleDetailListType = [1, n0, _rDLT,
    0, () => RoleDetail$
];
var roleListType = [1, n0, _rLT,
    0, () => Role$
];
var rolePermissionRestrictionArnListType = 64 | 0;
var RoleUsageListType = [1, n0, _RULT,
    0, () => RoleUsageType$
];
var SAMLProviderListType = [1, n0, _SAMLPLT,
    0, () => SAMLProviderListEntry$
];
var serverCertificateMetadataListType = [1, n0, _sCMLT,
    0, () => ServerCertificateMetadata$
];
var serviceNamespaceListType = 64 | 0;
var ServicesLastAccessed = [1, n0, _SLA,
    0, () => ServiceLastAccessed$
];
var ServiceSpecificCredentialsListType = [1, n0, _SSCLT,
    0, () => ServiceSpecificCredentialMetadata$
];
var SimulationPolicyListType = 64 | 0;
var SSHPublicKeyListType = [1, n0, _SSHPKLT,
    0, () => SSHPublicKeyMetadata$
];
var StatementListType = [1, n0, _SLT,
    0, () => Statement$
];
var tagKeyListType = 64 | 0;
var tagListType = [1, n0, _tLT,
    0, () => Tag$
];
var thumbprintListType = 64 | 0;
var TrackedActionsLastAccessed = [1, n0, _TALA,
    0, () => TrackedActionLastAccessed$
];
var userDetailListType = [1, n0, _uDLT,
    0, () => UserDetail$
];
var userListType = [1, n0, _uLT,
    0, () => User$
];
var virtualMFADeviceListType = [1, n0, _vMFADLT,
    0, [() => VirtualMFADevice$,
        0]
];
var CertificationMapType = 128 | 0;
var EvalDecisionDetailsType = 128 | 0;
var summaryMapType = 128 | 1;
exports.AcceptDelegationRequest$ = [9, n0, _ADR,
    0, () => AcceptDelegationRequestRequest$, () => __Unit
];
exports.AddClientIDToOpenIDConnectProvider$ = [9, n0, _ACIDTOIDCP,
    0, () => AddClientIDToOpenIDConnectProviderRequest$, () => __Unit
];
exports.AddRoleToInstanceProfile$ = [9, n0, _ARTIP,
    0, () => AddRoleToInstanceProfileRequest$, () => __Unit
];
exports.AddUserToGroup$ = [9, n0, _AUTG,
    0, () => AddUserToGroupRequest$, () => __Unit
];
exports.AssociateDelegationRequest$ = [9, n0, _ADRs,
    0, () => AssociateDelegationRequestRequest$, () => __Unit
];
exports.AttachGroupPolicy$ = [9, n0, _AGP,
    0, () => AttachGroupPolicyRequest$, () => __Unit
];
exports.AttachRolePolicy$ = [9, n0, _ARP,
    0, () => AttachRolePolicyRequest$, () => __Unit
];
exports.AttachUserPolicy$ = [9, n0, _AUP,
    0, () => AttachUserPolicyRequest$, () => __Unit
];
exports.ChangePassword$ = [9, n0, _CP,
    0, () => ChangePasswordRequest$, () => __Unit
];
exports.CreateAccessKey$ = [9, n0, _CAK,
    0, () => CreateAccessKeyRequest$, () => CreateAccessKeyResponse$
];
exports.CreateAccountAlias$ = [9, n0, _CAA,
    0, () => CreateAccountAliasRequest$, () => __Unit
];
exports.CreateDelegationRequest$ = [9, n0, _CDR,
    0, () => CreateDelegationRequestRequest$, () => CreateDelegationRequestResponse$
];
exports.CreateGroup$ = [9, n0, _CG,
    0, () => CreateGroupRequest$, () => CreateGroupResponse$
];
exports.CreateInstanceProfile$ = [9, n0, _CIP,
    0, () => CreateInstanceProfileRequest$, () => CreateInstanceProfileResponse$
];
exports.CreateLoginProfile$ = [9, n0, _CLP,
    0, () => CreateLoginProfileRequest$, () => CreateLoginProfileResponse$
];
exports.CreateOpenIDConnectProvider$ = [9, n0, _COIDCP,
    0, () => CreateOpenIDConnectProviderRequest$, () => CreateOpenIDConnectProviderResponse$
];
exports.CreatePolicy$ = [9, n0, _CPr,
    0, () => CreatePolicyRequest$, () => CreatePolicyResponse$
];
exports.CreatePolicyVersion$ = [9, n0, _CPV,
    0, () => CreatePolicyVersionRequest$, () => CreatePolicyVersionResponse$
];
exports.CreateRole$ = [9, n0, _CR,
    0, () => CreateRoleRequest$, () => CreateRoleResponse$
];
exports.CreateSAMLProvider$ = [9, n0, _CSAMLP,
    0, () => CreateSAMLProviderRequest$, () => CreateSAMLProviderResponse$
];
exports.CreateServiceLinkedRole$ = [9, n0, _CSLR,
    0, () => CreateServiceLinkedRoleRequest$, () => CreateServiceLinkedRoleResponse$
];
exports.CreateServiceSpecificCredential$ = [9, n0, _CSSC,
    0, () => CreateServiceSpecificCredentialRequest$, () => CreateServiceSpecificCredentialResponse$
];
exports.CreateUser$ = [9, n0, _CU,
    0, () => CreateUserRequest$, () => CreateUserResponse$
];
exports.CreateVirtualMFADevice$ = [9, n0, _CVMFAD,
    0, () => CreateVirtualMFADeviceRequest$, () => CreateVirtualMFADeviceResponse$
];
exports.DeactivateMFADevice$ = [9, n0, _DMFAD,
    0, () => DeactivateMFADeviceRequest$, () => __Unit
];
exports.DeleteAccessKey$ = [9, n0, _DAK,
    0, () => DeleteAccessKeyRequest$, () => __Unit
];
exports.DeleteAccountAlias$ = [9, n0, _DAA,
    0, () => DeleteAccountAliasRequest$, () => __Unit
];
exports.DeleteAccountPasswordPolicy$ = [9, n0, _DAPP,
    0, () => __Unit, () => __Unit
];
exports.DeleteGroup$ = [9, n0, _DG,
    0, () => DeleteGroupRequest$, () => __Unit
];
exports.DeleteGroupPolicy$ = [9, n0, _DGP,
    0, () => DeleteGroupPolicyRequest$, () => __Unit
];
exports.DeleteInstanceProfile$ = [9, n0, _DIP,
    0, () => DeleteInstanceProfileRequest$, () => __Unit
];
exports.DeleteLoginProfile$ = [9, n0, _DLP,
    0, () => DeleteLoginProfileRequest$, () => __Unit
];
exports.DeleteOpenIDConnectProvider$ = [9, n0, _DOIDCP,
    0, () => DeleteOpenIDConnectProviderRequest$, () => __Unit
];
exports.DeletePolicy$ = [9, n0, _DPe,
    0, () => DeletePolicyRequest$, () => __Unit
];
exports.DeletePolicyVersion$ = [9, n0, _DPV,
    0, () => DeletePolicyVersionRequest$, () => __Unit
];
exports.DeleteRole$ = [9, n0, _DRel,
    0, () => DeleteRoleRequest$, () => __Unit
];
exports.DeleteRolePermissionsBoundary$ = [9, n0, _DRPB,
    0, () => DeleteRolePermissionsBoundaryRequest$, () => __Unit
];
exports.DeleteRolePolicy$ = [9, n0, _DRP,
    0, () => DeleteRolePolicyRequest$, () => __Unit
];
exports.DeleteSAMLProvider$ = [9, n0, _DSAMLP,
    0, () => DeleteSAMLProviderRequest$, () => __Unit
];
exports.DeleteServerCertificate$ = [9, n0, _DSC,
    0, () => DeleteServerCertificateRequest$, () => __Unit
];
exports.DeleteServiceLinkedRole$ = [9, n0, _DSLR,
    0, () => DeleteServiceLinkedRoleRequest$, () => DeleteServiceLinkedRoleResponse$
];
exports.DeleteServiceSpecificCredential$ = [9, n0, _DSSC,
    0, () => DeleteServiceSpecificCredentialRequest$, () => __Unit
];
exports.DeleteSigningCertificate$ = [9, n0, _DSCe,
    0, () => DeleteSigningCertificateRequest$, () => __Unit
];
exports.DeleteSSHPublicKey$ = [9, n0, _DSSHPK,
    0, () => DeleteSSHPublicKeyRequest$, () => __Unit
];
exports.DeleteUser$ = [9, n0, _DU,
    0, () => DeleteUserRequest$, () => __Unit
];
exports.DeleteUserPermissionsBoundary$ = [9, n0, _DUPB,
    0, () => DeleteUserPermissionsBoundaryRequest$, () => __Unit
];
exports.DeleteUserPolicy$ = [9, n0, _DUP,
    0, () => DeleteUserPolicyRequest$, () => __Unit
];
exports.DeleteVirtualMFADevice$ = [9, n0, _DVMFAD,
    0, () => DeleteVirtualMFADeviceRequest$, () => __Unit
];
exports.DetachGroupPolicy$ = [9, n0, _DGPe,
    0, () => DetachGroupPolicyRequest$, () => __Unit
];
exports.DetachRolePolicy$ = [9, n0, _DRPe,
    0, () => DetachRolePolicyRequest$, () => __Unit
];
exports.DetachUserPolicy$ = [9, n0, _DUPe,
    0, () => DetachUserPolicyRequest$, () => __Unit
];
exports.DisableOrganizationsRootCredentialsManagement$ = [9, n0, _DORCM,
    0, () => DisableOrganizationsRootCredentialsManagementRequest$, () => DisableOrganizationsRootCredentialsManagementResponse$
];
exports.DisableOrganizationsRootSessions$ = [9, n0, _DORS,
    0, () => DisableOrganizationsRootSessionsRequest$, () => DisableOrganizationsRootSessionsResponse$
];
exports.DisableOutboundWebIdentityFederation$ = [9, n0, _DOWIF,
    0, () => __Unit, () => __Unit
];
exports.EnableMFADevice$ = [9, n0, _EMFAD,
    0, () => EnableMFADeviceRequest$, () => __Unit
];
exports.EnableOrganizationsRootCredentialsManagement$ = [9, n0, _EORCM,
    0, () => EnableOrganizationsRootCredentialsManagementRequest$, () => EnableOrganizationsRootCredentialsManagementResponse$
];
exports.EnableOrganizationsRootSessions$ = [9, n0, _EORS,
    0, () => EnableOrganizationsRootSessionsRequest$, () => EnableOrganizationsRootSessionsResponse$
];
exports.EnableOutboundWebIdentityFederation$ = [9, n0, _EOWIF,
    0, () => __Unit, () => EnableOutboundWebIdentityFederationResponse$
];
exports.GenerateCredentialReport$ = [9, n0, _GCR,
    0, () => __Unit, () => GenerateCredentialReportResponse$
];
exports.GenerateOrganizationsAccessReport$ = [9, n0, _GOAR,
    0, () => GenerateOrganizationsAccessReportRequest$, () => GenerateOrganizationsAccessReportResponse$
];
exports.GenerateServiceLastAccessedDetails$ = [9, n0, _GSLAD,
    0, () => GenerateServiceLastAccessedDetailsRequest$, () => GenerateServiceLastAccessedDetailsResponse$
];
exports.GetAccessKeyLastUsed$ = [9, n0, _GAKLU,
    0, () => GetAccessKeyLastUsedRequest$, () => GetAccessKeyLastUsedResponse$
];
exports.GetAccountAuthorizationDetails$ = [9, n0, _GAAD,
    0, () => GetAccountAuthorizationDetailsRequest$, () => GetAccountAuthorizationDetailsResponse$
];
exports.GetAccountPasswordPolicy$ = [9, n0, _GAPP,
    0, () => __Unit, () => GetAccountPasswordPolicyResponse$
];
exports.GetAccountSummary$ = [9, n0, _GAS,
    0, () => __Unit, () => GetAccountSummaryResponse$
];
exports.GetContextKeysForCustomPolicy$ = [9, n0, _GCKFCP,
    0, () => GetContextKeysForCustomPolicyRequest$, () => GetContextKeysForPolicyResponse$
];
exports.GetContextKeysForPrincipalPolicy$ = [9, n0, _GCKFPP,
    0, () => GetContextKeysForPrincipalPolicyRequest$, () => GetContextKeysForPolicyResponse$
];
exports.GetCredentialReport$ = [9, n0, _GCRe,
    0, () => __Unit, () => GetCredentialReportResponse$
];
exports.GetDelegationRequest$ = [9, n0, _GDR,
    0, () => GetDelegationRequestRequest$, () => GetDelegationRequestResponse$
];
exports.GetGroup$ = [9, n0, _GG,
    0, () => GetGroupRequest$, () => GetGroupResponse$
];
exports.GetGroupPolicy$ = [9, n0, _GGP,
    0, () => GetGroupPolicyRequest$, () => GetGroupPolicyResponse$
];
exports.GetHumanReadableSummary$ = [9, n0, _GHRS,
    0, () => GetHumanReadableSummaryRequest$, () => GetHumanReadableSummaryResponse$
];
exports.GetInstanceProfile$ = [9, n0, _GIP,
    0, () => GetInstanceProfileRequest$, () => GetInstanceProfileResponse$
];
exports.GetLoginProfile$ = [9, n0, _GLP,
    0, () => GetLoginProfileRequest$, () => GetLoginProfileResponse$
];
exports.GetMFADevice$ = [9, n0, _GMFAD,
    0, () => GetMFADeviceRequest$, () => GetMFADeviceResponse$
];
exports.GetOpenIDConnectProvider$ = [9, n0, _GOIDCP,
    0, () => GetOpenIDConnectProviderRequest$, () => GetOpenIDConnectProviderResponse$
];
exports.GetOrganizationsAccessReport$ = [9, n0, _GOARe,
    0, () => GetOrganizationsAccessReportRequest$, () => GetOrganizationsAccessReportResponse$
];
exports.GetOutboundWebIdentityFederationInfo$ = [9, n0, _GOWIFI,
    0, () => __Unit, () => GetOutboundWebIdentityFederationInfoResponse$
];
exports.GetPolicy$ = [9, n0, _GP,
    0, () => GetPolicyRequest$, () => GetPolicyResponse$
];
exports.GetPolicyVersion$ = [9, n0, _GPV,
    0, () => GetPolicyVersionRequest$, () => GetPolicyVersionResponse$
];
exports.GetRole$ = [9, n0, _GR,
    0, () => GetRoleRequest$, () => GetRoleResponse$
];
exports.GetRolePolicy$ = [9, n0, _GRP,
    0, () => GetRolePolicyRequest$, () => GetRolePolicyResponse$
];
exports.GetSAMLProvider$ = [9, n0, _GSAMLP,
    0, () => GetSAMLProviderRequest$, () => GetSAMLProviderResponse$
];
exports.GetServerCertificate$ = [9, n0, _GSC,
    0, () => GetServerCertificateRequest$, () => GetServerCertificateResponse$
];
exports.GetServiceLastAccessedDetails$ = [9, n0, _GSLADe,
    0, () => GetServiceLastAccessedDetailsRequest$, () => GetServiceLastAccessedDetailsResponse$
];
exports.GetServiceLastAccessedDetailsWithEntities$ = [9, n0, _GSLADWE,
    0, () => GetServiceLastAccessedDetailsWithEntitiesRequest$, () => GetServiceLastAccessedDetailsWithEntitiesResponse$
];
exports.GetServiceLinkedRoleDeletionStatus$ = [9, n0, _GSLRDS,
    0, () => GetServiceLinkedRoleDeletionStatusRequest$, () => GetServiceLinkedRoleDeletionStatusResponse$
];
exports.GetSSHPublicKey$ = [9, n0, _GSSHPK,
    0, () => GetSSHPublicKeyRequest$, () => GetSSHPublicKeyResponse$
];
exports.GetUser$ = [9, n0, _GU,
    0, () => GetUserRequest$, () => GetUserResponse$
];
exports.GetUserPolicy$ = [9, n0, _GUP,
    0, () => GetUserPolicyRequest$, () => GetUserPolicyResponse$
];
exports.ListAccessKeys$ = [9, n0, _LAK,
    0, () => ListAccessKeysRequest$, () => ListAccessKeysResponse$
];
exports.ListAccountAliases$ = [9, n0, _LAA,
    0, () => ListAccountAliasesRequest$, () => ListAccountAliasesResponse$
];
exports.ListAttachedGroupPolicies$ = [9, n0, _LAGP,
    0, () => ListAttachedGroupPoliciesRequest$, () => ListAttachedGroupPoliciesResponse$
];
exports.ListAttachedRolePolicies$ = [9, n0, _LARP,
    0, () => ListAttachedRolePoliciesRequest$, () => ListAttachedRolePoliciesResponse$
];
exports.ListAttachedUserPolicies$ = [9, n0, _LAUP,
    0, () => ListAttachedUserPoliciesRequest$, () => ListAttachedUserPoliciesResponse$
];
exports.ListDelegationRequests$ = [9, n0, _LDR,
    0, () => ListDelegationRequestsRequest$, () => ListDelegationRequestsResponse$
];
exports.ListEntitiesForPolicy$ = [9, n0, _LEFP,
    0, () => ListEntitiesForPolicyRequest$, () => ListEntitiesForPolicyResponse$
];
exports.ListGroupPolicies$ = [9, n0, _LGP,
    0, () => ListGroupPoliciesRequest$, () => ListGroupPoliciesResponse$
];
exports.ListGroups$ = [9, n0, _LG,
    0, () => ListGroupsRequest$, () => ListGroupsResponse$
];
exports.ListGroupsForUser$ = [9, n0, _LGFU,
    0, () => ListGroupsForUserRequest$, () => ListGroupsForUserResponse$
];
exports.ListInstanceProfiles$ = [9, n0, _LIP,
    0, () => ListInstanceProfilesRequest$, () => ListInstanceProfilesResponse$
];
exports.ListInstanceProfilesForRole$ = [9, n0, _LIPFR,
    0, () => ListInstanceProfilesForRoleRequest$, () => ListInstanceProfilesForRoleResponse$
];
exports.ListInstanceProfileTags$ = [9, n0, _LIPT,
    0, () => ListInstanceProfileTagsRequest$, () => ListInstanceProfileTagsResponse$
];
exports.ListMFADevices$ = [9, n0, _LMFAD,
    0, () => ListMFADevicesRequest$, () => ListMFADevicesResponse$
];
exports.ListMFADeviceTags$ = [9, n0, _LMFADT,
    0, () => ListMFADeviceTagsRequest$, () => ListMFADeviceTagsResponse$
];
exports.ListOpenIDConnectProviders$ = [9, n0, _LOIDCP,
    0, () => ListOpenIDConnectProvidersRequest$, () => ListOpenIDConnectProvidersResponse$
];
exports.ListOpenIDConnectProviderTags$ = [9, n0, _LOIDCPT,
    0, () => ListOpenIDConnectProviderTagsRequest$, () => ListOpenIDConnectProviderTagsResponse$
];
exports.ListOrganizationsFeatures$ = [9, n0, _LOF,
    0, () => ListOrganizationsFeaturesRequest$, () => ListOrganizationsFeaturesResponse$
];
exports.ListPolicies$ = [9, n0, _LPi,
    0, () => ListPoliciesRequest$, () => ListPoliciesResponse$
];
exports.ListPoliciesGrantingServiceAccess$ = [9, n0, _LPGSA,
    0, () => ListPoliciesGrantingServiceAccessRequest$, () => ListPoliciesGrantingServiceAccessResponse$
];
exports.ListPolicyTags$ = [9, n0, _LPT,
    0, () => ListPolicyTagsRequest$, () => ListPolicyTagsResponse$
];
exports.ListPolicyVersions$ = [9, n0, _LPV,
    0, () => ListPolicyVersionsRequest$, () => ListPolicyVersionsResponse$
];
exports.ListRolePolicies$ = [9, n0, _LRP,
    0, () => ListRolePoliciesRequest$, () => ListRolePoliciesResponse$
];
exports.ListRoles$ = [9, n0, _LR,
    0, () => ListRolesRequest$, () => ListRolesResponse$
];
exports.ListRoleTags$ = [9, n0, _LRT,
    0, () => ListRoleTagsRequest$, () => ListRoleTagsResponse$
];
exports.ListSAMLProviders$ = [9, n0, _LSAMLP,
    0, () => ListSAMLProvidersRequest$, () => ListSAMLProvidersResponse$
];
exports.ListSAMLProviderTags$ = [9, n0, _LSAMLPT,
    0, () => ListSAMLProviderTagsRequest$, () => ListSAMLProviderTagsResponse$
];
exports.ListServerCertificates$ = [9, n0, _LSC,
    0, () => ListServerCertificatesRequest$, () => ListServerCertificatesResponse$
];
exports.ListServerCertificateTags$ = [9, n0, _LSCT,
    0, () => ListServerCertificateTagsRequest$, () => ListServerCertificateTagsResponse$
];
exports.ListServiceSpecificCredentials$ = [9, n0, _LSSC,
    0, () => ListServiceSpecificCredentialsRequest$, () => ListServiceSpecificCredentialsResponse$
];
exports.ListSigningCertificates$ = [9, n0, _LSCi,
    0, () => ListSigningCertificatesRequest$, () => ListSigningCertificatesResponse$
];
exports.ListSSHPublicKeys$ = [9, n0, _LSSHPK,
    0, () => ListSSHPublicKeysRequest$, () => ListSSHPublicKeysResponse$
];
exports.ListUserPolicies$ = [9, n0, _LUP,
    0, () => ListUserPoliciesRequest$, () => ListUserPoliciesResponse$
];
exports.ListUsers$ = [9, n0, _LU,
    0, () => ListUsersRequest$, () => ListUsersResponse$
];
exports.ListUserTags$ = [9, n0, _LUT,
    0, () => ListUserTagsRequest$, () => ListUserTagsResponse$
];
exports.ListVirtualMFADevices$ = [9, n0, _LVMFAD,
    0, () => ListVirtualMFADevicesRequest$, () => ListVirtualMFADevicesResponse$
];
exports.PutGroupPolicy$ = [9, n0, _PGP,
    0, () => PutGroupPolicyRequest$, () => __Unit
];
exports.PutRolePermissionsBoundary$ = [9, n0, _PRPB,
    0, () => PutRolePermissionsBoundaryRequest$, () => __Unit
];
exports.PutRolePolicy$ = [9, n0, _PRPu,
    0, () => PutRolePolicyRequest$, () => __Unit
];
exports.PutUserPermissionsBoundary$ = [9, n0, _PUPB,
    0, () => PutUserPermissionsBoundaryRequest$, () => __Unit
];
exports.PutUserPolicy$ = [9, n0, _PUP,
    0, () => PutUserPolicyRequest$, () => __Unit
];
exports.RejectDelegationRequest$ = [9, n0, _RDR,
    0, () => RejectDelegationRequestRequest$, () => __Unit
];
exports.RemoveClientIDFromOpenIDConnectProvider$ = [9, n0, _RCIDFOIDCP,
    0, () => RemoveClientIDFromOpenIDConnectProviderRequest$, () => __Unit
];
exports.RemoveRoleFromInstanceProfile$ = [9, n0, _RRFIP,
    0, () => RemoveRoleFromInstanceProfileRequest$, () => __Unit
];
exports.RemoveUserFromGroup$ = [9, n0, _RUFG,
    0, () => RemoveUserFromGroupRequest$, () => __Unit
];
exports.ResetServiceSpecificCredential$ = [9, n0, _RSSC,
    0, () => ResetServiceSpecificCredentialRequest$, () => ResetServiceSpecificCredentialResponse$
];
exports.ResyncMFADevice$ = [9, n0, _RMFAD,
    0, () => ResyncMFADeviceRequest$, () => __Unit
];
exports.SendDelegationToken$ = [9, n0, _SDT,
    0, () => SendDelegationTokenRequest$, () => __Unit
];
exports.SetDefaultPolicyVersion$ = [9, n0, _SDPV,
    0, () => SetDefaultPolicyVersionRequest$, () => __Unit
];
exports.SetSecurityTokenServicePreferences$ = [9, n0, _SSTSP,
    0, () => SetSecurityTokenServicePreferencesRequest$, () => __Unit
];
exports.SimulateCustomPolicy$ = [9, n0, _SCP,
    0, () => SimulateCustomPolicyRequest$, () => SimulatePolicyResponse$
];
exports.SimulatePrincipalPolicy$ = [9, n0, _SPP,
    0, () => SimulatePrincipalPolicyRequest$, () => SimulatePolicyResponse$
];
exports.TagInstanceProfile$ = [9, n0, _TIP,
    0, () => TagInstanceProfileRequest$, () => __Unit
];
exports.TagMFADevice$ = [9, n0, _TMFAD,
    0, () => TagMFADeviceRequest$, () => __Unit
];
exports.TagOpenIDConnectProvider$ = [9, n0, _TOIDCP,
    0, () => TagOpenIDConnectProviderRequest$, () => __Unit
];
exports.TagPolicy$ = [9, n0, _TP,
    0, () => TagPolicyRequest$, () => __Unit
];
exports.TagRole$ = [9, n0, _TR,
    0, () => TagRoleRequest$, () => __Unit
];
exports.TagSAMLProvider$ = [9, n0, _TSAMLP,
    0, () => TagSAMLProviderRequest$, () => __Unit
];
exports.TagServerCertificate$ = [9, n0, _TSC,
    0, () => TagServerCertificateRequest$, () => __Unit
];
exports.TagUser$ = [9, n0, _TU,
    0, () => TagUserRequest$, () => __Unit
];
exports.UntagInstanceProfile$ = [9, n0, _UIP,
    0, () => UntagInstanceProfileRequest$, () => __Unit
];
exports.UntagMFADevice$ = [9, n0, _UMFAD,
    0, () => UntagMFADeviceRequest$, () => __Unit
];
exports.UntagOpenIDConnectProvider$ = [9, n0, _UOIDCP,
    0, () => UntagOpenIDConnectProviderRequest$, () => __Unit
];
exports.UntagPolicy$ = [9, n0, _UP,
    0, () => UntagPolicyRequest$, () => __Unit
];
exports.UntagRole$ = [9, n0, _UR,
    0, () => UntagRoleRequest$, () => __Unit
];
exports.UntagSAMLProvider$ = [9, n0, _USAMLP,
    0, () => UntagSAMLProviderRequest$, () => __Unit
];
exports.UntagServerCertificate$ = [9, n0, _USC,
    0, () => UntagServerCertificateRequest$, () => __Unit
];
exports.UntagUser$ = [9, n0, _UU,
    0, () => UntagUserRequest$, () => __Unit
];
exports.UpdateAccessKey$ = [9, n0, _UAK,
    0, () => UpdateAccessKeyRequest$, () => __Unit
];
exports.UpdateAccountPasswordPolicy$ = [9, n0, _UAPP,
    0, () => UpdateAccountPasswordPolicyRequest$, () => __Unit
];
exports.UpdateAssumeRolePolicy$ = [9, n0, _UARP,
    0, () => UpdateAssumeRolePolicyRequest$, () => __Unit
];
exports.UpdateDelegationRequest$ = [9, n0, _UDR,
    0, () => UpdateDelegationRequestRequest$, () => __Unit
];
exports.UpdateGroup$ = [9, n0, _UG,
    0, () => UpdateGroupRequest$, () => __Unit
];
exports.UpdateLoginProfile$ = [9, n0, _ULP,
    0, () => UpdateLoginProfileRequest$, () => __Unit
];
exports.UpdateOpenIDConnectProviderThumbprint$ = [9, n0, _UOIDCPT,
    0, () => UpdateOpenIDConnectProviderThumbprintRequest$, () => __Unit
];
exports.UpdateRole$ = [9, n0, _URp,
    0, () => UpdateRoleRequest$, () => UpdateRoleResponse$
];
exports.UpdateRoleDescription$ = [9, n0, _URD,
    0, () => UpdateRoleDescriptionRequest$, () => UpdateRoleDescriptionResponse$
];
exports.UpdateSAMLProvider$ = [9, n0, _USAMLPp,
    0, () => UpdateSAMLProviderRequest$, () => UpdateSAMLProviderResponse$
];
exports.UpdateServerCertificate$ = [9, n0, _USCp,
    0, () => UpdateServerCertificateRequest$, () => __Unit
];
exports.UpdateServiceSpecificCredential$ = [9, n0, _USSC,
    0, () => UpdateServiceSpecificCredentialRequest$, () => __Unit
];
exports.UpdateSigningCertificate$ = [9, n0, _USCpd,
    0, () => UpdateSigningCertificateRequest$, () => __Unit
];
exports.UpdateSSHPublicKey$ = [9, n0, _USSHPK,
    0, () => UpdateSSHPublicKeyRequest$, () => __Unit
];
exports.UpdateUser$ = [9, n0, _UUp,
    0, () => UpdateUserRequest$, () => __Unit
];
exports.UploadServerCertificate$ = [9, n0, _USCpl,
    0, () => UploadServerCertificateRequest$, () => UploadServerCertificateResponse$
];
exports.UploadSigningCertificate$ = [9, n0, _USCplo,
    0, () => UploadSigningCertificateRequest$, () => UploadSigningCertificateResponse$
];
exports.UploadSSHPublicKey$ = [9, n0, _USSHPKp,
    0, () => UploadSSHPublicKeyRequest$, () => UploadSSHPublicKeyResponse$
];
