// @aws-sdk/client-cognito-identity-provider/dist-cjs/schemas/schemas_0.js
const _A = "Actions";
const _AA = "ApplicationArn";
const _AACUO = "AllowAdminCreateUserOnly";
const _AAUTG = "AdminAddUserToGroup";
const _AAUTGR = "AdminAddUserToGroupRequest";
const _AAl = "AliasAttributes";
const _AAu = "AuthenticatorAttachment";
const _AC = "AvailableChallenges";
const _ACA = "AddCustomAttributes";
const _ACAR = "AddCustomAttributesRequest";
const _ACARd = "AddCustomAttributesResponse";
const _ACSU = "AdminConfirmSignUp";
const _ACSUR = "AdminConfirmSignUpRequest";
const _ACSURd = "AdminConfirmSignUpResponse";
const _ACT = "AnalyticsConfigurationType";
const _ACU = "AdminCreateUser";
const _ACUC = "AdminCreateUserConfig";
const _ACUCT = "AdminCreateUserConfigType";
const _ACUR = "AdminCreateUserRequest";
const _ACURd = "AdminCreateUserResponse";
const _ACn = "AnalyticsConfiguration";
const _ADE = "AccessDeniedException";
const _ADPFU = "AdminDisableProviderForUser";
const _ADPFUR = "AdminDisableProviderForUserRequest";
const _ADPFURd = "AdminDisableProviderForUserResponse";
const _ADT = "AttributeDataType";
const _ADU = "AdminDeleteUser";
const _ADUA = "AdminDeleteUserAttributes";
const _ADUAR = "AdminDeleteUserAttributesRequest";
const _ADUARd = "AdminDeleteUserAttributesResponse";
const _ADUR = "AdminDeleteUserRequest";
const _ADURd = "AdminDisableUserRequest";
const _ADURdm = "AdminDisableUserResponse";
const _ADUd = "AdminDisableUser";
const _AE = "AuthEvents";
const _AEE = "AliasExistsException";
const _AEI = "AnalyticsEndpointId";
const _AET = "AuthEventType";
const _AETu = "AuthEventsType";
const _AEU = "AdminEnableUser";
const _AEUR = "AdminEnableUserRequest";
const _AEURd = "AdminEnableUserResponse";
const _AF = "AuthFlow";
const _AFAF = "AllowedFirstAuthFactors";
const _AFD = "AdminForgetDevice";
const _AFDR = "AdminForgetDeviceRequest";
const _AGD = "AdminGetDevice";
const _AGDR = "AdminGetDeviceRequest";
const _AGDRd = "AdminGetDeviceResponse";
const _AGU = "AdminGetUser";
const _AGUR = "AdminGetUserRequest";
const _AGURd = "AdminGetUserResponse";
const _AI = "ApplicationId";
const _AIA = "AdminInitiateAuth";
const _AIAR = "AdminInitiateAuthRequest";
const _AIARd = "AdminInitiateAuthResponse";
const _ALD = "AdminListDevices";
const _ALDR = "AdminListDevicesRequest";
const _ALDRd = "AdminListDevicesResponse";
const _ALGFU = "AdminListGroupsForUser";
const _ALGFUR = "AdminListGroupsForUserRequest";
const _ALGFURd = "AdminListGroupsForUserResponse";
const _ALPFU = "AdminLinkProviderForUser";
const _ALPFUR = "AdminLinkProviderForUserRequest";
const _ALPFURd = "AdminLinkProviderForUserResponse";
const _ALT = "AssetListType";
const _ALTt = "AttributeListType";
const _ALUAE = "AdminListUserAuthEvents";
const _ALUAER = "AdminListUserAuthEventsRequest";
const _ALUAERd = "AdminListUserAuthEventsResponse";
const _AM = "AnalyticsMetadata";
const _AMT = "AnalyticsMetadataType";
const _AMt = "AttributeMapping";
const _AN = "AttributeName";
const _AOAF = "AllowedOAuthFlows";
const _AOAFUPC = "AllowedOAuthFlowsUserPoolClient";
const _AOAS = "AllowedOAuthScopes";
const _AP = "AuthParameters";
const _APT = "AuthParametersType";
const _AR = "AuthenticationResult";
const _ARS = "AccountRecoverySetting";
const _ARST = "AccountRecoverySettingType";
const _ART = "AuthenticationResultType";
const _ARTAC = "AdminRespondToAuthChallenge";
const _ARTACR = "AdminRespondToAuthChallengeRequest";
const _ARTACRd = "AdminRespondToAuthChallengeResponse";
const _ARUFG = "AdminRemoveUserFromGroup";
const _ARUFGR = "AdminRemoveUserFromGroupRequest";
const _ARUP = "AdminResetUserPassword";
const _ARUPR = "AdminResetUserPasswordRequest";
const _ARUPRd = "AdminResetUserPasswordResponse";
const _ARVBU = "AttributesRequireVerificationBeforeUpdate";
const _ASAF = "AdvancedSecurityAdditionalFlows";
const _ASAFT = "AdvancedSecurityAdditionalFlowsType";
const _ASM = "AdvancedSecurityMode";
const _AST = "AssociateSoftwareToken";
const _ASTR = "AssociateSoftwareTokenRequest";
const _ASTRs = "AssociateSoftwareTokenResponse";
const _ASUMFAP = "AdminSetUserMFAPreference";
const _ASUMFAPR = "AdminSetUserMFAPreferenceRequest";
const _ASUMFAPRd = "AdminSetUserMFAPreferenceResponse";
const _ASUP = "AdminSetUserPassword";
const _ASUPR = "AdminSetUserPasswordRequest";
const _ASUPRd = "AdminSetUserPasswordResponse";
const _ASUS = "AdminSetUserSettings";
const _ASUSR = "AdminSetUserSettingsRequest";
const _ASUSRd = "AdminSetUserSettingsResponse";
const _ASV = "AuthSessionValidity";
const _AT = "AssetType";
const _ATAT = "AccountTakeoverActionsType";
const _ATATc = "AccountTakeoverActionType";
const _ATG = "AttributesToGet";
const _ATRC = "AccountTakeoverRiskConfiguration";
const _ATRCT = "AccountTakeoverRiskConfigurationType";
const _ATV = "AccessTokenValidity";
const _ATc = "AccessToken";
const _ATt = "AttributeType";
const _ATu = "AuthenticatorTransports";
const _AUAEF = "AdminUpdateAuthEventFeedback";
const _AUAEFR = "AdminUpdateAuthEventFeedbackRequest";
const _AUAEFRd = "AdminUpdateAuthEventFeedbackResponse";
const _AUDS = "AdminUpdateDeviceStatus";
const _AUDSR = "AdminUpdateDeviceStatusRequest";
const _AUDSRd = "AdminUpdateDeviceStatusResponse";
const _AUGSO = "AdminUserGlobalSignOut";
const _AUGSOR = "AdminUserGlobalSignOutRequest";
const _AUGSORd = "AdminUserGlobalSignOutResponse";
const _AUPCS = "AddUserPoolClientSecret";
const _AUPCSR = "AddUserPoolClientSecretRequest";
const _AUPCSRd = "AddUserPoolClientSecretResponse";
const _AUUA = "AdminUpdateUserAttributes";
const _AUUAR = "AdminUpdateUserAttributesRequest";
const _AUUARd = "AdminUpdateUserAttributesResponse";
const _AVA = "AutoVerifiedAttributes";
const _AVT = "AttributeValueType";
const _AWSAI = "AWSAccountId";
const _Ar = "Arn";
const _As = "Assets";
const _At = "Attributes";
const _B = "Bytes";
const _BA = "BucketArn";
const _BE = "BlockEmail";
const _BIPRL = "BlockedIPRangeList";
const _C = "Category";
const _CA = "CustomAttributes";
const _CAC = "CreateAuthChallenge";
const _CALT = "CustomAttributesListType";
const _CAM = "CustomAuthMode";
const _CAe = "CertificateArn";
const _CAr = "CreatedAt";
const _CC = "ConfirmationCode";
const _CCAT = "CompromisedCredentialsActionsType";
const _CCD = "CompromisedCredentialsDetected";
const _CCO = "CredentialCreationOptions";
const _CCRC = "CompromisedCredentialsRiskConfiguration";
const _CCRCT = "CompromisedCredentialsRiskConfigurationType";
const _CD = "ContextData";
const _CDC = "CustomDomainConfig";
const _CDCT = "CustomDomainConfigType";
const _CDD = "CodeDeliveryDetails";
const _CDDL = "CodeDeliveryDetailsList";
const _CDDLT = "CodeDeliveryDetailsListType";
const _CDDT = "CodeDeliveryDetailsType";
const _CDFE = "CodeDeliveryFailureException";
const _CDR = "ConfirmDeviceRequest";
const _CDRo = "ConfirmDeviceResponse";
const _CDT = "ContextDataType";
const _CDo = "CompletionDate";
const _CDon = "ConfirmDevice";
const _CDr = "CreationDate";
const _CDu = "CustomDomain";
const _CELVCT = "CustomEmailLambdaVersionConfigType";
const _CES = "CustomEmailSender";
const _CFD = "CloudFrontDomain";
const _CFDl = "CloudFrontDistribution";
const _CFP = "ConfirmForgotPassword";
const _CFPR = "ConfirmForgotPasswordRequest";
const _CFPRo = "ConfirmForgotPasswordResponse";
const _CG = "CreateGroup";
const _CGR = "CreateGroupRequest";
const _CGRr = "CreateGroupResponse";
const _CI = "ClientId";
const _CIP = "CreateIdentityProvider";
const _CIPR = "CreateIdentityProviderRequest";
const _CIPRr = "CreateIdentityProviderResponse";
const _CIT = "ClientIdType";
const _CIr = "CredentialId";
const _CM = "ClientMetadata";
const _CME = "CodeMismatchException";
const _CMEo = "ConcurrentModificationException";
const _CMLB = "CreateManagedLoginBranding";
const _CMLBR = "CreateManagedLoginBrandingRequest";
const _CMLBRr = "CreateManagedLoginBrandingResponse";
const _CMo = "ColorMode";
const _CMom = "CompletionMessage";
const _CMu = "CustomMessage";
const _CN = "ChallengeName";
const _CNl = "ClientName";
const _CP = "ChallengeParameters";
const _CPR = "ChangePasswordRequest";
const _CPRh = "ChangePasswordResponse";
const _CPh = "ChangePassword";
const _CR = "ChallengeResponses";
const _CRLT = "ChallengeResponseListType";
const _CROND = "ChallengeRequiredOnNewDevice";
const _CRS = "CreateResourceServer";
const _CRSR = "CreateResourceServerRequest";
const _CRSRr = "CreateResourceServerResponse";
const _CRT = "ChallengeResponseType";
const _CRTh = "ChallengeResponsesType";
const _CRh = "ChallengeResponse";
const _CS = "ClientSecret";
const _CSCD = "ClientSecretCreateDate";
const _CSD = "ClientSecretDescriptor";
const _CSDLT = "ClientSecretDescriptorListType";
const _CSDT = "ClientSecretDescriptorType";
const _CSI = "ClientSecretId";
const _CSMSLVCT = "CustomSMSLambdaVersionConfigType";
const _CSMSS = "CustomSMSSender";
const _CSS = "CSS";
const _CSSV = "CSSVersion";
const _CST = "ClientSecretType";
const _CSU = "ConfirmSignUp";
const _CSUR = "ConfirmSignUpRequest";
const _CSURo = "ConfirmSignUpResponse";
const _CSV = "ClientSecretValue";
const _CSVH = "CSVHeader";
const _CSa = "CaseSensitive";
const _CSl = "ClientSecrets";
const _CSo = "ConfigurationSet";
const _CT = "CreateTerms";
const _CTR = "CreateTermsRequest";
const _CTRr = "CreateTermsResponse";
const _CUAF = "ConfiguredUserAuthFactors";
const _CUIJ = "CreateUserImportJob";
const _CUIJR = "CreateUserImportJobRequest";
const _CUIJRr = "CreateUserImportJobResponse";
const _CUP = "CreateUserPool";
const _CUPC = "CreateUserPoolClient";
const _CUPCR = "CreateUserPoolClientRequest";
const _CUPCRr = "CreateUserPoolClientResponse";
const _CUPD = "CreateUserPoolDomain";
const _CUPDR = "CreateUserPoolDomainRequest";
const _CUPDRr = "CreateUserPoolDomainResponse";
const _CUPR = "CreateUserPoolRequest";
const _CUPRR = "CreateUserPoolReplicaRequest";
const _CUPRRr = "CreateUserPoolReplicaResponse";
const _CUPRr = "CreateUserPoolResponse";
const _CUPRre = "CreateUserPoolReplica";
const _CURL = "CallbackURLs";
const _CWAR = "CompleteWebAuthnRegistration";
const _CWARR = "CompleteWebAuthnRegistrationRequest";
const _CWARRo = "CompleteWebAuthnRegistrationResponse";
const _CWLC = "CloudWatchLogsConfiguration";
const _CWLCT = "CloudWatchLogsConfigurationType";
const _CWLRA = "CloudWatchLogsRoleArn";
const _Ce = "Certificate";
const _Ci = "City";
const _Co = "Country";
const _Cod = "Code";
const _Cr = "Credential";
const _Cre = "Credentials";
const _D = "Device";
const _DA = "DeviceAttributes";
const _DAC = "DefineAuthChallenge";
const _DC = "DeviceConfiguration";
const _DCD = "DeviceCreateDate";
const _DCT = "DeviceConfigurationType";
const _DD = "DomainDescription";
const _DDM = "DesiredDeliveryMediums";
const _DDT = "DomainDescriptionType";
const _DEO = "DefaultEmailOption";
const _DG = "DeleteGroup";
const _DGK = "DeviceGroupKey";
const _DGR = "DeleteGroupRequest";
const _DIP = "DeleteIdentityProvider";
const _DIPR = "DeleteIdentityProviderRequest";
const _DIPRe = "DescribeIdentityProviderRequest";
const _DIPRes = "DescribeIdentityProviderResponse";
const _DIPe = "DescribeIdentityProvider";
const _DK = "DeviceKey";
const _DKEE = "DeviceKeyExistsException";
const _DLAD = "DeviceLastAuthenticatedDate";
const _DLMD = "DeviceLastModifiedDate";
const _DLT = "DeviceListType";
const _DM = "DeliveryMedium";
const _DMLB = "DeleteManagedLoginBranding";
const _DMLBBC = "DescribeManagedLoginBrandingByClient";
const _DMLBBCR = "DescribeManagedLoginBrandingByClientRequest";
const _DMLBBCRe = "DescribeManagedLoginBrandingByClientResponse";
const _DMLBR = "DeleteManagedLoginBrandingRequest";
const _DMLBRe = "DescribeManagedLoginBrandingRequest";
const _DMLBRes = "DescribeManagedLoginBrandingResponse";
const _DMLBe = "DescribeManagedLoginBranding";
const _DN = "DeviceName";
const _DOA = "DeveloperOnlyAttribute";
const _DOROUP = "DeviceOnlyRememberedOnUserPrompt";
const _DP = "DeletionProtection";
const _DPE = "DuplicateProviderException";
const _DRC = "DescribeRiskConfiguration";
const _DRCR = "DescribeRiskConfigurationRequest";
const _DRCRe = "DescribeRiskConfigurationResponse";
const _DRS = "DeviceRememberedStatus";
const _DRSR = "DeleteResourceServerRequest";
const _DRSRe = "DescribeResourceServerRequest";
const _DRSRes = "DescribeResourceServerResponse";
const _DRSe = "DeleteResourceServer";
const _DRSes = "DescribeResourceServer";
const _DRURI = "DefaultRedirectURI";
const _DSVC = "DeviceSecretVerifierConfig";
const _DSVCT = "DeviceSecretVerifierConfigType";
const _DT = "DeviceType";
const _DTR = "DeleteTermsRequest";
const _DTRe = "DescribeTermsRequest";
const _DTRes = "DescribeTermsResponse";
const _DTe = "DeleteTerms";
const _DTes = "DescribeTerms";
const _DU = "DestinationUser";
const _DUA = "DeleteUserAttributes";
const _DUAR = "DeleteUserAttributesRequest";
const _DUARe = "DeleteUserAttributesResponse";
const _DUIJ = "DescribeUserImportJob";
const _DUIJR = "DescribeUserImportJobRequest";
const _DUIJRe = "DescribeUserImportJobResponse";
const _DUP = "DeleteUserPool";
const _DUPC = "DeleteUserPoolClient";
const _DUPCR = "DeleteUserPoolClientRequest";
const _DUPCRe = "DescribeUserPoolClientRequest";
const _DUPCRes = "DescribeUserPoolClientResponse";
const _DUPCS = "DeleteUserPoolClientSecret";
const _DUPCSR = "DeleteUserPoolClientSecretRequest";
const _DUPCSRe = "DeleteUserPoolClientSecretResponse";
const _DUPCe = "DescribeUserPoolClient";
const _DUPD = "DeleteUserPoolDomain";
const _DUPDR = "DeleteUserPoolDomainRequest";
const _DUPDRe = "DeleteUserPoolDomainResponse";
const _DUPDRes = "DescribeUserPoolDomainRequest";
const _DUPDResc = "DescribeUserPoolDomainResponse";
const _DUPDe = "DescribeUserPoolDomain";
const _DUPR = "DeleteUserPoolRequest";
const _DUPRR = "DeleteUserPoolReplicaRequest";
const _DUPRRe = "DeleteUserPoolReplicaResponse";
const _DUPRe = "DescribeUserPoolRequest";
const _DUPRel = "DeleteUserPoolReplica";
const _DUPRes = "DescribeUserPoolResponse";
const _DUPe = "DescribeUserPool";
const _DUR = "DeleteUserRequest";
const _DUe = "DeleteUser";
const _DWAC = "DeleteWebAuthnCredential";
const _DWACR = "DeleteWebAuthnCredentialRequest";
const _DWACRe = "DeleteWebAuthnCredentialResponse";
const _De = "Devices";
const _Des = "Destination";
const _Desc = "Description";
const _Do = "Domain";
const _E = "Enabled";
const _EA = "EventAction";
const _EAF = "ExplicitAuthFlows";
const _EC = "EmailConfiguration";
const _ECD = "EventContextData";
const _ECDT = "EventContextDataType";
const _ECE = "ExpiredCodeException";
const _ECF = "EmailConfigurationFailure";
const _ECT = "EmailConfigurationType";
const _ED = "EncodedData";
const _EF = "EventFeedback";
const _EFT = "EventFeedbackType";
const _EFv = "EventFilter";
const _EI = "EventId";
const _EIx = "ExternalId";
const _EIxp = "ExpiresIn";
const _EM = "EmailMessage";
const _EMBL = "EmailMessageByLink";
const _EMC = "EmailMfaConfiguration";
const _EMCT = "EmailMfaConfigType";
const _EMS = "EmailMfaSettings";
const _EMST = "EmailMfaSettingsType";
const _ENOU = "EstimatedNumberOfUsers";
const _EPAUCD = "EnablePropagateAdditionalUserContextData";
const _ER = "EventResponse";
const _ERT = "EventRiskType";
const _ERv = "EventRisk";
const _ES = "EventSource";
const _ESA = "EmailSendingAccount";
const _ESBL = "EmailSubjectByLink";
const _ESTMFAE = "EnableSoftwareTokenMFAException";
const _ESm = "EmailSubject";
const _ET = "EventType";
const _ETR = "EnableTokenRevocation";
const _EVM = "EmailVerificationMessage";
const _EVS = "EmailVerificationSubject";
const _En = "Enforcement";
const _Ex = "Extension";
const _F = "From";
const _FAC = "ForceAliasCreation";
const _FC = "FirehoseConfiguration";
const _FCN = "FriendlyCredentialName";
const _FCT = "FirehoseConfigurationType";
const _FCa = "FactorConfiguration";
const _FD = "FeedbackDate";
const _FDN = "FriendlyDeviceName";
const _FDR = "ForgetDeviceRequest";
const _FDo = "ForgetDevice";
const _FE = "ForbiddenException";
const _FP = "ForgotPassword";
const _FPR = "ForgotPasswordRequest";
const _FPRo = "ForgotPasswordResponse";
const _FT = "FailoverType";
const _FTe = "FeedbackToken";
const _FU = "FailedUsers";
const _FUITE = "FeatureUnavailableInTierException";
const _FV = "FeedbackValue";
const _Fa = "Failover";
const _Fe = "Feature";
const _Fi = "Filter";
const _G = "Groups";
const _GCSVH = "GetCSVHeader";
const _GCSVHR = "GetCSVHeaderRequest";
const _GCSVHRe = "GetCSVHeaderResponse";
const _GD = "GetDevice";
const _GDR = "GetDeviceRequest";
const _GDRe = "GetDeviceResponse";
const _GEE = "GroupExistsException";
const _GG = "GetGroup";
const _GGR = "GetGroupRequest";
const _GGRe = "GetGroupResponse";
const _GIPBI = "GetIdentityProviderByIdentifier";
const _GIPBIR = "GetIdentityProviderByIdentifierRequest";
const _GIPBIRe = "GetIdentityProviderByIdentifierResponse";
const _GLDC = "GetLogDeliveryConfiguration";
const _GLDCR = "GetLogDeliveryConfigurationRequest";
const _GLDCRe = "GetLogDeliveryConfigurationResponse";
const _GLT = "GroupListType";
const _GN = "GroupName";
const _GS = "GenerateSecret";
const _GSC = "GetSigningCertificate";
const _GSCR = "GetSigningCertificateRequest";
const _GSCRe = "GetSigningCertificateResponse";
const _GSO = "GlobalSignOut";
const _GSOR = "GlobalSignOutRequest";
const _GSORl = "GlobalSignOutResponse";
const _GT = "GroupType";
const _GTFRT = "GetTokensFromRefreshToken";
const _GTFRTR = "GetTokensFromRefreshTokenRequest";
const _GTFRTRe = "GetTokensFromRefreshTokenResponse";
const _GU = "GetUser";
const _GUAF = "GetUserAuthFactors";
const _GUAFR = "GetUserAuthFactorsRequest";
const _GUAFRe = "GetUserAuthFactorsResponse";
const _GUAVC = "GetUserAttributeVerificationCode";
const _GUAVCR = "GetUserAttributeVerificationCodeRequest";
const _GUAVCRe = "GetUserAttributeVerificationCodeResponse";
const _GUIC = "GetUICustomization";
const _GUICR = "GetUICustomizationRequest";
const _GUICRe = "GetUICustomizationResponse";
const _GUPMC = "GetUserPoolMfaConfig";
const _GUPMCR = "GetUserPoolMfaConfigRequest";
const _GUPMCRe = "GetUserPoolMfaConfigResponse";
const _GUR = "GetUserRequest";
const _GURe = "GetUserResponse";
const _Gr = "Group";
const _HA = "HighAction";
const _HB = "HtmlBody";
const _HH = "HttpHeaders";
const _HHL = "HttpHeaderList";
const _HHt = "HttpHeader";
const _I = "Identifier";
const _IA = "IpAddress";
const _IAR = "InitiateAuthRequest";
const _IARn = "InitiateAuthResponse";
const _IAn = "InitiateAuth";
const _IC = "IssuerConfiguration";
const _ICT = "IssuerConfigurationType";
const _IEE = "InternalErrorException";
const _IERAPE = "InvalidEmailRoleAccessPolicyException";
const _IF = "InboundFederation";
const _IFLT = "InboundFederationLambdaType";
const _IFm = "ImageFile";
const _II = "IdpIdentifiers";
const _IId = "IdpIdentifier";
const _ILRE = "InvalidLambdaResponseException";
const _IMT = "InviteMessageTemplate";
const _IOAFE = "InvalidOAuthFlowException";
const _IP = "IdentityProvider";
const _IPE = "InvalidParameterException";
const _IPEn = "InvalidPasswordException";
const _IPT = "IdentityProviderType";
const _ISE = "InternalServerException";
const _ISRAPE = "InvalidSmsRoleAccessPolicyException";
const _ISRTRE = "InvalidSmsRoleTrustRelationshipException";
const _IT = "IdToken";
const _ITV = "IdTokenValidity";
const _IU = "ImageUrl";
const _IUPCE = "InvalidUserPoolConfigurationException";
const _IUm = "ImportedUsers";
const _Id = "Id";
const _JI = "JobId";
const _JN = "JobName";
const _KC = "KeyConfiguration";
const _KCT = "KeyConfigurationType";
const _KKA = "KmsKeyArn";
const _KMSKID = "KMSKeyID";
const _KT = "KeyType";
const _L = "Limit";
const _LA = "LowAction";
const _LAa = "LambdaArn";
const _LC = "LambdaConfig";
const _LCLT = "LogConfigurationListType";
const _LCT = "LambdaConfigType";
const _LCTo = "LogConfigurationType";
const _LCo = "LogConfigurations";
const _LD = "ListDevices";
const _LDC = "LogDeliveryConfiguration";
const _LDCT = "LogDeliveryConfigurationType";
const _LDR = "ListDevicesRequest";
const _LDRi = "ListDevicesResponse";
const _LEE = "LimitExceededException";
const _LG = "ListGroups";
const _LGA = "LogGroupArn";
const _LGR = "ListGroupsRequest";
const _LGRi = "ListGroupsResponse";
const _LIP = "ListIdentityProviders";
const _LIPR = "ListIdentityProvidersRequest";
const _LIPRi = "ListIdentityProvidersResponse";
const _LL = "LogLevel";
const _LMD = "LastModifiedDate";
const _LRS = "ListResourceServers";
const _LRSR = "ListResourceServersRequest";
const _LRSRi = "ListResourceServersResponse";
const _LT = "ListTerms";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTR = "ListTermsRequest";
const _LTRi = "ListTermsResponse";
const _LU = "ListUsers";
const _LUIG = "ListUsersInGroup";
const _LUIGR = "ListUsersInGroupRequest";
const _LUIGRi = "ListUsersInGroupResponse";
const _LUIJ = "ListUserImportJobs";
const _LUIJR = "ListUserImportJobsRequest";
const _LUIJRi = "ListUserImportJobsResponse";
const _LUP = "ListUserPools";
const _LUPC = "ListUserPoolClients";
const _LUPCR = "ListUserPoolClientsRequest";
const _LUPCRi = "ListUserPoolClientsResponse";
const _LUPCS = "ListUserPoolClientSecrets";
const _LUPCSR = "ListUserPoolClientSecretsRequest";
const _LUPCSRi = "ListUserPoolClientSecretsResponse";
const _LUPR = "ListUserPoolsRequest";
const _LUPRR = "ListUserPoolReplicasRequest";
const _LUPRRi = "ListUserPoolReplicasResponse";
const _LUPRi = "ListUserPoolsResponse";
const _LUPRis = "ListUserPoolReplicas";
const _LUR = "ListUsersRequest";
const _LURL = "LogoutURLs";
const _LURi = "ListUsersResponse";
const _LV = "LambdaVersion";
const _LWAC = "ListWebAuthnCredentials";
const _LWACR = "ListWebAuthnCredentialsRequest";
const _LWACRi = "ListWebAuthnCredentialsResponse";
const _Li = "Links";
const _M = "Message";
const _MA = "MediumAction";
const _MAe = "MessageAction";
const _MC = "MfaConfiguration";
const _ME = "MfaEmail";
const _MFAMNFE = "MFAMethodNotFoundException";
const _MFAO = "MFAOptions";
const _MFAOLT = "MFAOptionListType";
const _MFAOT = "MFAOptionType";
const _ML = "MinimumLength";
const _MLB = "ManagedLoginBranding";
const _MLBEE = "ManagedLoginBrandingExistsException";
const _MLBI = "ManagedLoginBrandingId";
const _MLBT = "ManagedLoginBrandingType";
const _MLV = "ManagedLoginVersion";
const _MLa = "MaxLength";
const _MLi = "MinLength";
const _MR = "MaxResults";
const _MTT = "MessageTemplateType";
const _MV = "MinValue";
const _MVa = "MaxValue";
const _Mu = "Mutable";
const _N = "Notify";
const _NAC = "NumberAttributeConstraints";
const _NACT = "NumberAttributeConstraintsType";
const _NAE = "NotAuthorizedException";
const _NAEo = "NoActionEmail";
const _NC = "NotifyConfiguration";
const _NCT = "NotifyConfigurationType";
const _NDM = "NewDeviceMetadata";
const _NDMT = "NewDeviceMetadataType";
const _NET = "NotifyEmailType";
const _NT = "NextToken";
const _Na = "Name";
const _ONEE = "OperationNotEnabledException";
const _P = "Password";
const _PA = "PreAuthentication";
const _PAN = "ProviderAttributeName";
const _PAV = "ProviderAttributeValue";
const _PAo = "PostAuthentication";
const _PC = "PostConfirmation";
const _PD = "ProviderDetails";
const _PDr = "ProviderDescription";
const _PHPVE = "PasswordHistoryPolicyViolationException";
const _PHS = "PasswordHistorySize";
const _PLT = "ProvidersListType";
const _PM = "PreferredMfa";
const _PMS = "PreferredMfaSetting";
const _PN = "ProviderName";
const _PNME = "PreconditionNotMetException";
const _PNo = "PoolName";
const _PP = "ProposedPassword";
const _PPT = "PasswordPolicyType";
const _PPa = "PasswordPolicy";
const _PPr = "PreviousPassword";
const _PRHCI = "PrimaryRoute53HealthCheckId";
const _PRRE = "PasswordResetRequiredException";
const _PSU = "PreSignUp";
const _PSUr = "PreSignedUrl";
const _PT = "PasswordType";
const _PTG = "PreTokenGeneration";
const _PTGC = "PreTokenGenerationConfig";
const _PTGVCT = "PreTokenGenerationVersionConfigType";
const _PTa = "PaginationToken";
const _PTr = "ProviderType";
const _PUEE = "PreventUserExistenceErrors";
const _PUIT = "ProviderUserIdentifierType";
const _PV = "PasswordVerifier";
const _Pe = "Permanent";
const _Po = "Policies";
const _Pr = "Precedence";
const _Pri = "Priority";
const _Pro = "Provider";
const _Prov = "Providers";
const _R = "Routing";
const _RA = "RoleArn";
const _RAe = "ReadAttributes";
const _RAes = "ResourceArn";
const _RC = "RiskConfiguration";
const _RCC = "ResendConfirmationCode";
const _RCCR = "ResendConfirmationCodeRequest";
const _RCCRe = "ResendConfirmationCodeResponse";
const _RCT = "RiskConfigurationType";
const _RD = "RiskDecision";
const _REC = "RiskExceptionConfiguration";
const _RECT = "RiskExceptionConfigurationType";
const _RGPS = "RetryGracePeriodSeconds";
const _RI = "ResourceId";
const _RL = "RiskLevel";
const _RLe = "RequireLowercase";
const _RM = "RecoveryMechanisms";
const _RMR = "ReturnMergedResources";
const _RMT = "RecoveryMechanismsType";
const _RN = "RegionName";
const _RNFE = "ResourceNotFoundException";
const _RNe = "RequireNumbers";
const _ROT = "RecoveryOptionType";
const _RPI = "RelyingPartyId";
const _RR = "ReplicaRegions";
const _RS = "ResourceServer";
const _RSLT = "ResourceServersListType";
const _RSSLT = "ResourceServerScopeListType";
const _RSST = "ResourceServerScopeType";
const _RST = "ResourceServerType";
const _RSe = "ResourceServers";
const _RSeq = "RequireSymbols";
const _RT = "RefreshToken";
const _RTAC = "RespondToAuthChallenge";
const _RTACR = "RespondToAuthChallengeRequest";
const _RTACRe = "RespondToAuthChallengeResponse";
const _RTEA = "ReplyToEmailAddress";
const _RTR = "RefreshTokenRotation";
const _RTRE = "RefreshTokenReuseException";
const _RTRT = "RefreshTokenRotationType";
const _RTRe = "RevokeTokenRequest";
const _RTRev = "RevokeTokenResponse";
const _RTV = "RefreshTokenValidity";
const _RTe = "ReplyTo";
const _RTev = "RevokeToken";
const _RTo = "RoutingType";
const _RU = "RequireUppercase";
const _Re = "Required";
const _Ro = "Role";
const _S = "Session";
const _SA = "SourceArn";
const _SAC = "StringAttributeConstraints";
const _SACT = "StringAttributeConstraintsType";
const _SALT = "SchemaAttributesListType";
const _SAM = "SmsAuthenticationMessage";
const _SAT = "SchemaAttributeType";
const _SAc = "SchemaAttributes";
const _SAt = "StreamArn";
const _SB = "S3Bucket";
const _SC = "SecretCode";
const _SCA = "SnsCallerArn";
const _SCF = "SmsConfigurationFailure";
const _SCT = "SecretCodeType";
const _SCTm = "SmsConfigurationType";
const _SCTo = "S3ConfigurationType";
const _SCm = "SmsConfiguration";
const _SCo = "S3Configuration";
const _SD = "ScopeDescription";
const _SDNEE = "ScopeDoesNotExistException";
const _SDt = "StartDate";
const _SH = "SecretHash";
const _SHT = "SecretHashType";
const _SIP = "SupportedIdentityProviders";
const _SIPRL = "SkippedIPRangeList";
const _SIPT = "SignInPolicyType";
const _SIPi = "SignInPolicy";
const _SLDC = "SetLogDeliveryConfiguration";
const _SLDCR = "SetLogDeliveryConfigurationRequest";
const _SLDCRe = "SetLogDeliveryConfigurationResponse";
const _SM = "SmsMessage";
const _SMC = "SmsMfaConfiguration";
const _SMCT = "SmsMfaConfigType";
const _SMSM = "SMSMessage";
const _SMSMS = "SMSMfaSettings";
const _SMSMST = "SMSMfaSettingsType";
const _SN = "ServerName";
const _SNc = "ScopeName";
const _SP = "ServerPath";
const _SR = "SecondaryRegion";
const _SRC = "SetRiskConfiguration";
const _SRCR = "SetRiskConfigurationRequest";
const _SRCRe = "SetRiskConfigurationResponse";
const _SRn = "SnsRegion";
const _ST = "SessionType";
const _STMC = "SoftwareTokenMfaConfiguration";
const _STMCT = "SoftwareTokenMfaConfigType";
const _STMFANFE = "SoftwareTokenMFANotFoundException";
const _STMFAUCT = "SoftwareTokenMFAUserCodeType";
const _STMS = "SoftwareTokenMfaSettings";
const _STMST = "SoftwareTokenMfaSettingsType";
const _SU = "SourceUser";
const _SUIC = "SetUICustomization";
const _SUICR = "SetUICustomizationRequest";
const _SUICRe = "SetUICustomizationResponse";
const _SUIJ = "StartUserImportJob";
const _SUIJR = "StartUserImportJobRequest";
const _SUIJRt = "StartUserImportJobResponse";
const _SUIJRto = "StopUserImportJobRequest";
const _SUIJRtop = "StopUserImportJobResponse";
const _SUIJt = "StopUserImportJob";
const _SUMFAP = "SetUserMFAPreference";
const _SUMFAPR = "SetUserMFAPreferenceRequest";
const _SUMFAPRe = "SetUserMFAPreferenceResponse";
const _SUPMC = "SetUserPoolMfaConfig";
const _SUPMCR = "SetUserPoolMfaConfigRequest";
const _SUPMCRe = "SetUserPoolMfaConfigResponse";
const _SUR = "SignUpRequest";
const _SURi = "SignUpResponse";
const _SUS = "SetUserSettings";
const _SUSR = "SetUserSettingsRequest";
const _SUSRe = "SetUserSettingsResponse";
const _SUi = "SignUp";
const _SUk = "SkippedUsers";
const _SVM = "SmsVerificationMessage";
const _SWAR = "StartWebAuthnRegistration";
const _SWARR = "StartWebAuthnRegistrationRequest";
const _SWARRt = "StartWebAuthnRegistrationResponse";
const _Sa = "Salt";
const _Sc = "Scopes";
const _Sch = "Schema";
const _Se = "Settings";
const _St = "Status";
const _Su = "Subject";
const _T = "Terms";
const _TB = "TextBody";
const _TCNAE = "TierChangeNotAllowedException";
const _TDLT = "TermsDescriptionListType";
const _TDT = "TermsDescriptionType";
const _TEE = "TermsExistsException";
const _TI = "TermsId";
const _TK = "TagKeys";
const _TMFAE = "TooManyFailedAttemptsException";
const _TMRE = "TooManyRequestsException";
const _TMT = "TokenModelType";
const _TN = "TermsName";
const _TP = "TemporaryPassword";
const _TPVD = "TemporaryPasswordValidityDays";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TS = "TermsSource";
const _TT = "TokenType";
const _TTe = "TermsType";
const _TVU = "TokenValidityUnits";
const _TVUT = "TokenValidityUnitsType";
const _Ta = "Tags";
const _Ti = "Timezone";
const _To = "Token";
const _Ty = "Type";
const _U = "Username";
const _UA = "UserAttributes";
const _UAEF = "UpdateAuthEventFeedback";
const _UAEFR = "UpdateAuthEventFeedbackRequest";
const _UAEFRp = "UpdateAuthEventFeedbackResponse";
const _UAN = "UserAttributeNames";
const _UAUS = "UserAttributeUpdateSettings";
const _UAUST = "UserAttributeUpdateSettingsType";
const _UAVD = "UnusedAccountValidityDays";
const _UAs = "UsernameAttributes";
const _UC = "UsernameConfiguration";
const _UCD = "UserCreateDate";
const _UCDT = "UserContextDataType";
const _UCDs = "UserContextData";
const _UCN = "UserConfirmationNecessary";
const _UCPV = "UseCognitoProvidedValues";
const _UCT = "UsernameConfigurationType";
const _UCs = "UserConfirmed";
const _UCse = "UserCode";
const _UDS = "UserDataShared";
const _UDSR = "UpdateDeviceStatusRequest";
const _UDSRp = "UpdateDeviceStatusResponse";
const _UDSp = "UpdateDeviceStatus";
const _UE = "UnauthorizedException";
const _UEE = "UsernameExistsException";
const _UG = "UpdateGroup";
const _UGR = "UpdateGroupRequest";
const _UGRp = "UpdateGroupResponse";
const _UIC = "UICustomization";
const _UICT = "UICustomizationType";
const _UIIPE = "UserImportInProgressException";
const _UIJ = "UserImportJob";
const _UIJLT = "UserImportJobsListType";
const _UIJT = "UserImportJobType";
const _UIJs = "UserImportJobs";
const _UIP = "UpdateIdentityProvider";
const _UIPE = "UnsupportedIdentityProviderException";
const _UIPR = "UpdateIdentityProviderRequest";
const _UIPRp = "UpdateIdentityProviderResponse";
const _ULE = "UnexpectedLambdaException";
const _ULMD = "UserLastModifiedDate";
const _ULT = "UsersListType";
const _ULVE = "UserLambdaValidationException";
const _UM = "UserMigration";
const _UMFASL = "UserMFASettingList";
const _UMLB = "UpdateManagedLoginBranding";
const _UMLBR = "UpdateManagedLoginBrandingRequest";
const _UMLBRp = "UpdateManagedLoginBrandingResponse";
const _UNCE = "UserNotConfirmedException";
const _UNFE = "UserNotFoundException";
const _UOE = "UnsupportedOperationException";
const _UP = "UserPool";
const _UPA = "UserPoolArn";
const _UPAO = "UserPoolAddOns";
const _UPAONEE = "UserPoolAddOnNotEnabledException";
const _UPAOT = "UserPoolAddOnsType";
const _UPC = "UserPoolClient";
const _UPCD = "UserPoolClientDescription";
const _UPCLT = "UserPoolClientListType";
const _UPCT = "UserPoolClientType";
const _UPCs = "UserPoolClients";
const _UPDT = "UserPoolDescriptionType";
const _UPI = "UserPoolId";
const _UPLT = "UserPoolListType";
const _UPPT = "UserPoolPolicyType";
const _UPR = "UserPoolReplica";
const _UPRLT = "UserPoolReplicaListType";
const _UPRT = "UserPoolReplicaType";
const _UPRs = "UserPoolReplicas";
const _UPT = "UserPoolTags";
const _UPTE = "UserPoolTaggingException";
const _UPTs = "UserPoolTier";
const _UPTse = "UserPoolType";
const _UPs = "UserPools";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _URS = "UpdateResourceServer";
const _URSR = "UpdateResourceServerRequest";
const _URSRp = "UpdateResourceServerResponse";
const _US = "UserStatus";
const _USs = "UserSub";
const _UT = "UsernameType";
const _UTR = "UpdateTermsRequest";
const _UTRp = "UpdateTermsResponse";
const _UTTE = "UnsupportedTokenTypeException";
const _UTp = "UpdateTerms";
const _UTs = "UserType";
const _UUA = "UpdateUserAttributes";
const _UUAR = "UpdateUserAttributesRequest";
const _UUARp = "UpdateUserAttributesResponse";
const _UUP = "UpdateUserPool";
const _UUPC = "UpdateUserPoolClient";
const _UUPCR = "UpdateUserPoolClientRequest";
const _UUPCRp = "UpdateUserPoolClientResponse";
const _UUPD = "UpdateUserPoolDomain";
const _UUPDR = "UpdateUserPoolDomainRequest";
const _UUPDRp = "UpdateUserPoolDomainResponse";
const _UUPR = "UpdateUserPoolRequest";
const _UUPRR = "UpdateUserPoolReplicaRequest";
const _UUPRRp = "UpdateUserPoolReplicaResponse";
const _UUPRp = "UpdateUserPoolResponse";
const _UUPRpd = "UpdateUserPoolReplica";
const _UUSE = "UnsupportedUserStateException";
const _UV = "UserVerification";
const _Us = "User";
const _Use = "Users";
const _V = "Value";
const _VACR = "VerifyAuthChallengeResponse";
const _VD = "ValidationData";
const _VMT = "VerificationMessageTemplate";
const _VMTT = "VerificationMessageTemplateType";
const _VST = "VerifySoftwareToken";
const _VSTR = "VerifySoftwareTokenRequest";
const _VSTRe = "VerifySoftwareTokenResponse";
const _VUA = "VerifyUserAttribute";
const _VUAR = "VerifyUserAttributeRequest";
const _VUARe = "VerifyUserAttributeResponse";
const _Ve = "Version";
const _WA = "WriteAttributes";
const _WAC = "WebAuthnConfiguration";
const _WACD = "WebAuthnCredentialDescription";
const _WACDLT = "WebAuthnCredentialDescriptionListType";
const _WACME = "WebAuthnClientMismatchException";
const _WACMEe = "WebAuthnConfigurationMissingException";
const _WACNFE = "WebAuthnChallengeNotFoundException";
const _WACNSE = "WebAuthnCredentialNotSupportedException";
const _WACT = "WebAuthnConfigurationType";
const _WAMS = "WebAuthnMfaSettings";
const _WAMST = "WebAuthnMfaSettingsType";
const _WANEE = "WebAuthnNotEnabledException";
const _WAONAE = "WebAuthnOriginNotAllowedException";
const _WARPME = "WebAuthnRelyingPartyMismatchException";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _hN = "headerName";
const _hV = "headerValue";
const _m = "message";
const _rC = "reasonCode";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.cognitoidentityprovider";
const _se = "server";
const n0 = "com.amazonaws.cognitoidentityprovider";
const { TypeRegistry } = require("@smithy/core/schema");
const { CognitoIdentityProviderServiceException } = require("../models/CognitoIdentityProviderServiceException");
const { AccessDeniedException, AliasExistsException, CodeDeliveryFailureException, CodeMismatchException, ConcurrentModificationException, DeviceKeyExistsException, DuplicateProviderException, EnableSoftwareTokenMFAException, ExpiredCodeException, FeatureUnavailableInTierException, ForbiddenException, GroupExistsException, InternalErrorException, InternalServerException, InvalidEmailRoleAccessPolicyException, InvalidLambdaResponseException, InvalidOAuthFlowException, InvalidParameterException, InvalidPasswordException, InvalidSmsRoleAccessPolicyException, InvalidSmsRoleTrustRelationshipException, InvalidUserPoolConfigurationException, LimitExceededException, ManagedLoginBrandingExistsException, MFAMethodNotFoundException, NotAuthorizedException, OperationNotEnabledException, PasswordHistoryPolicyViolationException, PasswordResetRequiredException, PreconditionNotMetException, RefreshTokenReuseException, ResourceNotFoundException, ScopeDoesNotExistException, SoftwareTokenMFANotFoundException, TermsExistsException, TierChangeNotAllowedException, TooManyFailedAttemptsException, TooManyRequestsException, UnauthorizedException, UnexpectedLambdaException, UnsupportedIdentityProviderException, UnsupportedOperationException, UnsupportedTokenTypeException, UnsupportedUserStateException, UserImportInProgressException, UserLambdaValidationException, UsernameExistsException, UserNotConfirmedException, UserNotFoundException, UserPoolAddOnNotEnabledException, UserPoolTaggingException, WebAuthnChallengeNotFoundException, WebAuthnClientMismatchException, WebAuthnConfigurationMissingException, WebAuthnCredentialNotSupportedException, WebAuthnNotEnabledException, WebAuthnOriginNotAllowedException, WebAuthnRelyingPartyMismatchException } = require("../models/errors");
const _s_registry = TypeRegistry.for(_s);
const CognitoIdentityProviderServiceException$ = [-3, _s, "CognitoIdentityProviderServiceException", 0, [], []];
exports.CognitoIdentityProviderServiceException$ = CognitoIdentityProviderServiceException$;
_s_registry.registerError(CognitoIdentityProviderServiceException$, CognitoIdentityProviderServiceException);
const n0_registry = TypeRegistry.for(n0);
const AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
exports.AccessDeniedException$ = AccessDeniedException$;
n0_registry.registerError(AccessDeniedException$, AccessDeniedException);
const AliasExistsException$ = [-3, n0, _AEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.AliasExistsException$ = AliasExistsException$;
n0_registry.registerError(AliasExistsException$, AliasExistsException);
const CodeDeliveryFailureException$ = [-3, n0, _CDFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.CodeDeliveryFailureException$ = CodeDeliveryFailureException$;
n0_registry.registerError(CodeDeliveryFailureException$, CodeDeliveryFailureException);
const CodeMismatchException$ = [-3, n0, _CME,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.CodeMismatchException$ = CodeMismatchException$;
n0_registry.registerError(CodeMismatchException$, CodeMismatchException);
const ConcurrentModificationException$ = [-3, n0, _CMEo,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.ConcurrentModificationException$ = ConcurrentModificationException$;
n0_registry.registerError(ConcurrentModificationException$, ConcurrentModificationException);
const DeviceKeyExistsException$ = [-3, n0, _DKEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.DeviceKeyExistsException$ = DeviceKeyExistsException$;
n0_registry.registerError(DeviceKeyExistsException$, DeviceKeyExistsException);
const DuplicateProviderException$ = [-3, n0, _DPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.DuplicateProviderException$ = DuplicateProviderException$;
n0_registry.registerError(DuplicateProviderException$, DuplicateProviderException);
const EnableSoftwareTokenMFAException$ = [-3, n0, _ESTMFAE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.EnableSoftwareTokenMFAException$ = EnableSoftwareTokenMFAException$;
n0_registry.registerError(EnableSoftwareTokenMFAException$, EnableSoftwareTokenMFAException);
const ExpiredCodeException$ = [-3, n0, _ECE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.ExpiredCodeException$ = ExpiredCodeException$;
n0_registry.registerError(ExpiredCodeException$, ExpiredCodeException);
const FeatureUnavailableInTierException$ = [-3, n0, _FUITE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
exports.FeatureUnavailableInTierException$ = FeatureUnavailableInTierException$;
n0_registry.registerError(FeatureUnavailableInTierException$, FeatureUnavailableInTierException);
const ForbiddenException$ = [-3, n0, _FE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
exports.ForbiddenException$ = ForbiddenException$;
n0_registry.registerError(ForbiddenException$, ForbiddenException);
const GroupExistsException$ = [-3, n0, _GEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.GroupExistsException$ = GroupExistsException$;
n0_registry.registerError(GroupExistsException$, GroupExistsException);
const InternalErrorException$ = [-3, n0, _IEE,
    { [_e]: _se },
    [_m],
    [0]
];
exports.InternalErrorException$ = InternalErrorException$;
n0_registry.registerError(InternalErrorException$, InternalErrorException);
const InternalServerException$ = [-3, n0, _ISE,
    { [_e]: _se },
    [_m],
    [0]
];
exports.InternalServerException$ = InternalServerException$;
n0_registry.registerError(InternalServerException$, InternalServerException);
const InvalidEmailRoleAccessPolicyException$ = [-3, n0, _IERAPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidEmailRoleAccessPolicyException$ = InvalidEmailRoleAccessPolicyException$;
n0_registry.registerError(InvalidEmailRoleAccessPolicyException$, InvalidEmailRoleAccessPolicyException);
const InvalidLambdaResponseException$ = [-3, n0, _ILRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidLambdaResponseException$ = InvalidLambdaResponseException$;
n0_registry.registerError(InvalidLambdaResponseException$, InvalidLambdaResponseException);
const InvalidOAuthFlowException$ = [-3, n0, _IOAFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidOAuthFlowException$ = InvalidOAuthFlowException$;
n0_registry.registerError(InvalidOAuthFlowException$, InvalidOAuthFlowException);
const InvalidParameterException$ = [-3, n0, _IPE,
    { [_e]: _c, [_hE]: 400 },
    [_m, _rC],
    [0, 0]
];
exports.InvalidParameterException$ = InvalidParameterException$;
n0_registry.registerError(InvalidParameterException$, InvalidParameterException);
const InvalidPasswordException$ = [-3, n0, _IPEn,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidPasswordException$ = InvalidPasswordException$;
n0_registry.registerError(InvalidPasswordException$, InvalidPasswordException);
const InvalidSmsRoleAccessPolicyException$ = [-3, n0, _ISRAPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidSmsRoleAccessPolicyException$ = InvalidSmsRoleAccessPolicyException$;
n0_registry.registerError(InvalidSmsRoleAccessPolicyException$, InvalidSmsRoleAccessPolicyException);
const InvalidSmsRoleTrustRelationshipException$ = [-3, n0, _ISRTRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidSmsRoleTrustRelationshipException$ = InvalidSmsRoleTrustRelationshipException$;
n0_registry.registerError(InvalidSmsRoleTrustRelationshipException$, InvalidSmsRoleTrustRelationshipException);
const InvalidUserPoolConfigurationException$ = [-3, n0, _IUPCE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidUserPoolConfigurationException$ = InvalidUserPoolConfigurationException$;
n0_registry.registerError(InvalidUserPoolConfigurationException$, InvalidUserPoolConfigurationException);
const LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.LimitExceededException$ = LimitExceededException$;
n0_registry.registerError(LimitExceededException$, LimitExceededException);
const ManagedLoginBrandingExistsException$ = [-3, n0, _MLBEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.ManagedLoginBrandingExistsException$ = ManagedLoginBrandingExistsException$;
n0_registry.registerError(ManagedLoginBrandingExistsException$, ManagedLoginBrandingExistsException);
const MFAMethodNotFoundException$ = [-3, n0, _MFAMNFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.MFAMethodNotFoundException$ = MFAMethodNotFoundException$;
n0_registry.registerError(MFAMethodNotFoundException$, MFAMethodNotFoundException);
const NotAuthorizedException$ = [-3, n0, _NAE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
exports.NotAuthorizedException$ = NotAuthorizedException$;
n0_registry.registerError(NotAuthorizedException$, NotAuthorizedException);
const OperationNotEnabledException$ = [-3, n0, _ONEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.OperationNotEnabledException$ = OperationNotEnabledException$;
n0_registry.registerError(OperationNotEnabledException$, OperationNotEnabledException);
const PasswordHistoryPolicyViolationException$ = [-3, n0, _PHPVE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.PasswordHistoryPolicyViolationException$ = PasswordHistoryPolicyViolationException$;
n0_registry.registerError(PasswordHistoryPolicyViolationException$, PasswordHistoryPolicyViolationException);
const PasswordResetRequiredException$ = [-3, n0, _PRRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.PasswordResetRequiredException$ = PasswordResetRequiredException$;
n0_registry.registerError(PasswordResetRequiredException$, PasswordResetRequiredException);
const PreconditionNotMetException$ = [-3, n0, _PNME,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.PreconditionNotMetException$ = PreconditionNotMetException$;
n0_registry.registerError(PreconditionNotMetException$, PreconditionNotMetException);
const RefreshTokenReuseException$ = [-3, n0, _RTRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.RefreshTokenReuseException$ = RefreshTokenReuseException$;
n0_registry.registerError(RefreshTokenReuseException$, RefreshTokenReuseException);
const ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
const ScopeDoesNotExistException$ = [-3, n0, _SDNEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.ScopeDoesNotExistException$ = ScopeDoesNotExistException$;
n0_registry.registerError(ScopeDoesNotExistException$, ScopeDoesNotExistException);
const SoftwareTokenMFANotFoundException$ = [-3, n0, _STMFANFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.SoftwareTokenMFANotFoundException$ = SoftwareTokenMFANotFoundException$;
n0_registry.registerError(SoftwareTokenMFANotFoundException$, SoftwareTokenMFANotFoundException);
const TermsExistsException$ = [-3, n0, _TEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.TermsExistsException$ = TermsExistsException$;
n0_registry.registerError(TermsExistsException$, TermsExistsException);
const TierChangeNotAllowedException$ = [-3, n0, _TCNAE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
exports.TierChangeNotAllowedException$ = TierChangeNotAllowedException$;
n0_registry.registerError(TierChangeNotAllowedException$, TierChangeNotAllowedException);
const TooManyFailedAttemptsException$ = [-3, n0, _TMFAE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.TooManyFailedAttemptsException$ = TooManyFailedAttemptsException$;
n0_registry.registerError(TooManyFailedAttemptsException$, TooManyFailedAttemptsException);
const TooManyRequestsException$ = [-3, n0, _TMRE,
    { [_e]: _c, [_hE]: 429 },
    [_m],
    [0]
];
exports.TooManyRequestsException$ = TooManyRequestsException$;
n0_registry.registerError(TooManyRequestsException$, TooManyRequestsException);
const UnauthorizedException$ = [-3, n0, _UE,
    { [_e]: _c, [_hE]: 401 },
    [_m],
    [0]
];
exports.UnauthorizedException$ = UnauthorizedException$;
n0_registry.registerError(UnauthorizedException$, UnauthorizedException);
const UnexpectedLambdaException$ = [-3, n0, _ULE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UnexpectedLambdaException$ = UnexpectedLambdaException$;
n0_registry.registerError(UnexpectedLambdaException$, UnexpectedLambdaException);
const UnsupportedIdentityProviderException$ = [-3, n0, _UIPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UnsupportedIdentityProviderException$ = UnsupportedIdentityProviderException$;
n0_registry.registerError(UnsupportedIdentityProviderException$, UnsupportedIdentityProviderException);
const UnsupportedOperationException$ = [-3, n0, _UOE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UnsupportedOperationException$ = UnsupportedOperationException$;
n0_registry.registerError(UnsupportedOperationException$, UnsupportedOperationException);
const UnsupportedTokenTypeException$ = [-3, n0, _UTTE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UnsupportedTokenTypeException$ = UnsupportedTokenTypeException$;
n0_registry.registerError(UnsupportedTokenTypeException$, UnsupportedTokenTypeException);
const UnsupportedUserStateException$ = [-3, n0, _UUSE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UnsupportedUserStateException$ = UnsupportedUserStateException$;
n0_registry.registerError(UnsupportedUserStateException$, UnsupportedUserStateException);
const UserImportInProgressException$ = [-3, n0, _UIIPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UserImportInProgressException$ = UserImportInProgressException$;
n0_registry.registerError(UserImportInProgressException$, UserImportInProgressException);
const UserLambdaValidationException$ = [-3, n0, _ULVE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UserLambdaValidationException$ = UserLambdaValidationException$;
n0_registry.registerError(UserLambdaValidationException$, UserLambdaValidationException);
const UsernameExistsException$ = [-3, n0, _UEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UsernameExistsException$ = UsernameExistsException$;
n0_registry.registerError(UsernameExistsException$, UsernameExistsException);
const UserNotConfirmedException$ = [-3, n0, _UNCE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UserNotConfirmedException$ = UserNotConfirmedException$;
n0_registry.registerError(UserNotConfirmedException$, UserNotConfirmedException);
const UserNotFoundException$ = [-3, n0, _UNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
exports.UserNotFoundException$ = UserNotFoundException$;
n0_registry.registerError(UserNotFoundException$, UserNotFoundException);
const UserPoolAddOnNotEnabledException$ = [-3, n0, _UPAONEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UserPoolAddOnNotEnabledException$ = UserPoolAddOnNotEnabledException$;
n0_registry.registerError(UserPoolAddOnNotEnabledException$, UserPoolAddOnNotEnabledException);
const UserPoolTaggingException$ = [-3, n0, _UPTE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.UserPoolTaggingException$ = UserPoolTaggingException$;
n0_registry.registerError(UserPoolTaggingException$, UserPoolTaggingException);
const WebAuthnChallengeNotFoundException$ = [-3, n0, _WACNFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.WebAuthnChallengeNotFoundException$ = WebAuthnChallengeNotFoundException$;
n0_registry.registerError(WebAuthnChallengeNotFoundException$, WebAuthnChallengeNotFoundException);
const WebAuthnClientMismatchException$ = [-3, n0, _WACME,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.WebAuthnClientMismatchException$ = WebAuthnClientMismatchException$;
n0_registry.registerError(WebAuthnClientMismatchException$, WebAuthnClientMismatchException);
const WebAuthnConfigurationMissingException$ = [-3, n0, _WACMEe,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.WebAuthnConfigurationMissingException$ = WebAuthnConfigurationMissingException$;
n0_registry.registerError(WebAuthnConfigurationMissingException$, WebAuthnConfigurationMissingException);
const WebAuthnCredentialNotSupportedException$ = [-3, n0, _WACNSE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.WebAuthnCredentialNotSupportedException$ = WebAuthnCredentialNotSupportedException$;
n0_registry.registerError(WebAuthnCredentialNotSupportedException$, WebAuthnCredentialNotSupportedException);
const WebAuthnNotEnabledException$ = [-3, n0, _WANEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.WebAuthnNotEnabledException$ = WebAuthnNotEnabledException$;
n0_registry.registerError(WebAuthnNotEnabledException$, WebAuthnNotEnabledException);
const WebAuthnOriginNotAllowedException$ = [-3, n0, _WAONAE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.WebAuthnOriginNotAllowedException$ = WebAuthnOriginNotAllowedException$;
n0_registry.registerError(WebAuthnOriginNotAllowedException$, WebAuthnOriginNotAllowedException);
const WebAuthnRelyingPartyMismatchException$ = [-3, n0, _WARPME,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.WebAuthnRelyingPartyMismatchException$ = WebAuthnRelyingPartyMismatchException$;
n0_registry.registerError(WebAuthnRelyingPartyMismatchException$, WebAuthnRelyingPartyMismatchException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AttributeValueType = [0, n0, _AVT, 8, 0];
var ClientIdType = [0, n0, _CIT, 8, 0];
var ClientSecretType = [0, n0, _CST, 8, 0];
var PasswordType = [0, n0, _PT, 8, 0];
var SecretCodeType = [0, n0, _SCT, 8, 0];
var SecretHashType = [0, n0, _SHT, 8, 0];
var SessionType = [0, n0, _ST, 8, 0];
var SoftwareTokenMFAUserCodeType = [0, n0, _STMFAUCT, 8, 0];
var TokenModelType = [0, n0, _TMT, 8, 0];
var UsernameType = [0, n0, _UT, 8, 0];
const AccountRecoverySettingType$ = [3, n0, _ARST,
    0,
    [_RM],
    [() => RecoveryMechanismsType]
];
exports.AccountRecoverySettingType$ = AccountRecoverySettingType$;
const AccountTakeoverActionsType$ = [3, n0, _ATAT,
    0,
    [_LA, _MA, _HA],
    [() => AccountTakeoverActionType$, () => AccountTakeoverActionType$, () => AccountTakeoverActionType$]
];
exports.AccountTakeoverActionsType$ = AccountTakeoverActionsType$;
const AccountTakeoverActionType$ = [3, n0, _ATATc,
    0,
    [_N, _EA],
    [2, 0], 2
];
exports.AccountTakeoverActionType$ = AccountTakeoverActionType$;
const AccountTakeoverRiskConfigurationType$ = [3, n0, _ATRCT,
    0,
    [_A, _NC],
    [() => AccountTakeoverActionsType$, () => NotifyConfigurationType$], 1
];
exports.AccountTakeoverRiskConfigurationType$ = AccountTakeoverRiskConfigurationType$;
const AddCustomAttributesRequest$ = [3, n0, _ACAR,
    0,
    [_UPI, _CA],
    [0, () => CustomAttributesListType], 2
];
exports.AddCustomAttributesRequest$ = AddCustomAttributesRequest$;
const AddCustomAttributesResponse$ = [3, n0, _ACARd,
    0,
    [],
    []
];
exports.AddCustomAttributesResponse$ = AddCustomAttributesResponse$;
const AddUserPoolClientSecretRequest$ = [3, n0, _AUPCSR,
    0,
    [_UPI, _CI, _CS],
    [0, [() => ClientIdType, 0], [() => ClientSecretType, 0]], 2
];
exports.AddUserPoolClientSecretRequest$ = AddUserPoolClientSecretRequest$;
const AddUserPoolClientSecretResponse$ = [3, n0, _AUPCSRd,
    0,
    [_CSD],
    [[() => ClientSecretDescriptorType$, 0]]
];
exports.AddUserPoolClientSecretResponse$ = AddUserPoolClientSecretResponse$;
const AdminAddUserToGroupRequest$ = [3, n0, _AAUTGR,
    0,
    [_UPI, _U, _GN],
    [0, [() => UsernameType, 0], 0], 3
];
exports.AdminAddUserToGroupRequest$ = AdminAddUserToGroupRequest$;
const AdminConfirmSignUpRequest$ = [3, n0, _ACSUR,
    0,
    [_UPI, _U, _CM],
    [0, [() => UsernameType, 0], 128 | 0], 2
];
exports.AdminConfirmSignUpRequest$ = AdminConfirmSignUpRequest$;
const AdminConfirmSignUpResponse$ = [3, n0, _ACSURd,
    0,
    [],
    []
];
exports.AdminConfirmSignUpResponse$ = AdminConfirmSignUpResponse$;
const AdminCreateUserConfigType$ = [3, n0, _ACUCT,
    0,
    [_AACUO, _UAVD, _IMT],
    [2, 1, () => MessageTemplateType$]
];
exports.AdminCreateUserConfigType$ = AdminCreateUserConfigType$;
const AdminCreateUserRequest$ = [3, n0, _ACUR,
    0,
    [_UPI, _U, _UA, _VD, _TP, _FAC, _MAe, _DDM, _CM],
    [0, [() => UsernameType, 0], [() => AttributeListType, 0], [() => AttributeListType, 0], [() => PasswordType, 0], 2, 0, 64 | 0, 128 | 0], 2
];
exports.AdminCreateUserRequest$ = AdminCreateUserRequest$;
const AdminCreateUserResponse$ = [3, n0, _ACURd,
    0,
    [_Us],
    [[() => UserType$, 0]]
];
exports.AdminCreateUserResponse$ = AdminCreateUserResponse$;
const AdminDeleteUserAttributesRequest$ = [3, n0, _ADUAR,
    0,
    [_UPI, _U, _UAN],
    [0, [() => UsernameType, 0], 64 | 0], 3
];
exports.AdminDeleteUserAttributesRequest$ = AdminDeleteUserAttributesRequest$;
const AdminDeleteUserAttributesResponse$ = [3, n0, _ADUARd,
    0,
    [],
    []
];
exports.AdminDeleteUserAttributesResponse$ = AdminDeleteUserAttributesResponse$;
const AdminDeleteUserRequest$ = [3, n0, _ADUR,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminDeleteUserRequest$ = AdminDeleteUserRequest$;
const AdminDisableProviderForUserRequest$ = [3, n0, _ADPFUR,
    0,
    [_UPI, _Us],
    [0, () => ProviderUserIdentifierType$], 2
];
exports.AdminDisableProviderForUserRequest$ = AdminDisableProviderForUserRequest$;
const AdminDisableProviderForUserResponse$ = [3, n0, _ADPFURd,
    0,
    [],
    []
];
exports.AdminDisableProviderForUserResponse$ = AdminDisableProviderForUserResponse$;
const AdminDisableUserRequest$ = [3, n0, _ADURd,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminDisableUserRequest$ = AdminDisableUserRequest$;
const AdminDisableUserResponse$ = [3, n0, _ADURdm,
    0,
    [],
    []
];
exports.AdminDisableUserResponse$ = AdminDisableUserResponse$;
const AdminEnableUserRequest$ = [3, n0, _AEUR,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminEnableUserRequest$ = AdminEnableUserRequest$;
const AdminEnableUserResponse$ = [3, n0, _AEURd,
    0,
    [],
    []
];
exports.AdminEnableUserResponse$ = AdminEnableUserResponse$;
const AdminForgetDeviceRequest$ = [3, n0, _AFDR,
    0,
    [_UPI, _U, _DK],
    [0, [() => UsernameType, 0], 0], 3
];
exports.AdminForgetDeviceRequest$ = AdminForgetDeviceRequest$;
const AdminGetDeviceRequest$ = [3, n0, _AGDR,
    0,
    [_DK, _UPI, _U],
    [0, 0, [() => UsernameType, 0]], 3
];
exports.AdminGetDeviceRequest$ = AdminGetDeviceRequest$;
const AdminGetDeviceResponse$ = [3, n0, _AGDRd,
    0,
    [_D],
    [[() => DeviceType$, 0]], 1
];
exports.AdminGetDeviceResponse$ = AdminGetDeviceResponse$;
const AdminGetUserRequest$ = [3, n0, _AGUR,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminGetUserRequest$ = AdminGetUserRequest$;
const AdminGetUserResponse$ = [3, n0, _AGURd,
    0,
    [_U, _UA, _UCD, _ULMD, _E, _US, _MFAO, _PMS, _UMFASL],
    [[() => UsernameType, 0], [() => AttributeListType, 0], 4, 4, 2, 0, () => MFAOptionListType, 0, 64 | 0], 1
];
exports.AdminGetUserResponse$ = AdminGetUserResponse$;
const AdminInitiateAuthRequest$ = [3, n0, _AIAR,
    0,
    [_UPI, _CI, _AF, _AP, _CM, _AM, _CD, _S],
    [0, [() => ClientIdType, 0], 0, [() => AuthParametersType, 0], 128 | 0, () => AnalyticsMetadataType$, () => ContextDataType$, [() => SessionType, 0]], 3
];
exports.AdminInitiateAuthRequest$ = AdminInitiateAuthRequest$;
const AdminInitiateAuthResponse$ = [3, n0, _AIARd,
    0,
    [_CN, _S, _CP, _AR, _AC],
    [0, [() => SessionType, 0], 128 | 0, [() => AuthenticationResultType$, 0], 64 | 0]
];
exports.AdminInitiateAuthResponse$ = AdminInitiateAuthResponse$;
const AdminLinkProviderForUserRequest$ = [3, n0, _ALPFUR,
    0,
    [_UPI, _DU, _SU],
    [0, () => ProviderUserIdentifierType$, () => ProviderUserIdentifierType$], 3
];
exports.AdminLinkProviderForUserRequest$ = AdminLinkProviderForUserRequest$;
const AdminLinkProviderForUserResponse$ = [3, n0, _ALPFURd,
    0,
    [],
    []
];
exports.AdminLinkProviderForUserResponse$ = AdminLinkProviderForUserResponse$;
const AdminListDevicesRequest$ = [3, n0, _ALDR,
    0,
    [_UPI, _U, _L, _PTa],
    [0, [() => UsernameType, 0], 1, 0], 2
];
exports.AdminListDevicesRequest$ = AdminListDevicesRequest$;
const AdminListDevicesResponse$ = [3, n0, _ALDRd,
    0,
    [_De, _PTa],
    [[() => DeviceListType, 0], 0]
];
exports.AdminListDevicesResponse$ = AdminListDevicesResponse$;
const AdminListGroupsForUserRequest$ = [3, n0, _ALGFUR,
    0,
    [_U, _UPI, _L, _NT],
    [[() => UsernameType, 0], 0, 1, 0], 2
];
exports.AdminListGroupsForUserRequest$ = AdminListGroupsForUserRequest$;
const AdminListGroupsForUserResponse$ = [3, n0, _ALGFURd,
    0,
    [_G, _NT],
    [() => GroupListType, 0]
];
exports.AdminListGroupsForUserResponse$ = AdminListGroupsForUserResponse$;
const AdminListUserAuthEventsRequest$ = [3, n0, _ALUAER,
    0,
    [_UPI, _U, _MR, _NT],
    [0, [() => UsernameType, 0], 1, 0], 2
];
exports.AdminListUserAuthEventsRequest$ = AdminListUserAuthEventsRequest$;
const AdminListUserAuthEventsResponse$ = [3, n0, _ALUAERd,
    0,
    [_AE, _NT],
    [() => AuthEventsType, 0]
];
exports.AdminListUserAuthEventsResponse$ = AdminListUserAuthEventsResponse$;
const AdminRemoveUserFromGroupRequest$ = [3, n0, _ARUFGR,
    0,
    [_UPI, _U, _GN],
    [0, [() => UsernameType, 0], 0], 3
];
exports.AdminRemoveUserFromGroupRequest$ = AdminRemoveUserFromGroupRequest$;
const AdminResetUserPasswordRequest$ = [3, n0, _ARUPR,
    0,
    [_UPI, _U, _CM],
    [0, [() => UsernameType, 0], 128 | 0], 2
];
exports.AdminResetUserPasswordRequest$ = AdminResetUserPasswordRequest$;
const AdminResetUserPasswordResponse$ = [3, n0, _ARUPRd,
    0,
    [],
    []
];
exports.AdminResetUserPasswordResponse$ = AdminResetUserPasswordResponse$;
const AdminRespondToAuthChallengeRequest$ = [3, n0, _ARTACR,
    0,
    [_UPI, _CI, _CN, _CR, _S, _AM, _CD, _CM],
    [0, [() => ClientIdType, 0], 0, [() => ChallengeResponsesType, 0], [() => SessionType, 0], () => AnalyticsMetadataType$, () => ContextDataType$, 128 | 0], 3
];
exports.AdminRespondToAuthChallengeRequest$ = AdminRespondToAuthChallengeRequest$;
const AdminRespondToAuthChallengeResponse$ = [3, n0, _ARTACRd,
    0,
    [_CN, _S, _CP, _AR],
    [0, [() => SessionType, 0], 128 | 0, [() => AuthenticationResultType$, 0]]
];
exports.AdminRespondToAuthChallengeResponse$ = AdminRespondToAuthChallengeResponse$;
const AdminSetUserMFAPreferenceRequest$ = [3, n0, _ASUMFAPR,
    0,
    [_U, _UPI, _SMSMS, _STMS, _EMS, _WAMS],
    [[() => UsernameType, 0], 0, () => SMSMfaSettingsType$, () => SoftwareTokenMfaSettingsType$, () => EmailMfaSettingsType$, () => WebAuthnMfaSettingsType$], 2
];
exports.AdminSetUserMFAPreferenceRequest$ = AdminSetUserMFAPreferenceRequest$;
const AdminSetUserMFAPreferenceResponse$ = [3, n0, _ASUMFAPRd,
    0,
    [],
    []
];
exports.AdminSetUserMFAPreferenceResponse$ = AdminSetUserMFAPreferenceResponse$;
const AdminSetUserPasswordRequest$ = [3, n0, _ASUPR,
    0,
    [_UPI, _U, _P, _Pe],
    [0, [() => UsernameType, 0], [() => PasswordType, 0], 2], 3
];
exports.AdminSetUserPasswordRequest$ = AdminSetUserPasswordRequest$;
const AdminSetUserPasswordResponse$ = [3, n0, _ASUPRd,
    0,
    [],
    []
];
exports.AdminSetUserPasswordResponse$ = AdminSetUserPasswordResponse$;
const AdminSetUserSettingsRequest$ = [3, n0, _ASUSR,
    0,
    [_UPI, _U, _MFAO],
    [0, [() => UsernameType, 0], () => MFAOptionListType], 3
];
exports.AdminSetUserSettingsRequest$ = AdminSetUserSettingsRequest$;
const AdminSetUserSettingsResponse$ = [3, n0, _ASUSRd,
    0,
    [],
    []
];
exports.AdminSetUserSettingsResponse$ = AdminSetUserSettingsResponse$;
const AdminUpdateAuthEventFeedbackRequest$ = [3, n0, _AUAEFR,
    0,
    [_UPI, _U, _EI, _FV],
    [0, [() => UsernameType, 0], 0, 0], 4
];
exports.AdminUpdateAuthEventFeedbackRequest$ = AdminUpdateAuthEventFeedbackRequest$;
const AdminUpdateAuthEventFeedbackResponse$ = [3, n0, _AUAEFRd,
    0,
    [],
    []
];
exports.AdminUpdateAuthEventFeedbackResponse$ = AdminUpdateAuthEventFeedbackResponse$;
const AdminUpdateDeviceStatusRequest$ = [3, n0, _AUDSR,
    0,
    [_UPI, _U, _DK, _DRS],
    [0, [() => UsernameType, 0], 0, 0], 3
];
exports.AdminUpdateDeviceStatusRequest$ = AdminUpdateDeviceStatusRequest$;
const AdminUpdateDeviceStatusResponse$ = [3, n0, _AUDSRd,
    0,
    [],
    []
];
exports.AdminUpdateDeviceStatusResponse$ = AdminUpdateDeviceStatusResponse$;
const AdminUpdateUserAttributesRequest$ = [3, n0, _AUUAR,
    0,
    [_UPI, _U, _UA, _CM],
    [0, [() => UsernameType, 0], [() => AttributeListType, 0], 128 | 0], 3
];
exports.AdminUpdateUserAttributesRequest$ = AdminUpdateUserAttributesRequest$;
const AdminUpdateUserAttributesResponse$ = [3, n0, _AUUARd,
    0,
    [],
    []
];
exports.AdminUpdateUserAttributesResponse$ = AdminUpdateUserAttributesResponse$;
const AdminUserGlobalSignOutRequest$ = [3, n0, _AUGSOR,
    0,
    [_UPI, _U],
    [0, [() => UsernameType, 0]], 2
];
exports.AdminUserGlobalSignOutRequest$ = AdminUserGlobalSignOutRequest$;
const AdminUserGlobalSignOutResponse$ = [3, n0, _AUGSORd,
    0,
    [],
    []
];
exports.AdminUserGlobalSignOutResponse$ = AdminUserGlobalSignOutResponse$;
const AdvancedSecurityAdditionalFlowsType$ = [3, n0, _ASAFT,
    0,
    [_CAM],
    [0]
];
exports.AdvancedSecurityAdditionalFlowsType$ = AdvancedSecurityAdditionalFlowsType$;
const AnalyticsConfigurationType$ = [3, n0, _ACT,
    0,
    [_AI, _AA, _RA, _EIx, _UDS],
    [0, 0, 0, 0, 2]
];
exports.AnalyticsConfigurationType$ = AnalyticsConfigurationType$;
const AnalyticsMetadataType$ = [3, n0, _AMT,
    0,
    [_AEI],
    [0]
];
exports.AnalyticsMetadataType$ = AnalyticsMetadataType$;
const AssetType$ = [3, n0, _AT,
    0,
    [_C, _CMo, _Ex, _B, _RI],
    [0, 0, 0, 21, 0], 3
];
exports.AssetType$ = AssetType$;
const AssociateSoftwareTokenRequest$ = [3, n0, _ASTR,
    0,
    [_ATc, _S],
    [[() => TokenModelType, 0], [() => SessionType, 0]]
];
exports.AssociateSoftwareTokenRequest$ = AssociateSoftwareTokenRequest$;
const AssociateSoftwareTokenResponse$ = [3, n0, _ASTRs,
    0,
    [_SC, _S],
    [[() => SecretCodeType, 0], [() => SessionType, 0]]
];
exports.AssociateSoftwareTokenResponse$ = AssociateSoftwareTokenResponse$;
const AttributeType$ = [3, n0, _ATt,
    0,
    [_Na, _V],
    [0, [() => AttributeValueType, 0]], 1
];
exports.AttributeType$ = AttributeType$;
const AuthenticationResultType$ = [3, n0, _ART,
    0,
    [_ATc, _EIxp, _TT, _RT, _IT, _NDM],
    [[() => TokenModelType, 0], 1, 0, [() => TokenModelType, 0], [() => TokenModelType, 0], () => NewDeviceMetadataType$]
];
exports.AuthenticationResultType$ = AuthenticationResultType$;
const AuthEventType$ = [3, n0, _AET,
    0,
    [_EI, _ET, _CDr, _ER, _ERv, _CR, _ECD, _EF],
    [0, 0, 4, 0, () => EventRiskType$, () => ChallengeResponseListType, () => EventContextDataType$, () => EventFeedbackType$]
];
exports.AuthEventType$ = AuthEventType$;
const ChallengeResponseType$ = [3, n0, _CRT,
    0,
    [_CN, _CRh],
    [0, 0]
];
exports.ChallengeResponseType$ = ChallengeResponseType$;
const ChangePasswordRequest$ = [3, n0, _CPR,
    0,
    [_PP, _ATc, _PPr],
    [[() => PasswordType, 0], [() => TokenModelType, 0], [() => PasswordType, 0]], 2
];
exports.ChangePasswordRequest$ = ChangePasswordRequest$;
const ChangePasswordResponse$ = [3, n0, _CPRh,
    0,
    [],
    []
];
exports.ChangePasswordResponse$ = ChangePasswordResponse$;
const ClientSecretDescriptorType$ = [3, n0, _CSDT,
    0,
    [_CSI, _CSV, _CSCD],
    [0, [() => ClientSecretType, 0], 4]
];
exports.ClientSecretDescriptorType$ = ClientSecretDescriptorType$;
const CloudWatchLogsConfigurationType$ = [3, n0, _CWLCT,
    0,
    [_LGA],
    [0]
];
exports.CloudWatchLogsConfigurationType$ = CloudWatchLogsConfigurationType$;
const CodeDeliveryDetailsType$ = [3, n0, _CDDT,
    0,
    [_Des, _DM, _AN],
    [0, 0, 0]
];
exports.CodeDeliveryDetailsType$ = CodeDeliveryDetailsType$;
const CompleteWebAuthnRegistrationRequest$ = [3, n0, _CWARR,
    0,
    [_ATc, _Cr],
    [[() => TokenModelType, 0], 15], 2
];
exports.CompleteWebAuthnRegistrationRequest$ = CompleteWebAuthnRegistrationRequest$;
const CompleteWebAuthnRegistrationResponse$ = [3, n0, _CWARRo,
    0,
    [],
    []
];
exports.CompleteWebAuthnRegistrationResponse$ = CompleteWebAuthnRegistrationResponse$;
const CompromisedCredentialsActionsType$ = [3, n0, _CCAT,
    0,
    [_EA],
    [0], 1
];
exports.CompromisedCredentialsActionsType$ = CompromisedCredentialsActionsType$;
const CompromisedCredentialsRiskConfigurationType$ = [3, n0, _CCRCT,
    0,
    [_A, _EFv],
    [() => CompromisedCredentialsActionsType$, 64 | 0], 1
];
exports.CompromisedCredentialsRiskConfigurationType$ = CompromisedCredentialsRiskConfigurationType$;
const ConfirmDeviceRequest$ = [3, n0, _CDR,
    0,
    [_ATc, _DK, _DSVC, _DN],
    [[() => TokenModelType, 0], 0, () => DeviceSecretVerifierConfigType$, 0], 2
];
exports.ConfirmDeviceRequest$ = ConfirmDeviceRequest$;
const ConfirmDeviceResponse$ = [3, n0, _CDRo,
    0,
    [_UCN],
    [2]
];
exports.ConfirmDeviceResponse$ = ConfirmDeviceResponse$;
const ConfirmForgotPasswordRequest$ = [3, n0, _CFPR,
    0,
    [_CI, _U, _CC, _P, _SH, _AM, _UCDs, _CM],
    [[() => ClientIdType, 0], [() => UsernameType, 0], 0, [() => PasswordType, 0], [() => SecretHashType, 0], () => AnalyticsMetadataType$, [() => UserContextDataType$, 0], 128 | 0], 4
];
exports.ConfirmForgotPasswordRequest$ = ConfirmForgotPasswordRequest$;
const ConfirmForgotPasswordResponse$ = [3, n0, _CFPRo,
    0,
    [],
    []
];
exports.ConfirmForgotPasswordResponse$ = ConfirmForgotPasswordResponse$;
const ConfirmSignUpRequest$ = [3, n0, _CSUR,
    0,
    [_CI, _U, _CC, _SH, _FAC, _AM, _UCDs, _CM, _S],
    [[() => ClientIdType, 0], [() => UsernameType, 0], 0, [() => SecretHashType, 0], 2, () => AnalyticsMetadataType$, [() => UserContextDataType$, 0], 128 | 0, [() => SessionType, 0]], 3
];
exports.ConfirmSignUpRequest$ = ConfirmSignUpRequest$;
const ConfirmSignUpResponse$ = [3, n0, _CSURo,
    0,
    [_S],
    [[() => SessionType, 0]]
];
exports.ConfirmSignUpResponse$ = ConfirmSignUpResponse$;
const ContextDataType$ = [3, n0, _CDT,
    0,
    [_IA, _SN, _SP, _HH, _ED],
    [0, 0, 0, () => HttpHeaderList, 0], 4
];
exports.ContextDataType$ = ContextDataType$;
const CreateGroupRequest$ = [3, n0, _CGR,
    0,
    [_GN, _UPI, _Desc, _RA, _Pr],
    [0, 0, 0, 0, 1], 2
];
exports.CreateGroupRequest$ = CreateGroupRequest$;
const CreateGroupResponse$ = [3, n0, _CGRr,
    0,
    [_Gr],
    [() => GroupType$]
];
exports.CreateGroupResponse$ = CreateGroupResponse$;
const CreateIdentityProviderRequest$ = [3, n0, _CIPR,
    0,
    [_UPI, _PN, _PTr, _PD, _AMt, _II],
    [0, 0, 0, 128 | 0, 128 | 0, 64 | 0], 4
];
exports.CreateIdentityProviderRequest$ = CreateIdentityProviderRequest$;
const CreateIdentityProviderResponse$ = [3, n0, _CIPRr,
    0,
    [_IP],
    [() => IdentityProviderType$], 1
];
exports.CreateIdentityProviderResponse$ = CreateIdentityProviderResponse$;
const CreateManagedLoginBrandingRequest$ = [3, n0, _CMLBR,
    0,
    [_UPI, _CI, _UCPV, _Se, _As],
    [0, [() => ClientIdType, 0], 2, 15, () => AssetListType], 2
];
exports.CreateManagedLoginBrandingRequest$ = CreateManagedLoginBrandingRequest$;
const CreateManagedLoginBrandingResponse$ = [3, n0, _CMLBRr,
    0,
    [_MLB],
    [() => ManagedLoginBrandingType$]
];
exports.CreateManagedLoginBrandingResponse$ = CreateManagedLoginBrandingResponse$;
const CreateResourceServerRequest$ = [3, n0, _CRSR,
    0,
    [_UPI, _I, _Na, _Sc],
    [0, 0, 0, () => ResourceServerScopeListType], 3
];
exports.CreateResourceServerRequest$ = CreateResourceServerRequest$;
const CreateResourceServerResponse$ = [3, n0, _CRSRr,
    0,
    [_RS],
    [() => ResourceServerType$], 1
];
exports.CreateResourceServerResponse$ = CreateResourceServerResponse$;
const CreateTermsRequest$ = [3, n0, _CTR,
    0,
    [_UPI, _CI, _TN, _TS, _En, _Li],
    [0, [() => ClientIdType, 0], 0, 0, 0, 128 | 0], 5
];
exports.CreateTermsRequest$ = CreateTermsRequest$;
const CreateTermsResponse$ = [3, n0, _CTRr,
    0,
    [_T],
    [[() => TermsType$, 0]]
];
exports.CreateTermsResponse$ = CreateTermsResponse$;
const CreateUserImportJobRequest$ = [3, n0, _CUIJR,
    0,
    [_JN, _UPI, _CWLRA],
    [0, 0, 0], 3
];
exports.CreateUserImportJobRequest$ = CreateUserImportJobRequest$;
const CreateUserImportJobResponse$ = [3, n0, _CUIJRr,
    0,
    [_UIJ],
    [() => UserImportJobType$]
];
exports.CreateUserImportJobResponse$ = CreateUserImportJobResponse$;
const CreateUserPoolClientRequest$ = [3, n0, _CUPCR,
    0,
    [_UPI, _CNl, _GS, _CS, _RTV, _ATV, _ITV, _TVU, _RAe, _WA, _EAF, _SIP, _CURL, _LURL, _DRURI, _AOAF, _AOAS, _AOAFUPC, _ACn, _PUEE, _ETR, _EPAUCD, _ASV, _RTR],
    [0, 0, 2, [() => ClientSecretType, 0], 1, 1, 1, () => TokenValidityUnitsType$, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 0, 64 | 0, 64 | 0, 2, () => AnalyticsConfigurationType$, 0, 2, 2, 1, () => RefreshTokenRotationType$], 2
];
exports.CreateUserPoolClientRequest$ = CreateUserPoolClientRequest$;
const CreateUserPoolClientResponse$ = [3, n0, _CUPCRr,
    0,
    [_UPC],
    [[() => UserPoolClientType$, 0]]
];
exports.CreateUserPoolClientResponse$ = CreateUserPoolClientResponse$;
const CreateUserPoolDomainRequest$ = [3, n0, _CUPDR,
    0,
    [_Do, _UPI, _MLV, _CDC, _R],
    [0, 0, 1, () => CustomDomainConfigType$, () => RoutingType$], 2
];
exports.CreateUserPoolDomainRequest$ = CreateUserPoolDomainRequest$;
const CreateUserPoolDomainResponse$ = [3, n0, _CUPDRr,
    0,
    [_MLV, _CFD, _R],
    [1, 0, () => RoutingType$]
];
exports.CreateUserPoolDomainResponse$ = CreateUserPoolDomainResponse$;
const CreateUserPoolReplicaRequest$ = [3, n0, _CUPRR,
    0,
    [_UPI, _RN, _UPT],
    [0, 0, 128 | 0], 2
];
exports.CreateUserPoolReplicaRequest$ = CreateUserPoolReplicaRequest$;
const CreateUserPoolReplicaResponse$ = [3, n0, _CUPRRr,
    0,
    [_UPR],
    [() => UserPoolReplicaType$]
];
exports.CreateUserPoolReplicaResponse$ = CreateUserPoolReplicaResponse$;
const CreateUserPoolRequest$ = [3, n0, _CUPR,
    0,
    [_PNo, _Po, _DP, _LC, _AVA, _AAl, _UAs, _SVM, _EVM, _EVS, _VMT, _SAM, _MC, _UAUS, _DC, _EC, _SCm, _UPT, _ACUC, _Sch, _UPAO, _UC, _ARS, _UPTs, _KC, _IC],
    [0, () => UserPoolPolicyType$, 0, () => LambdaConfigType$, 64 | 0, 64 | 0, 64 | 0, 0, 0, 0, () => VerificationMessageTemplateType$, 0, 0, () => UserAttributeUpdateSettingsType$, () => DeviceConfigurationType$, () => EmailConfigurationType$, () => SmsConfigurationType$, 128 | 0, () => AdminCreateUserConfigType$, () => SchemaAttributesListType, () => UserPoolAddOnsType$, () => UsernameConfigurationType$, () => AccountRecoverySettingType$, 0, () => KeyConfigurationType$, () => IssuerConfigurationType$], 1
];
exports.CreateUserPoolRequest$ = CreateUserPoolRequest$;
const CreateUserPoolResponse$ = [3, n0, _CUPRr,
    0,
    [_UP],
    [() => UserPoolType$]
];
exports.CreateUserPoolResponse$ = CreateUserPoolResponse$;
const CustomDomainConfigType$ = [3, n0, _CDCT,
    0,
    [_CAe],
    [0], 1
];
exports.CustomDomainConfigType$ = CustomDomainConfigType$;
const CustomEmailLambdaVersionConfigType$ = [3, n0, _CELVCT,
    0,
    [_LV, _LAa],
    [0, 0], 2
];
exports.CustomEmailLambdaVersionConfigType$ = CustomEmailLambdaVersionConfigType$;
const CustomSMSLambdaVersionConfigType$ = [3, n0, _CSMSLVCT,
    0,
    [_LV, _LAa],
    [0, 0], 2
];
exports.CustomSMSLambdaVersionConfigType$ = CustomSMSLambdaVersionConfigType$;
const DeleteGroupRequest$ = [3, n0, _DGR,
    0,
    [_GN, _UPI],
    [0, 0], 2
];
exports.DeleteGroupRequest$ = DeleteGroupRequest$;
const DeleteIdentityProviderRequest$ = [3, n0, _DIPR,
    0,
    [_UPI, _PN],
    [0, 0], 2
];
exports.DeleteIdentityProviderRequest$ = DeleteIdentityProviderRequest$;
const DeleteManagedLoginBrandingRequest$ = [3, n0, _DMLBR,
    0,
    [_MLBI, _UPI],
    [0, 0], 2
];
exports.DeleteManagedLoginBrandingRequest$ = DeleteManagedLoginBrandingRequest$;
const DeleteResourceServerRequest$ = [3, n0, _DRSR,
    0,
    [_UPI, _I],
    [0, 0], 2
];
exports.DeleteResourceServerRequest$ = DeleteResourceServerRequest$;
const DeleteTermsRequest$ = [3, n0, _DTR,
    0,
    [_TI, _UPI],
    [0, 0], 2
];
exports.DeleteTermsRequest$ = DeleteTermsRequest$;
const DeleteUserAttributesRequest$ = [3, n0, _DUAR,
    0,
    [_UAN, _ATc],
    [64 | 0, [() => TokenModelType, 0]], 2
];
exports.DeleteUserAttributesRequest$ = DeleteUserAttributesRequest$;
const DeleteUserAttributesResponse$ = [3, n0, _DUARe,
    0,
    [],
    []
];
exports.DeleteUserAttributesResponse$ = DeleteUserAttributesResponse$;
const DeleteUserPoolClientRequest$ = [3, n0, _DUPCR,
    0,
    [_UPI, _CI],
    [0, [() => ClientIdType, 0]], 2
];
exports.DeleteUserPoolClientRequest$ = DeleteUserPoolClientRequest$;
const DeleteUserPoolClientSecretRequest$ = [3, n0, _DUPCSR,
    0,
    [_UPI, _CI, _CSI],
    [0, [() => ClientIdType, 0], 0], 3
];
exports.DeleteUserPoolClientSecretRequest$ = DeleteUserPoolClientSecretRequest$;
const DeleteUserPoolClientSecretResponse$ = [3, n0, _DUPCSRe,
    0,
    [],
    []
];
exports.DeleteUserPoolClientSecretResponse$ = DeleteUserPoolClientSecretResponse$;
const DeleteUserPoolDomainRequest$ = [3, n0, _DUPDR,
    0,
    [_Do, _UPI],
    [0, 0], 2
];
exports.DeleteUserPoolDomainRequest$ = DeleteUserPoolDomainRequest$;
const DeleteUserPoolDomainResponse$ = [3, n0, _DUPDRe,
    0,
    [],
    []
];
exports.DeleteUserPoolDomainResponse$ = DeleteUserPoolDomainResponse$;
const DeleteUserPoolReplicaRequest$ = [3, n0, _DUPRR,
    0,
    [_UPI, _RN],
    [0, 0], 2
];
exports.DeleteUserPoolReplicaRequest$ = DeleteUserPoolReplicaRequest$;
const DeleteUserPoolReplicaResponse$ = [3, n0, _DUPRRe,
    0,
    [_UPR],
    [() => UserPoolReplicaType$]
];
exports.DeleteUserPoolReplicaResponse$ = DeleteUserPoolReplicaResponse$;
const DeleteUserPoolRequest$ = [3, n0, _DUPR,
    0,
    [_UPI],
    [0], 1
];
exports.DeleteUserPoolRequest$ = DeleteUserPoolRequest$;
const DeleteUserRequest$ = [3, n0, _DUR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.DeleteUserRequest$ = DeleteUserRequest$;
const DeleteWebAuthnCredentialRequest$ = [3, n0, _DWACR,
    0,
    [_ATc, _CIr],
    [[() => TokenModelType, 0], 0], 2
];
exports.DeleteWebAuthnCredentialRequest$ = DeleteWebAuthnCredentialRequest$;
const DeleteWebAuthnCredentialResponse$ = [3, n0, _DWACRe,
    0,
    [],
    []
];
exports.DeleteWebAuthnCredentialResponse$ = DeleteWebAuthnCredentialResponse$;
const DescribeIdentityProviderRequest$ = [3, n0, _DIPRe,
    0,
    [_UPI, _PN],
    [0, 0], 2
];
exports.DescribeIdentityProviderRequest$ = DescribeIdentityProviderRequest$;
const DescribeIdentityProviderResponse$ = [3, n0, _DIPRes,
    0,
    [_IP],
    [() => IdentityProviderType$], 1
];
exports.DescribeIdentityProviderResponse$ = DescribeIdentityProviderResponse$;
const DescribeManagedLoginBrandingByClientRequest$ = [3, n0, _DMLBBCR,
    0,
    [_UPI, _CI, _RMR],
    [0, [() => ClientIdType, 0], 2], 2
];
exports.DescribeManagedLoginBrandingByClientRequest$ = DescribeManagedLoginBrandingByClientRequest$;
const DescribeManagedLoginBrandingByClientResponse$ = [3, n0, _DMLBBCRe,
    0,
    [_MLB],
    [() => ManagedLoginBrandingType$]
];
exports.DescribeManagedLoginBrandingByClientResponse$ = DescribeManagedLoginBrandingByClientResponse$;
const DescribeManagedLoginBrandingRequest$ = [3, n0, _DMLBRe,
    0,
    [_UPI, _MLBI, _RMR],
    [0, 0, 2], 2
];
exports.DescribeManagedLoginBrandingRequest$ = DescribeManagedLoginBrandingRequest$;
const DescribeManagedLoginBrandingResponse$ = [3, n0, _DMLBRes,
    0,
    [_MLB],
    [() => ManagedLoginBrandingType$]
];
exports.DescribeManagedLoginBrandingResponse$ = DescribeManagedLoginBrandingResponse$;
const DescribeResourceServerRequest$ = [3, n0, _DRSRe,
    0,
    [_UPI, _I],
    [0, 0], 2
];
exports.DescribeResourceServerRequest$ = DescribeResourceServerRequest$;
const DescribeResourceServerResponse$ = [3, n0, _DRSRes,
    0,
    [_RS],
    [() => ResourceServerType$], 1
];
exports.DescribeResourceServerResponse$ = DescribeResourceServerResponse$;
const DescribeRiskConfigurationRequest$ = [3, n0, _DRCR,
    0,
    [_UPI, _CI],
    [0, [() => ClientIdType, 0]], 1
];
exports.DescribeRiskConfigurationRequest$ = DescribeRiskConfigurationRequest$;
const DescribeRiskConfigurationResponse$ = [3, n0, _DRCRe,
    0,
    [_RC],
    [[() => RiskConfigurationType$, 0]], 1
];
exports.DescribeRiskConfigurationResponse$ = DescribeRiskConfigurationResponse$;
const DescribeTermsRequest$ = [3, n0, _DTRe,
    0,
    [_TI, _UPI],
    [0, 0], 2
];
exports.DescribeTermsRequest$ = DescribeTermsRequest$;
const DescribeTermsResponse$ = [3, n0, _DTRes,
    0,
    [_T],
    [[() => TermsType$, 0]]
];
exports.DescribeTermsResponse$ = DescribeTermsResponse$;
const DescribeUserImportJobRequest$ = [3, n0, _DUIJR,
    0,
    [_UPI, _JI],
    [0, 0], 2
];
exports.DescribeUserImportJobRequest$ = DescribeUserImportJobRequest$;
const DescribeUserImportJobResponse$ = [3, n0, _DUIJRe,
    0,
    [_UIJ],
    [() => UserImportJobType$]
];
exports.DescribeUserImportJobResponse$ = DescribeUserImportJobResponse$;
const DescribeUserPoolClientRequest$ = [3, n0, _DUPCRe,
    0,
    [_UPI, _CI],
    [0, [() => ClientIdType, 0]], 2
];
exports.DescribeUserPoolClientRequest$ = DescribeUserPoolClientRequest$;
const DescribeUserPoolClientResponse$ = [3, n0, _DUPCRes,
    0,
    [_UPC],
    [[() => UserPoolClientType$, 0]]
];
exports.DescribeUserPoolClientResponse$ = DescribeUserPoolClientResponse$;
const DescribeUserPoolDomainRequest$ = [3, n0, _DUPDRes,
    0,
    [_Do],
    [0], 1
];
exports.DescribeUserPoolDomainRequest$ = DescribeUserPoolDomainRequest$;
const DescribeUserPoolDomainResponse$ = [3, n0, _DUPDResc,
    0,
    [_DD],
    [() => DomainDescriptionType$]
];
exports.DescribeUserPoolDomainResponse$ = DescribeUserPoolDomainResponse$;
const DescribeUserPoolRequest$ = [3, n0, _DUPRe,
    0,
    [_UPI],
    [0], 1
];
exports.DescribeUserPoolRequest$ = DescribeUserPoolRequest$;
const DescribeUserPoolResponse$ = [3, n0, _DUPRes,
    0,
    [_UP],
    [() => UserPoolType$]
];
exports.DescribeUserPoolResponse$ = DescribeUserPoolResponse$;
const DeviceConfigurationType$ = [3, n0, _DCT,
    0,
    [_CROND, _DOROUP],
    [2, 2]
];
exports.DeviceConfigurationType$ = DeviceConfigurationType$;
const DeviceSecretVerifierConfigType$ = [3, n0, _DSVCT,
    0,
    [_PV, _Sa],
    [0, 0]
];
exports.DeviceSecretVerifierConfigType$ = DeviceSecretVerifierConfigType$;
const DeviceType$ = [3, n0, _DT,
    0,
    [_DK, _DA, _DCD, _DLMD, _DLAD],
    [0, [() => AttributeListType, 0], 4, 4, 4]
];
exports.DeviceType$ = DeviceType$;
const DomainDescriptionType$ = [3, n0, _DDT,
    0,
    [_UPI, _AWSAI, _Do, _SB, _CFDl, _Ve, _St, _CDC, _MLV, _R],
    [0, 0, 0, 0, 0, 0, 0, () => CustomDomainConfigType$, 1, () => RoutingType$]
];
exports.DomainDescriptionType$ = DomainDescriptionType$;
const EmailConfigurationType$ = [3, n0, _ECT,
    0,
    [_SA, _RTEA, _ESA, _F, _CSo],
    [0, 0, 0, 0, 0]
];
exports.EmailConfigurationType$ = EmailConfigurationType$;
const EmailMfaConfigType$ = [3, n0, _EMCT,
    0,
    [_M, _Su],
    [0, 0]
];
exports.EmailMfaConfigType$ = EmailMfaConfigType$;
const EmailMfaSettingsType$ = [3, n0, _EMST,
    0,
    [_E, _PM],
    [2, 2]
];
exports.EmailMfaSettingsType$ = EmailMfaSettingsType$;
const EventContextDataType$ = [3, n0, _ECDT,
    0,
    [_IA, _DN, _Ti, _Ci, _Co],
    [0, 0, 0, 0, 0]
];
exports.EventContextDataType$ = EventContextDataType$;
const EventFeedbackType$ = [3, n0, _EFT,
    0,
    [_FV, _Pro, _FD],
    [0, 0, 4], 2
];
exports.EventFeedbackType$ = EventFeedbackType$;
const EventRiskType$ = [3, n0, _ERT,
    0,
    [_RD, _RL, _CCD],
    [0, 0, 2]
];
exports.EventRiskType$ = EventRiskType$;
const FailoverType$ = [3, n0, _FT,
    0,
    [_SR, _PRHCI],
    [0, 0], 2
];
exports.FailoverType$ = FailoverType$;
const FirehoseConfigurationType$ = [3, n0, _FCT,
    0,
    [_SAt],
    [0]
];
exports.FirehoseConfigurationType$ = FirehoseConfigurationType$;
const ForgetDeviceRequest$ = [3, n0, _FDR,
    0,
    [_DK, _ATc],
    [0, [() => TokenModelType, 0]], 1
];
exports.ForgetDeviceRequest$ = ForgetDeviceRequest$;
const ForgotPasswordRequest$ = [3, n0, _FPR,
    0,
    [_CI, _U, _SH, _UCDs, _AM, _CM],
    [[() => ClientIdType, 0], [() => UsernameType, 0], [() => SecretHashType, 0], [() => UserContextDataType$, 0], () => AnalyticsMetadataType$, 128 | 0], 2
];
exports.ForgotPasswordRequest$ = ForgotPasswordRequest$;
const ForgotPasswordResponse$ = [3, n0, _FPRo,
    0,
    [_CDD],
    [() => CodeDeliveryDetailsType$]
];
exports.ForgotPasswordResponse$ = ForgotPasswordResponse$;
const GetCSVHeaderRequest$ = [3, n0, _GCSVHR,
    0,
    [_UPI],
    [0], 1
];
exports.GetCSVHeaderRequest$ = GetCSVHeaderRequest$;
const GetCSVHeaderResponse$ = [3, n0, _GCSVHRe,
    0,
    [_UPI, _CSVH],
    [0, 64 | 0]
];
exports.GetCSVHeaderResponse$ = GetCSVHeaderResponse$;
const GetDeviceRequest$ = [3, n0, _GDR,
    0,
    [_DK, _ATc],
    [0, [() => TokenModelType, 0]], 1
];
exports.GetDeviceRequest$ = GetDeviceRequest$;
const GetDeviceResponse$ = [3, n0, _GDRe,
    0,
    [_D],
    [[() => DeviceType$, 0]], 1
];
exports.GetDeviceResponse$ = GetDeviceResponse$;
const GetGroupRequest$ = [3, n0, _GGR,
    0,
    [_GN, _UPI],
    [0, 0], 2
];
exports.GetGroupRequest$ = GetGroupRequest$;
const GetGroupResponse$ = [3, n0, _GGRe,
    0,
    [_Gr],
    [() => GroupType$]
];
exports.GetGroupResponse$ = GetGroupResponse$;
const GetIdentityProviderByIdentifierRequest$ = [3, n0, _GIPBIR,
    0,
    [_UPI, _IId],
    [0, 0], 2
];
exports.GetIdentityProviderByIdentifierRequest$ = GetIdentityProviderByIdentifierRequest$;
const GetIdentityProviderByIdentifierResponse$ = [3, n0, _GIPBIRe,
    0,
    [_IP],
    [() => IdentityProviderType$], 1
];
exports.GetIdentityProviderByIdentifierResponse$ = GetIdentityProviderByIdentifierResponse$;
const GetLogDeliveryConfigurationRequest$ = [3, n0, _GLDCR,
    0,
    [_UPI],
    [0], 1
];
exports.GetLogDeliveryConfigurationRequest$ = GetLogDeliveryConfigurationRequest$;
const GetLogDeliveryConfigurationResponse$ = [3, n0, _GLDCRe,
    0,
    [_LDC],
    [() => LogDeliveryConfigurationType$]
];
exports.GetLogDeliveryConfigurationResponse$ = GetLogDeliveryConfigurationResponse$;
const GetSigningCertificateRequest$ = [3, n0, _GSCR,
    0,
    [_UPI],
    [0], 1
];
exports.GetSigningCertificateRequest$ = GetSigningCertificateRequest$;
const GetSigningCertificateResponse$ = [3, n0, _GSCRe,
    0,
    [_Ce],
    [0]
];
exports.GetSigningCertificateResponse$ = GetSigningCertificateResponse$;
const GetTokensFromRefreshTokenRequest$ = [3, n0, _GTFRTR,
    0,
    [_RT, _CI, _CS, _DK, _CM],
    [[() => TokenModelType, 0], [() => ClientIdType, 0], [() => ClientSecretType, 0], 0, 128 | 0], 2
];
exports.GetTokensFromRefreshTokenRequest$ = GetTokensFromRefreshTokenRequest$;
const GetTokensFromRefreshTokenResponse$ = [3, n0, _GTFRTRe,
    0,
    [_AR],
    [[() => AuthenticationResultType$, 0]]
];
exports.GetTokensFromRefreshTokenResponse$ = GetTokensFromRefreshTokenResponse$;
const GetUICustomizationRequest$ = [3, n0, _GUICR,
    0,
    [_UPI, _CI],
    [0, [() => ClientIdType, 0]], 1
];
exports.GetUICustomizationRequest$ = GetUICustomizationRequest$;
const GetUICustomizationResponse$ = [3, n0, _GUICRe,
    0,
    [_UIC],
    [[() => UICustomizationType$, 0]], 1
];
exports.GetUICustomizationResponse$ = GetUICustomizationResponse$;
const GetUserAttributeVerificationCodeRequest$ = [3, n0, _GUAVCR,
    0,
    [_ATc, _AN, _CM],
    [[() => TokenModelType, 0], 0, 128 | 0], 2
];
exports.GetUserAttributeVerificationCodeRequest$ = GetUserAttributeVerificationCodeRequest$;
const GetUserAttributeVerificationCodeResponse$ = [3, n0, _GUAVCRe,
    0,
    [_CDD],
    [() => CodeDeliveryDetailsType$]
];
exports.GetUserAttributeVerificationCodeResponse$ = GetUserAttributeVerificationCodeResponse$;
const GetUserAuthFactorsRequest$ = [3, n0, _GUAFR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.GetUserAuthFactorsRequest$ = GetUserAuthFactorsRequest$;
const GetUserAuthFactorsResponse$ = [3, n0, _GUAFRe,
    0,
    [_U, _PMS, _UMFASL, _CUAF],
    [[() => UsernameType, 0], 0, 64 | 0, 64 | 0], 1
];
exports.GetUserAuthFactorsResponse$ = GetUserAuthFactorsResponse$;
const GetUserPoolMfaConfigRequest$ = [3, n0, _GUPMCR,
    0,
    [_UPI],
    [0], 1
];
exports.GetUserPoolMfaConfigRequest$ = GetUserPoolMfaConfigRequest$;
const GetUserPoolMfaConfigResponse$ = [3, n0, _GUPMCRe,
    0,
    [_SMC, _STMC, _EMC, _MC, _WAC],
    [() => SmsMfaConfigType$, () => SoftwareTokenMfaConfigType$, () => EmailMfaConfigType$, 0, () => WebAuthnConfigurationType$]
];
exports.GetUserPoolMfaConfigResponse$ = GetUserPoolMfaConfigResponse$;
const GetUserRequest$ = [3, n0, _GUR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.GetUserRequest$ = GetUserRequest$;
const GetUserResponse$ = [3, n0, _GURe,
    0,
    [_U, _UA, _MFAO, _PMS, _UMFASL],
    [[() => UsernameType, 0], [() => AttributeListType, 0], () => MFAOptionListType, 0, 64 | 0], 2
];
exports.GetUserResponse$ = GetUserResponse$;
const GlobalSignOutRequest$ = [3, n0, _GSOR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.GlobalSignOutRequest$ = GlobalSignOutRequest$;
const GlobalSignOutResponse$ = [3, n0, _GSORl,
    0,
    [],
    []
];
exports.GlobalSignOutResponse$ = GlobalSignOutResponse$;
const GroupType$ = [3, n0, _GT,
    0,
    [_GN, _UPI, _Desc, _RA, _Pr, _LMD, _CDr],
    [0, 0, 0, 0, 1, 4, 4]
];
exports.GroupType$ = GroupType$;
const HttpHeader$ = [3, n0, _HHt,
    0,
    [_hN, _hV],
    [0, 0]
];
exports.HttpHeader$ = HttpHeader$;
const IdentityProviderType$ = [3, n0, _IPT,
    0,
    [_UPI, _PN, _PTr, _PD, _AMt, _II, _LMD, _CDr],
    [0, 0, 0, 128 | 0, 128 | 0, 64 | 0, 4, 4]
];
exports.IdentityProviderType$ = IdentityProviderType$;
const InboundFederationLambdaType$ = [3, n0, _IFLT,
    0,
    [_LV, _LAa],
    [0, 0], 2
];
exports.InboundFederationLambdaType$ = InboundFederationLambdaType$;
const InitiateAuthRequest$ = [3, n0, _IAR,
    0,
    [_AF, _CI, _AP, _CM, _AM, _UCDs, _S],
    [0, [() => ClientIdType, 0], [() => AuthParametersType, 0], 128 | 0, () => AnalyticsMetadataType$, [() => UserContextDataType$, 0], [() => SessionType, 0]], 2
];
exports.InitiateAuthRequest$ = InitiateAuthRequest$;
const InitiateAuthResponse$ = [3, n0, _IARn,
    0,
    [_CN, _S, _CP, _AR, _AC],
    [0, [() => SessionType, 0], 128 | 0, [() => AuthenticationResultType$, 0], 64 | 0]
];
exports.InitiateAuthResponse$ = InitiateAuthResponse$;
const IssuerConfigurationType$ = [3, n0, _ICT,
    0,
    [_Ty],
    [0]
];
exports.IssuerConfigurationType$ = IssuerConfigurationType$;
const KeyConfigurationType$ = [3, n0, _KCT,
    0,
    [_KT, _KKA],
    [0, 0]
];
exports.KeyConfigurationType$ = KeyConfigurationType$;
const LambdaConfigType$ = [3, n0, _LCT,
    0,
    [_PSU, _CMu, _PC, _PA, _PAo, _DAC, _CAC, _VACR, _PTG, _UM, _PTGC, _CSMSS, _CES, _KMSKID, _IF],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, () => PreTokenGenerationVersionConfigType$, () => CustomSMSLambdaVersionConfigType$, () => CustomEmailLambdaVersionConfigType$, 0, () => InboundFederationLambdaType$]
];
exports.LambdaConfigType$ = LambdaConfigType$;
const ListDevicesRequest$ = [3, n0, _LDR,
    0,
    [_ATc, _L, _PTa],
    [[() => TokenModelType, 0], 1, 0], 1
];
exports.ListDevicesRequest$ = ListDevicesRequest$;
const ListDevicesResponse$ = [3, n0, _LDRi,
    0,
    [_De, _PTa],
    [[() => DeviceListType, 0], 0]
];
exports.ListDevicesResponse$ = ListDevicesResponse$;
const ListGroupsRequest$ = [3, n0, _LGR,
    0,
    [_UPI, _L, _NT],
    [0, 1, 0], 1
];
exports.ListGroupsRequest$ = ListGroupsRequest$;
const ListGroupsResponse$ = [3, n0, _LGRi,
    0,
    [_G, _NT],
    [() => GroupListType, 0]
];
exports.ListGroupsResponse$ = ListGroupsResponse$;
const ListIdentityProvidersRequest$ = [3, n0, _LIPR,
    0,
    [_UPI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListIdentityProvidersRequest$ = ListIdentityProvidersRequest$;
const ListIdentityProvidersResponse$ = [3, n0, _LIPRi,
    0,
    [_Prov, _NT],
    [() => ProvidersListType, 0], 1
];
exports.ListIdentityProvidersResponse$ = ListIdentityProvidersResponse$;
const ListResourceServersRequest$ = [3, n0, _LRSR,
    0,
    [_UPI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListResourceServersRequest$ = ListResourceServersRequest$;
const ListResourceServersResponse$ = [3, n0, _LRSRi,
    0,
    [_RSe, _NT],
    [() => ResourceServersListType, 0], 1
];
exports.ListResourceServersResponse$ = ListResourceServersResponse$;
const ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RAes],
    [0], 1
];
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
const ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [128 | 0]
];
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
const ListTermsRequest$ = [3, n0, _LTR,
    0,
    [_UPI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListTermsRequest$ = ListTermsRequest$;
const ListTermsResponse$ = [3, n0, _LTRi,
    0,
    [_T, _NT],
    [() => TermsDescriptionListType, 0], 1
];
exports.ListTermsResponse$ = ListTermsResponse$;
const ListUserImportJobsRequest$ = [3, n0, _LUIJR,
    0,
    [_UPI, _MR, _PTa],
    [0, 1, 0], 2
];
exports.ListUserImportJobsRequest$ = ListUserImportJobsRequest$;
const ListUserImportJobsResponse$ = [3, n0, _LUIJRi,
    0,
    [_UIJs, _PTa],
    [() => UserImportJobsListType, 0]
];
exports.ListUserImportJobsResponse$ = ListUserImportJobsResponse$;
const ListUserPoolClientSecretsRequest$ = [3, n0, _LUPCSR,
    0,
    [_UPI, _CI, _NT],
    [0, [() => ClientIdType, 0], 0], 2
];
exports.ListUserPoolClientSecretsRequest$ = ListUserPoolClientSecretsRequest$;
const ListUserPoolClientSecretsResponse$ = [3, n0, _LUPCSRi,
    0,
    [_CSl, _NT],
    [[() => ClientSecretDescriptorListType, 0], 0]
];
exports.ListUserPoolClientSecretsResponse$ = ListUserPoolClientSecretsResponse$;
const ListUserPoolClientsRequest$ = [3, n0, _LUPCR,
    0,
    [_UPI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListUserPoolClientsRequest$ = ListUserPoolClientsRequest$;
const ListUserPoolClientsResponse$ = [3, n0, _LUPCRi,
    0,
    [_UPCs, _NT],
    [[() => UserPoolClientListType, 0], 0]
];
exports.ListUserPoolClientsResponse$ = ListUserPoolClientsResponse$;
const ListUserPoolReplicasRequest$ = [3, n0, _LUPRR,
    0,
    [_UPI, _NT],
    [0, 0], 1
];
exports.ListUserPoolReplicasRequest$ = ListUserPoolReplicasRequest$;
const ListUserPoolReplicasResponse$ = [3, n0, _LUPRRi,
    0,
    [_UPRs, _NT],
    [() => UserPoolReplicaListType, 0]
];
exports.ListUserPoolReplicasResponse$ = ListUserPoolReplicasResponse$;
const ListUserPoolsRequest$ = [3, n0, _LUPR,
    0,
    [_MR, _NT],
    [1, 0], 1
];
exports.ListUserPoolsRequest$ = ListUserPoolsRequest$;
const ListUserPoolsResponse$ = [3, n0, _LUPRi,
    0,
    [_UPs, _NT],
    [() => UserPoolListType, 0]
];
exports.ListUserPoolsResponse$ = ListUserPoolsResponse$;
const ListUsersInGroupRequest$ = [3, n0, _LUIGR,
    0,
    [_UPI, _GN, _L, _NT],
    [0, 0, 1, 0], 2
];
exports.ListUsersInGroupRequest$ = ListUsersInGroupRequest$;
const ListUsersInGroupResponse$ = [3, n0, _LUIGRi,
    0,
    [_Use, _NT],
    [[() => UsersListType, 0], 0]
];
exports.ListUsersInGroupResponse$ = ListUsersInGroupResponse$;
const ListUsersRequest$ = [3, n0, _LUR,
    0,
    [_UPI, _ATG, _L, _PTa, _Fi],
    [0, 64 | 0, 1, 0, 0], 1
];
exports.ListUsersRequest$ = ListUsersRequest$;
const ListUsersResponse$ = [3, n0, _LURi,
    0,
    [_Use, _PTa],
    [[() => UsersListType, 0], 0]
];
exports.ListUsersResponse$ = ListUsersResponse$;
const ListWebAuthnCredentialsRequest$ = [3, n0, _LWACR,
    0,
    [_ATc, _NT, _MR],
    [[() => TokenModelType, 0], 0, 1], 1
];
exports.ListWebAuthnCredentialsRequest$ = ListWebAuthnCredentialsRequest$;
const ListWebAuthnCredentialsResponse$ = [3, n0, _LWACRi,
    0,
    [_Cre, _NT],
    [() => WebAuthnCredentialDescriptionListType, 0], 1
];
exports.ListWebAuthnCredentialsResponse$ = ListWebAuthnCredentialsResponse$;
const LogConfigurationType$ = [3, n0, _LCTo,
    0,
    [_LL, _ES, _CWLC, _SCo, _FC],
    [0, 0, () => CloudWatchLogsConfigurationType$, () => S3ConfigurationType$, () => FirehoseConfigurationType$], 2
];
exports.LogConfigurationType$ = LogConfigurationType$;
const LogDeliveryConfigurationType$ = [3, n0, _LDCT,
    0,
    [_UPI, _LCo],
    [0, () => LogConfigurationListType], 2
];
exports.LogDeliveryConfigurationType$ = LogDeliveryConfigurationType$;
const ManagedLoginBrandingType$ = [3, n0, _MLBT,
    0,
    [_MLBI, _UPI, _UCPV, _Se, _As, _CDr, _LMD],
    [0, 0, 2, 15, () => AssetListType, 4, 4]
];
exports.ManagedLoginBrandingType$ = ManagedLoginBrandingType$;
const MessageTemplateType$ = [3, n0, _MTT,
    0,
    [_SMSM, _EM, _ESm],
    [0, 0, 0]
];
exports.MessageTemplateType$ = MessageTemplateType$;
const MFAOptionType$ = [3, n0, _MFAOT,
    0,
    [_DM, _AN],
    [0, 0]
];
exports.MFAOptionType$ = MFAOptionType$;
const NewDeviceMetadataType$ = [3, n0, _NDMT,
    0,
    [_DK, _DGK],
    [0, 0]
];
exports.NewDeviceMetadataType$ = NewDeviceMetadataType$;
const NotifyConfigurationType$ = [3, n0, _NCT,
    0,
    [_SA, _F, _RTe, _BE, _NAEo, _ME],
    [0, 0, 0, () => NotifyEmailType$, () => NotifyEmailType$, () => NotifyEmailType$], 1
];
exports.NotifyConfigurationType$ = NotifyConfigurationType$;
const NotifyEmailType$ = [3, n0, _NET,
    0,
    [_Su, _HB, _TB],
    [0, 0, 0], 1
];
exports.NotifyEmailType$ = NotifyEmailType$;
const NumberAttributeConstraintsType$ = [3, n0, _NACT,
    0,
    [_MV, _MVa],
    [0, 0]
];
exports.NumberAttributeConstraintsType$ = NumberAttributeConstraintsType$;
const PasswordPolicyType$ = [3, n0, _PPT,
    0,
    [_ML, _RU, _RLe, _RNe, _RSeq, _PHS, _TPVD],
    [1, 2, 2, 2, 2, 1, 1]
];
exports.PasswordPolicyType$ = PasswordPolicyType$;
const PreTokenGenerationVersionConfigType$ = [3, n0, _PTGVCT,
    0,
    [_LV, _LAa],
    [0, 0], 2
];
exports.PreTokenGenerationVersionConfigType$ = PreTokenGenerationVersionConfigType$;
const ProviderDescription$ = [3, n0, _PDr,
    0,
    [_PN, _PTr, _LMD, _CDr],
    [0, 0, 4, 4]
];
exports.ProviderDescription$ = ProviderDescription$;
const ProviderUserIdentifierType$ = [3, n0, _PUIT,
    0,
    [_PN, _PAN, _PAV],
    [0, 0, 0]
];
exports.ProviderUserIdentifierType$ = ProviderUserIdentifierType$;
const RecoveryOptionType$ = [3, n0, _ROT,
    0,
    [_Pri, _Na],
    [1, 0], 2
];
exports.RecoveryOptionType$ = RecoveryOptionType$;
const RefreshTokenRotationType$ = [3, n0, _RTRT,
    0,
    [_Fe, _RGPS],
    [0, 1], 1
];
exports.RefreshTokenRotationType$ = RefreshTokenRotationType$;
const ResendConfirmationCodeRequest$ = [3, n0, _RCCR,
    0,
    [_CI, _U, _SH, _UCDs, _AM, _CM],
    [[() => ClientIdType, 0], [() => UsernameType, 0], [() => SecretHashType, 0], [() => UserContextDataType$, 0], () => AnalyticsMetadataType$, 128 | 0], 2
];
exports.ResendConfirmationCodeRequest$ = ResendConfirmationCodeRequest$;
const ResendConfirmationCodeResponse$ = [3, n0, _RCCRe,
    0,
    [_CDD],
    [() => CodeDeliveryDetailsType$]
];
exports.ResendConfirmationCodeResponse$ = ResendConfirmationCodeResponse$;
const ResourceServerScopeType$ = [3, n0, _RSST,
    0,
    [_SNc, _SD],
    [0, 0], 2
];
exports.ResourceServerScopeType$ = ResourceServerScopeType$;
const ResourceServerType$ = [3, n0, _RST,
    0,
    [_UPI, _I, _Na, _Sc],
    [0, 0, 0, () => ResourceServerScopeListType]
];
exports.ResourceServerType$ = ResourceServerType$;
const RespondToAuthChallengeRequest$ = [3, n0, _RTACR,
    0,
    [_CI, _CN, _S, _CR, _AM, _UCDs, _CM],
    [[() => ClientIdType, 0], 0, [() => SessionType, 0], [() => ChallengeResponsesType, 0], () => AnalyticsMetadataType$, [() => UserContextDataType$, 0], 128 | 0], 2
];
exports.RespondToAuthChallengeRequest$ = RespondToAuthChallengeRequest$;
const RespondToAuthChallengeResponse$ = [3, n0, _RTACRe,
    0,
    [_CN, _S, _CP, _AR],
    [0, [() => SessionType, 0], 128 | 0, [() => AuthenticationResultType$, 0]]
];
exports.RespondToAuthChallengeResponse$ = RespondToAuthChallengeResponse$;
const RevokeTokenRequest$ = [3, n0, _RTRe,
    0,
    [_To, _CI, _CS],
    [[() => TokenModelType, 0], [() => ClientIdType, 0], [() => ClientSecretType, 0]], 2
];
exports.RevokeTokenRequest$ = RevokeTokenRequest$;
const RevokeTokenResponse$ = [3, n0, _RTRev,
    0,
    [],
    []
];
exports.RevokeTokenResponse$ = RevokeTokenResponse$;
const RiskConfigurationType$ = [3, n0, _RCT,
    0,
    [_UPI, _CI, _CCRC, _ATRC, _REC, _LMD],
    [0, [() => ClientIdType, 0], () => CompromisedCredentialsRiskConfigurationType$, () => AccountTakeoverRiskConfigurationType$, () => RiskExceptionConfigurationType$, 4]
];
exports.RiskConfigurationType$ = RiskConfigurationType$;
const RiskExceptionConfigurationType$ = [3, n0, _RECT,
    0,
    [_BIPRL, _SIPRL],
    [64 | 0, 64 | 0]
];
exports.RiskExceptionConfigurationType$ = RiskExceptionConfigurationType$;
const RoutingType$ = [3, n0, _RTo,
    0,
    [_Fa],
    [() => FailoverType$]
];
exports.RoutingType$ = RoutingType$;
const S3ConfigurationType$ = [3, n0, _SCTo,
    0,
    [_BA],
    [0]
];
exports.S3ConfigurationType$ = S3ConfigurationType$;
const SchemaAttributeType$ = [3, n0, _SAT,
    0,
    [_Na, _ADT, _DOA, _Mu, _Re, _NAC, _SAC],
    [0, 0, 2, 2, 2, () => NumberAttributeConstraintsType$, () => StringAttributeConstraintsType$]
];
exports.SchemaAttributeType$ = SchemaAttributeType$;
const SetLogDeliveryConfigurationRequest$ = [3, n0, _SLDCR,
    0,
    [_UPI, _LCo],
    [0, () => LogConfigurationListType], 2
];
exports.SetLogDeliveryConfigurationRequest$ = SetLogDeliveryConfigurationRequest$;
const SetLogDeliveryConfigurationResponse$ = [3, n0, _SLDCRe,
    0,
    [_LDC],
    [() => LogDeliveryConfigurationType$]
];
exports.SetLogDeliveryConfigurationResponse$ = SetLogDeliveryConfigurationResponse$;
const SetRiskConfigurationRequest$ = [3, n0, _SRCR,
    0,
    [_UPI, _CI, _CCRC, _ATRC, _REC],
    [0, [() => ClientIdType, 0], () => CompromisedCredentialsRiskConfigurationType$, () => AccountTakeoverRiskConfigurationType$, () => RiskExceptionConfigurationType$], 1
];
exports.SetRiskConfigurationRequest$ = SetRiskConfigurationRequest$;
const SetRiskConfigurationResponse$ = [3, n0, _SRCRe,
    0,
    [_RC],
    [[() => RiskConfigurationType$, 0]], 1
];
exports.SetRiskConfigurationResponse$ = SetRiskConfigurationResponse$;
const SetUICustomizationRequest$ = [3, n0, _SUICR,
    0,
    [_UPI, _CI, _CSS, _IFm],
    [0, [() => ClientIdType, 0], 0, 21], 1
];
exports.SetUICustomizationRequest$ = SetUICustomizationRequest$;
const SetUICustomizationResponse$ = [3, n0, _SUICRe,
    0,
    [_UIC],
    [[() => UICustomizationType$, 0]], 1
];
exports.SetUICustomizationResponse$ = SetUICustomizationResponse$;
const SetUserMFAPreferenceRequest$ = [3, n0, _SUMFAPR,
    0,
    [_ATc, _SMSMS, _STMS, _EMS, _WAMS],
    [[() => TokenModelType, 0], () => SMSMfaSettingsType$, () => SoftwareTokenMfaSettingsType$, () => EmailMfaSettingsType$, () => WebAuthnMfaSettingsType$], 1
];
exports.SetUserMFAPreferenceRequest$ = SetUserMFAPreferenceRequest$;
const SetUserMFAPreferenceResponse$ = [3, n0, _SUMFAPRe,
    0,
    [],
    []
];
exports.SetUserMFAPreferenceResponse$ = SetUserMFAPreferenceResponse$;
const SetUserPoolMfaConfigRequest$ = [3, n0, _SUPMCR,
    0,
    [_UPI, _SMC, _STMC, _EMC, _MC, _WAC],
    [0, () => SmsMfaConfigType$, () => SoftwareTokenMfaConfigType$, () => EmailMfaConfigType$, 0, () => WebAuthnConfigurationType$], 1
];
exports.SetUserPoolMfaConfigRequest$ = SetUserPoolMfaConfigRequest$;
const SetUserPoolMfaConfigResponse$ = [3, n0, _SUPMCRe,
    0,
    [_SMC, _STMC, _EMC, _MC, _WAC],
    [() => SmsMfaConfigType$, () => SoftwareTokenMfaConfigType$, () => EmailMfaConfigType$, 0, () => WebAuthnConfigurationType$]
];
exports.SetUserPoolMfaConfigResponse$ = SetUserPoolMfaConfigResponse$;
const SetUserSettingsRequest$ = [3, n0, _SUSR,
    0,
    [_ATc, _MFAO],
    [[() => TokenModelType, 0], () => MFAOptionListType], 2
];
exports.SetUserSettingsRequest$ = SetUserSettingsRequest$;
const SetUserSettingsResponse$ = [3, n0, _SUSRe,
    0,
    [],
    []
];
exports.SetUserSettingsResponse$ = SetUserSettingsResponse$;
const SignInPolicyType$ = [3, n0, _SIPT,
    0,
    [_AFAF],
    [64 | 0]
];
exports.SignInPolicyType$ = SignInPolicyType$;
const SignUpRequest$ = [3, n0, _SUR,
    0,
    [_CI, _U, _SH, _P, _UA, _VD, _AM, _UCDs, _CM],
    [[() => ClientIdType, 0], [() => UsernameType, 0], [() => SecretHashType, 0], [() => PasswordType, 0], [() => AttributeListType, 0], [() => AttributeListType, 0], () => AnalyticsMetadataType$, [() => UserContextDataType$, 0], 128 | 0], 2
];
exports.SignUpRequest$ = SignUpRequest$;
const SignUpResponse$ = [3, n0, _SURi,
    0,
    [_UCs, _USs, _CDD, _S],
    [2, 0, () => CodeDeliveryDetailsType$, [() => SessionType, 0]], 2
];
exports.SignUpResponse$ = SignUpResponse$;
const SmsConfigurationType$ = [3, n0, _SCTm,
    0,
    [_SCA, _EIx, _SRn],
    [0, 0, 0], 1
];
exports.SmsConfigurationType$ = SmsConfigurationType$;
const SmsMfaConfigType$ = [3, n0, _SMCT,
    0,
    [_SAM, _SCm],
    [0, () => SmsConfigurationType$]
];
exports.SmsMfaConfigType$ = SmsMfaConfigType$;
const SMSMfaSettingsType$ = [3, n0, _SMSMST,
    0,
    [_E, _PM],
    [2, 2]
];
exports.SMSMfaSettingsType$ = SMSMfaSettingsType$;
const SoftwareTokenMfaConfigType$ = [3, n0, _STMCT,
    0,
    [_E],
    [2]
];
exports.SoftwareTokenMfaConfigType$ = SoftwareTokenMfaConfigType$;
const SoftwareTokenMfaSettingsType$ = [3, n0, _STMST,
    0,
    [_E, _PM],
    [2, 2]
];
exports.SoftwareTokenMfaSettingsType$ = SoftwareTokenMfaSettingsType$;
const StartUserImportJobRequest$ = [3, n0, _SUIJR,
    0,
    [_UPI, _JI],
    [0, 0], 2
];
exports.StartUserImportJobRequest$ = StartUserImportJobRequest$;
const StartUserImportJobResponse$ = [3, n0, _SUIJRt,
    0,
    [_UIJ],
    [() => UserImportJobType$]
];
exports.StartUserImportJobResponse$ = StartUserImportJobResponse$;
const StartWebAuthnRegistrationRequest$ = [3, n0, _SWARR,
    0,
    [_ATc],
    [[() => TokenModelType, 0]], 1
];
exports.StartWebAuthnRegistrationRequest$ = StartWebAuthnRegistrationRequest$;
const StartWebAuthnRegistrationResponse$ = [3, n0, _SWARRt,
    0,
    [_CCO],
    [15], 1
];
exports.StartWebAuthnRegistrationResponse$ = StartWebAuthnRegistrationResponse$;
const StopUserImportJobRequest$ = [3, n0, _SUIJRto,
    0,
    [_UPI, _JI],
    [0, 0], 2
];
exports.StopUserImportJobRequest$ = StopUserImportJobRequest$;
const StopUserImportJobResponse$ = [3, n0, _SUIJRtop,
    0,
    [_UIJ],
    [() => UserImportJobType$]
];
exports.StopUserImportJobResponse$ = StopUserImportJobResponse$;
const StringAttributeConstraintsType$ = [3, n0, _SACT,
    0,
    [_MLi, _MLa],
    [0, 0]
];
exports.StringAttributeConstraintsType$ = StringAttributeConstraintsType$;
const TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RAes, _Ta],
    [0, 128 | 0], 2
];
exports.TagResourceRequest$ = TagResourceRequest$;
const TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
exports.TagResourceResponse$ = TagResourceResponse$;
const TermsDescriptionType$ = [3, n0, _TDT,
    0,
    [_TI, _TN, _En, _CDr, _LMD],
    [0, 0, 0, 4, 4], 5
];
exports.TermsDescriptionType$ = TermsDescriptionType$;
const TermsType$ = [3, n0, _TTe,
    0,
    [_TI, _UPI, _CI, _TN, _TS, _En, _Li, _CDr, _LMD],
    [0, 0, [() => ClientIdType, 0], 0, 0, 0, 128 | 0, 4, 4], 9
];
exports.TermsType$ = TermsType$;
const TokenValidityUnitsType$ = [3, n0, _TVUT,
    0,
    [_ATc, _IT, _RT],
    [0, 0, 0]
];
exports.TokenValidityUnitsType$ = TokenValidityUnitsType$;
const UICustomizationType$ = [3, n0, _UICT,
    0,
    [_UPI, _CI, _IU, _CSS, _CSSV, _LMD, _CDr],
    [0, [() => ClientIdType, 0], 0, 0, 0, 4, 4]
];
exports.UICustomizationType$ = UICustomizationType$;
const UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RAes, _TK],
    [0, 64 | 0], 2
];
exports.UntagResourceRequest$ = UntagResourceRequest$;
const UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
exports.UntagResourceResponse$ = UntagResourceResponse$;
const UpdateAuthEventFeedbackRequest$ = [3, n0, _UAEFR,
    0,
    [_UPI, _U, _EI, _FTe, _FV],
    [0, [() => UsernameType, 0], 0, [() => TokenModelType, 0], 0], 5
];
exports.UpdateAuthEventFeedbackRequest$ = UpdateAuthEventFeedbackRequest$;
const UpdateAuthEventFeedbackResponse$ = [3, n0, _UAEFRp,
    0,
    [],
    []
];
exports.UpdateAuthEventFeedbackResponse$ = UpdateAuthEventFeedbackResponse$;
const UpdateDeviceStatusRequest$ = [3, n0, _UDSR,
    0,
    [_ATc, _DK, _DRS],
    [[() => TokenModelType, 0], 0, 0], 2
];
exports.UpdateDeviceStatusRequest$ = UpdateDeviceStatusRequest$;
const UpdateDeviceStatusResponse$ = [3, n0, _UDSRp,
    0,
    [],
    []
];
exports.UpdateDeviceStatusResponse$ = UpdateDeviceStatusResponse$;
const UpdateGroupRequest$ = [3, n0, _UGR,
    0,
    [_GN, _UPI, _Desc, _RA, _Pr],
    [0, 0, 0, 0, 1], 2
];
exports.UpdateGroupRequest$ = UpdateGroupRequest$;
const UpdateGroupResponse$ = [3, n0, _UGRp,
    0,
    [_Gr],
    [() => GroupType$]
];
exports.UpdateGroupResponse$ = UpdateGroupResponse$;
const UpdateIdentityProviderRequest$ = [3, n0, _UIPR,
    0,
    [_UPI, _PN, _PD, _AMt, _II],
    [0, 0, 128 | 0, 128 | 0, 64 | 0], 2
];
exports.UpdateIdentityProviderRequest$ = UpdateIdentityProviderRequest$;
const UpdateIdentityProviderResponse$ = [3, n0, _UIPRp,
    0,
    [_IP],
    [() => IdentityProviderType$], 1
];
exports.UpdateIdentityProviderResponse$ = UpdateIdentityProviderResponse$;
const UpdateManagedLoginBrandingRequest$ = [3, n0, _UMLBR,
    0,
    [_UPI, _MLBI, _UCPV, _Se, _As],
    [0, 0, 2, 15, () => AssetListType]
];
exports.UpdateManagedLoginBrandingRequest$ = UpdateManagedLoginBrandingRequest$;
const UpdateManagedLoginBrandingResponse$ = [3, n0, _UMLBRp,
    0,
    [_MLB],
    [() => ManagedLoginBrandingType$]
];
exports.UpdateManagedLoginBrandingResponse$ = UpdateManagedLoginBrandingResponse$;
const UpdateResourceServerRequest$ = [3, n0, _URSR,
    0,
    [_UPI, _I, _Na, _Sc],
    [0, 0, 0, () => ResourceServerScopeListType], 3
];
exports.UpdateResourceServerRequest$ = UpdateResourceServerRequest$;
const UpdateResourceServerResponse$ = [3, n0, _URSRp,
    0,
    [_RS],
    [() => ResourceServerType$], 1
];
exports.UpdateResourceServerResponse$ = UpdateResourceServerResponse$;
const UpdateTermsRequest$ = [3, n0, _UTR,
    0,
    [_TI, _UPI, _TN, _TS, _En, _Li],
    [0, 0, 0, 0, 0, 128 | 0], 2
];
exports.UpdateTermsRequest$ = UpdateTermsRequest$;
const UpdateTermsResponse$ = [3, n0, _UTRp,
    0,
    [_T],
    [[() => TermsType$, 0]]
];
exports.UpdateTermsResponse$ = UpdateTermsResponse$;
const UpdateUserAttributesRequest$ = [3, n0, _UUAR,
    0,
    [_UA, _ATc, _CM],
    [[() => AttributeListType, 0], [() => TokenModelType, 0], 128 | 0], 2
];
exports.UpdateUserAttributesRequest$ = UpdateUserAttributesRequest$;
const UpdateUserAttributesResponse$ = [3, n0, _UUARp,
    0,
    [_CDDL],
    [() => CodeDeliveryDetailsListType]
];
exports.UpdateUserAttributesResponse$ = UpdateUserAttributesResponse$;
const UpdateUserPoolClientRequest$ = [3, n0, _UUPCR,
    0,
    [_UPI, _CI, _CNl, _RTV, _ATV, _ITV, _TVU, _RAe, _WA, _EAF, _SIP, _CURL, _LURL, _DRURI, _AOAF, _AOAS, _AOAFUPC, _ACn, _PUEE, _ETR, _EPAUCD, _ASV, _RTR],
    [0, [() => ClientIdType, 0], 0, 1, 1, 1, () => TokenValidityUnitsType$, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 0, 64 | 0, 64 | 0, 2, () => AnalyticsConfigurationType$, 0, 2, 2, 1, () => RefreshTokenRotationType$], 2
];
exports.UpdateUserPoolClientRequest$ = UpdateUserPoolClientRequest$;
const UpdateUserPoolClientResponse$ = [3, n0, _UUPCRp,
    0,
    [_UPC],
    [[() => UserPoolClientType$, 0]]
];
exports.UpdateUserPoolClientResponse$ = UpdateUserPoolClientResponse$;
const UpdateUserPoolDomainRequest$ = [3, n0, _UUPDR,
    0,
    [_Do, _UPI, _MLV, _CDC, _R],
    [0, 0, 1, () => CustomDomainConfigType$, () => RoutingType$], 2
];
exports.UpdateUserPoolDomainRequest$ = UpdateUserPoolDomainRequest$;
const UpdateUserPoolDomainResponse$ = [3, n0, _UUPDRp,
    0,
    [_MLV, _CFD, _R],
    [1, 0, () => RoutingType$]
];
exports.UpdateUserPoolDomainResponse$ = UpdateUserPoolDomainResponse$;
const UpdateUserPoolReplicaRequest$ = [3, n0, _UUPRR,
    0,
    [_UPI, _RN, _St],
    [0, 0, 0], 3
];
exports.UpdateUserPoolReplicaRequest$ = UpdateUserPoolReplicaRequest$;
const UpdateUserPoolReplicaResponse$ = [3, n0, _UUPRRp,
    0,
    [_UPR],
    [() => UserPoolReplicaType$]
];
exports.UpdateUserPoolReplicaResponse$ = UpdateUserPoolReplicaResponse$;
const UpdateUserPoolRequest$ = [3, n0, _UUPR,
    0,
    [_UPI, _Po, _DP, _LC, _AVA, _SVM, _EVM, _EVS, _VMT, _SAM, _UAUS, _MC, _DC, _EC, _SCm, _UPT, _ACUC, _UPAO, _ARS, _PNo, _UPTs, _KC, _IC],
    [0, () => UserPoolPolicyType$, 0, () => LambdaConfigType$, 64 | 0, 0, 0, 0, () => VerificationMessageTemplateType$, 0, () => UserAttributeUpdateSettingsType$, 0, () => DeviceConfigurationType$, () => EmailConfigurationType$, () => SmsConfigurationType$, 128 | 0, () => AdminCreateUserConfigType$, () => UserPoolAddOnsType$, () => AccountRecoverySettingType$, 0, 0, () => KeyConfigurationType$, () => IssuerConfigurationType$], 1
];
exports.UpdateUserPoolRequest$ = UpdateUserPoolRequest$;
const UpdateUserPoolResponse$ = [3, n0, _UUPRp,
    0,
    [],
    []
];
exports.UpdateUserPoolResponse$ = UpdateUserPoolResponse$;
const UserAttributeUpdateSettingsType$ = [3, n0, _UAUST,
    0,
    [_ARVBU],
    [64 | 0]
];
exports.UserAttributeUpdateSettingsType$ = UserAttributeUpdateSettingsType$;
const UserContextDataType$ = [3, n0, _UCDT,
    8,
    [_IA, _ED],
    [0, 0]
];
exports.UserContextDataType$ = UserContextDataType$;
const UserImportJobType$ = [3, n0, _UIJT,
    0,
    [_JN, _JI, _UPI, _PSUr, _CDr, _SDt, _CDo, _St, _CWLRA, _IUm, _SUk, _FU, _CMom],
    [0, 0, 0, 0, 4, 4, 4, 0, 0, 1, 1, 1, 0]
];
exports.UserImportJobType$ = UserImportJobType$;
const UsernameConfigurationType$ = [3, n0, _UCT,
    0,
    [_CSa],
    [2], 1
];
exports.UsernameConfigurationType$ = UsernameConfigurationType$;
const UserPoolAddOnsType$ = [3, n0, _UPAOT,
    0,
    [_ASM, _ASAF],
    [0, () => AdvancedSecurityAdditionalFlowsType$], 1
];
exports.UserPoolAddOnsType$ = UserPoolAddOnsType$;
const UserPoolClientDescription$ = [3, n0, _UPCD,
    0,
    [_CI, _UPI, _CNl],
    [[() => ClientIdType, 0], 0, 0]
];
exports.UserPoolClientDescription$ = UserPoolClientDescription$;
const UserPoolClientType$ = [3, n0, _UPCT,
    0,
    [_UPI, _CNl, _CI, _CS, _LMD, _CDr, _RTV, _ATV, _ITV, _TVU, _RAe, _WA, _EAF, _SIP, _CURL, _LURL, _DRURI, _AOAF, _AOAS, _AOAFUPC, _ACn, _PUEE, _ETR, _EPAUCD, _ASV, _RTR],
    [0, 0, [() => ClientIdType, 0], [() => ClientSecretType, 0], 4, 4, 1, 1, 1, () => TokenValidityUnitsType$, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 64 | 0, 0, 64 | 0, 64 | 0, 2, () => AnalyticsConfigurationType$, 0, 2, 2, 1, () => RefreshTokenRotationType$]
];
exports.UserPoolClientType$ = UserPoolClientType$;
const UserPoolDescriptionType$ = [3, n0, _UPDT,
    0,
    [_Id, _Na, _LC, _St, _LMD, _CDr, _RR],
    [0, 0, () => LambdaConfigType$, 0, 4, 4, 64 | 0]
];
exports.UserPoolDescriptionType$ = UserPoolDescriptionType$;
const UserPoolPolicyType$ = [3, n0, _UPPT,
    0,
    [_PPa, _SIPi],
    [() => PasswordPolicyType$, () => SignInPolicyType$]
];
exports.UserPoolPolicyType$ = UserPoolPolicyType$;
const UserPoolReplicaType$ = [3, n0, _UPRT,
    0,
    [_RN, _St, _Ro, _UPA],
    [0, 0, 0, 0]
];
exports.UserPoolReplicaType$ = UserPoolReplicaType$;
const UserPoolType$ = [3, n0, _UPTse,
    0,
    [_Id, _Na, _Po, _DP, _LC, _St, _LMD, _CDr, _SAc, _AVA, _AAl, _UAs, _SVM, _EVM, _EVS, _VMT, _SAM, _UAUS, _MC, _DC, _ENOU, _EC, _SCm, _UPT, _SCF, _ECF, _Do, _CDu, _ACUC, _UPAO, _UC, _Ar, _ARS, _UPTs, _KC, _IC],
    [0, 0, () => UserPoolPolicyType$, 0, () => LambdaConfigType$, 0, 4, 4, () => SchemaAttributesListType, 64 | 0, 64 | 0, 64 | 0, 0, 0, 0, () => VerificationMessageTemplateType$, 0, () => UserAttributeUpdateSettingsType$, 0, () => DeviceConfigurationType$, 1, () => EmailConfigurationType$, () => SmsConfigurationType$, 128 | 0, 0, 0, 0, 0, () => AdminCreateUserConfigType$, () => UserPoolAddOnsType$, () => UsernameConfigurationType$, 0, () => AccountRecoverySettingType$, 0, () => KeyConfigurationType$, () => IssuerConfigurationType$]
];
exports.UserPoolType$ = UserPoolType$;
const UserType$ = [3, n0, _UTs,
    0,
    [_U, _At, _UCD, _ULMD, _E, _US, _MFAO],
    [[() => UsernameType, 0], [() => AttributeListType, 0], 4, 4, 2, 0, () => MFAOptionListType]
];
exports.UserType$ = UserType$;
const VerificationMessageTemplateType$ = [3, n0, _VMTT,
    0,
    [_SM, _EM, _ESm, _EMBL, _ESBL, _DEO],
    [0, 0, 0, 0, 0, 0]
];
exports.VerificationMessageTemplateType$ = VerificationMessageTemplateType$;
const VerifySoftwareTokenRequest$ = [3, n0, _VSTR,
    0,
    [_UCse, _ATc, _S, _FDN],
    [[() => SoftwareTokenMFAUserCodeType, 0], [() => TokenModelType, 0], [() => SessionType, 0], 0], 1
];
exports.VerifySoftwareTokenRequest$ = VerifySoftwareTokenRequest$;
const VerifySoftwareTokenResponse$ = [3, n0, _VSTRe,
    0,
    [_St, _S],
    [0, [() => SessionType, 0]]
];
exports.VerifySoftwareTokenResponse$ = VerifySoftwareTokenResponse$;
const VerifyUserAttributeRequest$ = [3, n0, _VUAR,
    0,
    [_ATc, _AN, _Cod],
    [[() => TokenModelType, 0], 0, 0], 3
];
exports.VerifyUserAttributeRequest$ = VerifyUserAttributeRequest$;
const VerifyUserAttributeResponse$ = [3, n0, _VUARe,
    0,
    [],
    []
];
exports.VerifyUserAttributeResponse$ = VerifyUserAttributeResponse$;
const WebAuthnConfigurationType$ = [3, n0, _WACT,
    0,
    [_RPI, _UV, _FCa],
    [0, 0, 0]
];
exports.WebAuthnConfigurationType$ = WebAuthnConfigurationType$;
const WebAuthnCredentialDescription$ = [3, n0, _WACD,
    0,
    [_CIr, _FCN, _RPI, _ATu, _CAr, _AAu],
    [0, 0, 0, 64 | 0, 4, 0], 5
];
exports.WebAuthnCredentialDescription$ = WebAuthnCredentialDescription$;
const WebAuthnMfaSettingsType$ = [3, n0, _WAMST,
    0,
    [_E],
    [2]
];
exports.WebAuthnMfaSettingsType$ = WebAuthnMfaSettingsType$;
var __Unit = "unit";
var AliasAttributesListType = 64 | 0;
var AllowedFirstAuthFactorsListType = 64 | 0;
var AssetListType = [1, n0, _ALT,
    0, () => AssetType$
];
var AttributeListType = [1, n0, _ALTt,
    0, [() => AttributeType$,
        0]
];
var AttributeNameListType = 64 | 0;
var AttributesRequireVerificationBeforeUpdateType = 64 | 0;
var AuthEventsType = [1, n0, _AETu,
    0, () => AuthEventType$
];
var AvailableChallengeListType = 64 | 0;
var BlockedIPRangeListType = 64 | 0;
var CallbackURLsListType = 64 | 0;
var ChallengeResponseListType = [1, n0, _CRLT,
    0, () => ChallengeResponseType$
];
var ClientPermissionListType = 64 | 0;
var ClientSecretDescriptorListType = [1, n0, _CSDLT,
    0, [() => ClientSecretDescriptorType$,
        0]
];
var CodeDeliveryDetailsListType = [1, n0, _CDDLT,
    0, () => CodeDeliveryDetailsType$
];
var ConfiguredUserAuthFactorsListType = 64 | 0;
var CustomAttributesListType = [1, n0, _CALT,
    0, () => SchemaAttributeType$
];
var DeliveryMediumListType = 64 | 0;
var DeviceListType = [1, n0, _DLT,
    0, [() => DeviceType$,
        0]
];
var EventFiltersType = 64 | 0;
var ExplicitAuthFlowsListType = 64 | 0;
var GroupListType = [1, n0, _GLT,
    0, () => GroupType$
];
var HttpHeaderList = [1, n0, _HHL,
    0, () => HttpHeader$
];
var IdpIdentifiersListType = 64 | 0;
var ListOfStringTypes = 64 | 0;
var LogConfigurationListType = [1, n0, _LCLT,
    0, () => LogConfigurationType$
];
var LogoutURLsListType = 64 | 0;
var MFAOptionListType = [1, n0, _MFAOLT,
    0, () => MFAOptionType$
];
var OAuthFlowsType = 64 | 0;
var ProvidersListType = [1, n0, _PLT,
    0, () => ProviderDescription$
];
var RecoveryMechanismsType = [1, n0, _RMT,
    0, () => RecoveryOptionType$
];
var ReplicaRegionsType = 64 | 0;
var ResourceServerScopeListType = [1, n0, _RSSLT,
    0, () => ResourceServerScopeType$
];
var ResourceServersListType = [1, n0, _RSLT,
    0, () => ResourceServerType$
];
var SchemaAttributesListType = [1, n0, _SALT,
    0, () => SchemaAttributeType$
];
var ScopeListType = 64 | 0;
var SearchedAttributeNamesListType = 64 | 0;
var SkippedIPRangeListType = 64 | 0;
var SupportedIdentityProvidersListType = 64 | 0;
var TermsDescriptionListType = [1, n0, _TDLT,
    0, () => TermsDescriptionType$
];
var UserImportJobsListType = [1, n0, _UIJLT,
    0, () => UserImportJobType$
];
var UserMFASettingListType = 64 | 0;
var UsernameAttributesListType = 64 | 0;
var UserPoolClientListType = [1, n0, _UPCLT,
    0, [() => UserPoolClientDescription$,
        0]
];
var UserPoolListType = [1, n0, _UPLT,
    0, () => UserPoolDescriptionType$
];
var UserPoolReplicaListType = [1, n0, _UPRLT,
    0, () => UserPoolReplicaType$
];
var UserPoolTagsListType = 64 | 0;
var UsersListType = [1, n0, _ULT,
    0, [() => UserType$,
        0]
];
var VerifiedAttributesListType = 64 | 0;
var WebAuthnAuthenticatorTransportsList = 64 | 0;
var WebAuthnCredentialDescriptionListType = [1, n0, _WACDLT,
    0, () => WebAuthnCredentialDescription$
];
var AttributeMappingType = 128 | 0;
var AuthParametersType = [2, n0, _APT,
    8, 0, 0
];
var ChallengeParametersType = 128 | 0;
var ChallengeResponsesType = [2, n0, _CRTh,
    8, 0, 0
];
var ClientMetadataType = 128 | 0;
var LinksType = 128 | 0;
var ProviderDetailsType = 128 | 0;
var UserPoolTagsType = 128 | 0;
exports.AddCustomAttributes$ = [9, n0, _ACA,
    0, () => AddCustomAttributesRequest$, () => AddCustomAttributesResponse$
];
exports.AddUserPoolClientSecret$ = [9, n0, _AUPCS,
    0, () => AddUserPoolClientSecretRequest$, () => AddUserPoolClientSecretResponse$
];
exports.AdminAddUserToGroup$ = [9, n0, _AAUTG,
    0, () => AdminAddUserToGroupRequest$, () => __Unit
];
exports.AdminConfirmSignUp$ = [9, n0, _ACSU,
    0, () => AdminConfirmSignUpRequest$, () => AdminConfirmSignUpResponse$
];
exports.AdminCreateUser$ = [9, n0, _ACU,
    0, () => AdminCreateUserRequest$, () => AdminCreateUserResponse$
];
exports.AdminDeleteUser$ = [9, n0, _ADU,
    0, () => AdminDeleteUserRequest$, () => __Unit
];
exports.AdminDeleteUserAttributes$ = [9, n0, _ADUA,
    0, () => AdminDeleteUserAttributesRequest$, () => AdminDeleteUserAttributesResponse$
];
exports.AdminDisableProviderForUser$ = [9, n0, _ADPFU,
    0, () => AdminDisableProviderForUserRequest$, () => AdminDisableProviderForUserResponse$
];
exports.AdminDisableUser$ = [9, n0, _ADUd,
    0, () => AdminDisableUserRequest$, () => AdminDisableUserResponse$
];
exports.AdminEnableUser$ = [9, n0, _AEU,
    0, () => AdminEnableUserRequest$, () => AdminEnableUserResponse$
];
exports.AdminForgetDevice$ = [9, n0, _AFD,
    0, () => AdminForgetDeviceRequest$, () => __Unit
];
exports.AdminGetDevice$ = [9, n0, _AGD,
    0, () => AdminGetDeviceRequest$, () => AdminGetDeviceResponse$
];
exports.AdminGetUser$ = [9, n0, _AGU,
    0, () => AdminGetUserRequest$, () => AdminGetUserResponse$
];
exports.AdminInitiateAuth$ = [9, n0, _AIA,
    0, () => AdminInitiateAuthRequest$, () => AdminInitiateAuthResponse$
];
exports.AdminLinkProviderForUser$ = [9, n0, _ALPFU,
    0, () => AdminLinkProviderForUserRequest$, () => AdminLinkProviderForUserResponse$
];
exports.AdminListDevices$ = [9, n0, _ALD,
    0, () => AdminListDevicesRequest$, () => AdminListDevicesResponse$
];
exports.AdminListGroupsForUser$ = [9, n0, _ALGFU,
    0, () => AdminListGroupsForUserRequest$, () => AdminListGroupsForUserResponse$
];
exports.AdminListUserAuthEvents$ = [9, n0, _ALUAE,
    0, () => AdminListUserAuthEventsRequest$, () => AdminListUserAuthEventsResponse$
];
exports.AdminRemoveUserFromGroup$ = [9, n0, _ARUFG,
    0, () => AdminRemoveUserFromGroupRequest$, () => __Unit
];
exports.AdminResetUserPassword$ = [9, n0, _ARUP,
    0, () => AdminResetUserPasswordRequest$, () => AdminResetUserPasswordResponse$
];
exports.AdminRespondToAuthChallenge$ = [9, n0, _ARTAC,
    0, () => AdminRespondToAuthChallengeRequest$, () => AdminRespondToAuthChallengeResponse$
];
exports.AdminSetUserMFAPreference$ = [9, n0, _ASUMFAP,
    0, () => AdminSetUserMFAPreferenceRequest$, () => AdminSetUserMFAPreferenceResponse$
];
exports.AdminSetUserPassword$ = [9, n0, _ASUP,
    0, () => AdminSetUserPasswordRequest$, () => AdminSetUserPasswordResponse$
];
exports.AdminSetUserSettings$ = [9, n0, _ASUS,
    0, () => AdminSetUserSettingsRequest$, () => AdminSetUserSettingsResponse$
];
exports.AdminUpdateAuthEventFeedback$ = [9, n0, _AUAEF,
    0, () => AdminUpdateAuthEventFeedbackRequest$, () => AdminUpdateAuthEventFeedbackResponse$
];
exports.AdminUpdateDeviceStatus$ = [9, n0, _AUDS,
    0, () => AdminUpdateDeviceStatusRequest$, () => AdminUpdateDeviceStatusResponse$
];
exports.AdminUpdateUserAttributes$ = [9, n0, _AUUA,
    0, () => AdminUpdateUserAttributesRequest$, () => AdminUpdateUserAttributesResponse$
];
exports.AdminUserGlobalSignOut$ = [9, n0, _AUGSO,
    0, () => AdminUserGlobalSignOutRequest$, () => AdminUserGlobalSignOutResponse$
];
exports.AssociateSoftwareToken$ = [9, n0, _AST,
    0, () => AssociateSoftwareTokenRequest$, () => AssociateSoftwareTokenResponse$
];
exports.ChangePassword$ = [9, n0, _CPh,
    0, () => ChangePasswordRequest$, () => ChangePasswordResponse$
];
exports.CompleteWebAuthnRegistration$ = [9, n0, _CWAR,
    0, () => CompleteWebAuthnRegistrationRequest$, () => CompleteWebAuthnRegistrationResponse$
];
exports.ConfirmDevice$ = [9, n0, _CDon,
    0, () => ConfirmDeviceRequest$, () => ConfirmDeviceResponse$
];
exports.ConfirmForgotPassword$ = [9, n0, _CFP,
    0, () => ConfirmForgotPasswordRequest$, () => ConfirmForgotPasswordResponse$
];
exports.ConfirmSignUp$ = [9, n0, _CSU,
    0, () => ConfirmSignUpRequest$, () => ConfirmSignUpResponse$
];
exports.CreateGroup$ = [9, n0, _CG,
    0, () => CreateGroupRequest$, () => CreateGroupResponse$
];
exports.CreateIdentityProvider$ = [9, n0, _CIP,
    0, () => CreateIdentityProviderRequest$, () => CreateIdentityProviderResponse$
];
exports.CreateManagedLoginBranding$ = [9, n0, _CMLB,
    0, () => CreateManagedLoginBrandingRequest$, () => CreateManagedLoginBrandingResponse$
];
exports.CreateResourceServer$ = [9, n0, _CRS,
    0, () => CreateResourceServerRequest$, () => CreateResourceServerResponse$
];
exports.CreateTerms$ = [9, n0, _CT,
    0, () => CreateTermsRequest$, () => CreateTermsResponse$
];
exports.CreateUserImportJob$ = [9, n0, _CUIJ,
    0, () => CreateUserImportJobRequest$, () => CreateUserImportJobResponse$
];
exports.CreateUserPool$ = [9, n0, _CUP,
    0, () => CreateUserPoolRequest$, () => CreateUserPoolResponse$
];
exports.CreateUserPoolClient$ = [9, n0, _CUPC,
    0, () => CreateUserPoolClientRequest$, () => CreateUserPoolClientResponse$
];
exports.CreateUserPoolDomain$ = [9, n0, _CUPD,
    0, () => CreateUserPoolDomainRequest$, () => CreateUserPoolDomainResponse$
];
exports.CreateUserPoolReplica$ = [9, n0, _CUPRre,
    0, () => CreateUserPoolReplicaRequest$, () => CreateUserPoolReplicaResponse$
];
exports.DeleteGroup$ = [9, n0, _DG,
    0, () => DeleteGroupRequest$, () => __Unit
];
exports.DeleteIdentityProvider$ = [9, n0, _DIP,
    0, () => DeleteIdentityProviderRequest$, () => __Unit
];
exports.DeleteManagedLoginBranding$ = [9, n0, _DMLB,
    0, () => DeleteManagedLoginBrandingRequest$, () => __Unit
];
exports.DeleteResourceServer$ = [9, n0, _DRSe,
    0, () => DeleteResourceServerRequest$, () => __Unit
];
exports.DeleteTerms$ = [9, n0, _DTe,
    0, () => DeleteTermsRequest$, () => __Unit
];
exports.DeleteUser$ = [9, n0, _DUe,
    0, () => DeleteUserRequest$, () => __Unit
];
exports.DeleteUserAttributes$ = [9, n0, _DUA,
    0, () => DeleteUserAttributesRequest$, () => DeleteUserAttributesResponse$
];
exports.DeleteUserPool$ = [9, n0, _DUP,
    0, () => DeleteUserPoolRequest$, () => __Unit
];
exports.DeleteUserPoolClient$ = [9, n0, _DUPC,
    0, () => DeleteUserPoolClientRequest$, () => __Unit
];
exports.DeleteUserPoolClientSecret$ = [9, n0, _DUPCS,
    0, () => DeleteUserPoolClientSecretRequest$, () => DeleteUserPoolClientSecretResponse$
];
exports.DeleteUserPoolDomain$ = [9, n0, _DUPD,
    0, () => DeleteUserPoolDomainRequest$, () => DeleteUserPoolDomainResponse$
];
exports.DeleteUserPoolReplica$ = [9, n0, _DUPRel,
    0, () => DeleteUserPoolReplicaRequest$, () => DeleteUserPoolReplicaResponse$
];
exports.DeleteWebAuthnCredential$ = [9, n0, _DWAC,
    0, () => DeleteWebAuthnCredentialRequest$, () => DeleteWebAuthnCredentialResponse$
];
exports.DescribeIdentityProvider$ = [9, n0, _DIPe,
    0, () => DescribeIdentityProviderRequest$, () => DescribeIdentityProviderResponse$
];
exports.DescribeManagedLoginBranding$ = [9, n0, _DMLBe,
    0, () => DescribeManagedLoginBrandingRequest$, () => DescribeManagedLoginBrandingResponse$
];
exports.DescribeManagedLoginBrandingByClient$ = [9, n0, _DMLBBC,
    0, () => DescribeManagedLoginBrandingByClientRequest$, () => DescribeManagedLoginBrandingByClientResponse$
];
exports.DescribeResourceServer$ = [9, n0, _DRSes,
    0, () => DescribeResourceServerRequest$, () => DescribeResourceServerResponse$
];
exports.DescribeRiskConfiguration$ = [9, n0, _DRC,
    0, () => DescribeRiskConfigurationRequest$, () => DescribeRiskConfigurationResponse$
];
exports.DescribeTerms$ = [9, n0, _DTes,
    0, () => DescribeTermsRequest$, () => DescribeTermsResponse$
];
exports.DescribeUserImportJob$ = [9, n0, _DUIJ,
    0, () => DescribeUserImportJobRequest$, () => DescribeUserImportJobResponse$
];
exports.DescribeUserPool$ = [9, n0, _DUPe,
    0, () => DescribeUserPoolRequest$, () => DescribeUserPoolResponse$
];
exports.DescribeUserPoolClient$ = [9, n0, _DUPCe,
    0, () => DescribeUserPoolClientRequest$, () => DescribeUserPoolClientResponse$
];
exports.DescribeUserPoolDomain$ = [9, n0, _DUPDe,
    0, () => DescribeUserPoolDomainRequest$, () => DescribeUserPoolDomainResponse$
];
exports.ForgetDevice$ = [9, n0, _FDo,
    0, () => ForgetDeviceRequest$, () => __Unit
];
exports.ForgotPassword$ = [9, n0, _FP,
    0, () => ForgotPasswordRequest$, () => ForgotPasswordResponse$
];
exports.GetCSVHeader$ = [9, n0, _GCSVH,
    0, () => GetCSVHeaderRequest$, () => GetCSVHeaderResponse$
];
exports.GetDevice$ = [9, n0, _GD,
    0, () => GetDeviceRequest$, () => GetDeviceResponse$
];
exports.GetGroup$ = [9, n0, _GG,
    0, () => GetGroupRequest$, () => GetGroupResponse$
];
exports.GetIdentityProviderByIdentifier$ = [9, n0, _GIPBI,
    0, () => GetIdentityProviderByIdentifierRequest$, () => GetIdentityProviderByIdentifierResponse$
];
exports.GetLogDeliveryConfiguration$ = [9, n0, _GLDC,
    0, () => GetLogDeliveryConfigurationRequest$, () => GetLogDeliveryConfigurationResponse$
];
exports.GetSigningCertificate$ = [9, n0, _GSC,
    0, () => GetSigningCertificateRequest$, () => GetSigningCertificateResponse$
];
exports.GetTokensFromRefreshToken$ = [9, n0, _GTFRT,
    0, () => GetTokensFromRefreshTokenRequest$, () => GetTokensFromRefreshTokenResponse$
];
exports.GetUICustomization$ = [9, n0, _GUIC,
    0, () => GetUICustomizationRequest$, () => GetUICustomizationResponse$
];
exports.GetUser$ = [9, n0, _GU,
    0, () => GetUserRequest$, () => GetUserResponse$
];
exports.GetUserAttributeVerificationCode$ = [9, n0, _GUAVC,
    0, () => GetUserAttributeVerificationCodeRequest$, () => GetUserAttributeVerificationCodeResponse$
];
exports.GetUserAuthFactors$ = [9, n0, _GUAF,
    0, () => GetUserAuthFactorsRequest$, () => GetUserAuthFactorsResponse$
];
exports.GetUserPoolMfaConfig$ = [9, n0, _GUPMC,
    0, () => GetUserPoolMfaConfigRequest$, () => GetUserPoolMfaConfigResponse$
];
exports.GlobalSignOut$ = [9, n0, _GSO,
    0, () => GlobalSignOutRequest$, () => GlobalSignOutResponse$
];
exports.InitiateAuth$ = [9, n0, _IAn,
    0, () => InitiateAuthRequest$, () => InitiateAuthResponse$
];
exports.ListDevices$ = [9, n0, _LD,
    0, () => ListDevicesRequest$, () => ListDevicesResponse$
];
exports.ListGroups$ = [9, n0, _LG,
    0, () => ListGroupsRequest$, () => ListGroupsResponse$
];
exports.ListIdentityProviders$ = [9, n0, _LIP,
    0, () => ListIdentityProvidersRequest$, () => ListIdentityProvidersResponse$
];
exports.ListResourceServers$ = [9, n0, _LRS,
    0, () => ListResourceServersRequest$, () => ListResourceServersResponse$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
exports.ListTerms$ = [9, n0, _LT,
    0, () => ListTermsRequest$, () => ListTermsResponse$
];
exports.ListUserImportJobs$ = [9, n0, _LUIJ,
    0, () => ListUserImportJobsRequest$, () => ListUserImportJobsResponse$
];
exports.ListUserPoolClients$ = [9, n0, _LUPC,
    0, () => ListUserPoolClientsRequest$, () => ListUserPoolClientsResponse$
];
exports.ListUserPoolClientSecrets$ = [9, n0, _LUPCS,
    0, () => ListUserPoolClientSecretsRequest$, () => ListUserPoolClientSecretsResponse$
];
exports.ListUserPoolReplicas$ = [9, n0, _LUPRis,
    0, () => ListUserPoolReplicasRequest$, () => ListUserPoolReplicasResponse$
];
exports.ListUserPools$ = [9, n0, _LUP,
    0, () => ListUserPoolsRequest$, () => ListUserPoolsResponse$
];
exports.ListUsers$ = [9, n0, _LU,
    0, () => ListUsersRequest$, () => ListUsersResponse$
];
exports.ListUsersInGroup$ = [9, n0, _LUIG,
    0, () => ListUsersInGroupRequest$, () => ListUsersInGroupResponse$
];
exports.ListWebAuthnCredentials$ = [9, n0, _LWAC,
    0, () => ListWebAuthnCredentialsRequest$, () => ListWebAuthnCredentialsResponse$
];
exports.ResendConfirmationCode$ = [9, n0, _RCC,
    0, () => ResendConfirmationCodeRequest$, () => ResendConfirmationCodeResponse$
];
exports.RespondToAuthChallenge$ = [9, n0, _RTAC,
    0, () => RespondToAuthChallengeRequest$, () => RespondToAuthChallengeResponse$
];
exports.RevokeToken$ = [9, n0, _RTev,
    0, () => RevokeTokenRequest$, () => RevokeTokenResponse$
];
exports.SetLogDeliveryConfiguration$ = [9, n0, _SLDC,
    0, () => SetLogDeliveryConfigurationRequest$, () => SetLogDeliveryConfigurationResponse$
];
exports.SetRiskConfiguration$ = [9, n0, _SRC,
    0, () => SetRiskConfigurationRequest$, () => SetRiskConfigurationResponse$
];
exports.SetUICustomization$ = [9, n0, _SUIC,
    0, () => SetUICustomizationRequest$, () => SetUICustomizationResponse$
];
exports.SetUserMFAPreference$ = [9, n0, _SUMFAP,
    0, () => SetUserMFAPreferenceRequest$, () => SetUserMFAPreferenceResponse$
];
exports.SetUserPoolMfaConfig$ = [9, n0, _SUPMC,
    0, () => SetUserPoolMfaConfigRequest$, () => SetUserPoolMfaConfigResponse$
];
exports.SetUserSettings$ = [9, n0, _SUS,
    0, () => SetUserSettingsRequest$, () => SetUserSettingsResponse$
];
exports.SignUp$ = [9, n0, _SUi,
    0, () => SignUpRequest$, () => SignUpResponse$
];
exports.StartUserImportJob$ = [9, n0, _SUIJ,
    0, () => StartUserImportJobRequest$, () => StartUserImportJobResponse$
];
exports.StartWebAuthnRegistration$ = [9, n0, _SWAR,
    0, () => StartWebAuthnRegistrationRequest$, () => StartWebAuthnRegistrationResponse$
];
exports.StopUserImportJob$ = [9, n0, _SUIJt,
    0, () => StopUserImportJobRequest$, () => StopUserImportJobResponse$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
exports.UpdateAuthEventFeedback$ = [9, n0, _UAEF,
    0, () => UpdateAuthEventFeedbackRequest$, () => UpdateAuthEventFeedbackResponse$
];
exports.UpdateDeviceStatus$ = [9, n0, _UDSp,
    0, () => UpdateDeviceStatusRequest$, () => UpdateDeviceStatusResponse$
];
exports.UpdateGroup$ = [9, n0, _UG,
    0, () => UpdateGroupRequest$, () => UpdateGroupResponse$
];
exports.UpdateIdentityProvider$ = [9, n0, _UIP,
    0, () => UpdateIdentityProviderRequest$, () => UpdateIdentityProviderResponse$
];
exports.UpdateManagedLoginBranding$ = [9, n0, _UMLB,
    0, () => UpdateManagedLoginBrandingRequest$, () => UpdateManagedLoginBrandingResponse$
];
exports.UpdateResourceServer$ = [9, n0, _URS,
    0, () => UpdateResourceServerRequest$, () => UpdateResourceServerResponse$
];
exports.UpdateTerms$ = [9, n0, _UTp,
    0, () => UpdateTermsRequest$, () => UpdateTermsResponse$
];
exports.UpdateUserAttributes$ = [9, n0, _UUA,
    0, () => UpdateUserAttributesRequest$, () => UpdateUserAttributesResponse$
];
exports.UpdateUserPool$ = [9, n0, _UUP,
    0, () => UpdateUserPoolRequest$, () => UpdateUserPoolResponse$
];
exports.UpdateUserPoolClient$ = [9, n0, _UUPC,
    0, () => UpdateUserPoolClientRequest$, () => UpdateUserPoolClientResponse$
];
exports.UpdateUserPoolDomain$ = [9, n0, _UUPD,
    0, () => UpdateUserPoolDomainRequest$, () => UpdateUserPoolDomainResponse$
];
exports.UpdateUserPoolReplica$ = [9, n0, _UUPRpd,
    0, () => UpdateUserPoolReplicaRequest$, () => UpdateUserPoolReplicaResponse$
];
exports.VerifySoftwareToken$ = [9, n0, _VST,
    0, () => VerifySoftwareTokenRequest$, () => VerifySoftwareTokenResponse$
];
exports.VerifyUserAttribute$ = [9, n0, _VUA,
    0, () => VerifyUserAttributeRequest$, () => VerifyUserAttributeResponse$
];
